{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "7f760e82",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import pandas as pd\n",
    "import yaml\n",
    "import uuid"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6eb10c2",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### C3N-05929 test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "643351b1",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>case</th>\n",
       "      <th>disease</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>data_format</th>\n",
       "      <th>reference</th>\n",
       "      <th>UUID</th>\n",
       "      <th>system</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3N-05929.WGS.T</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>WGS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3N-05929.WGS.N</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>WGS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample_name       case disease experimental_strategy   sample_type  \\\n",
       "0  C3N-05929.WGS.T  C3N-05929    LSCC                   WGS         tumor   \n",
       "1  C3N-05929.WGS.N  C3N-05929    LSCC                   WGS  blood_normal   \n",
       "\n",
       "                                           data_path  filesize data_format  \\\n",
       "0  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...  12345678         BAM   \n",
       "1  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...  12345678         BAM   \n",
       "\n",
       "  reference                                  UUID    system  \n",
       "0      hg38  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83  storage1  \n",
       "1      hg38  7d8f4ede-1911-456e-9eee-2fafe7132f50  storage1  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bammap = pd.read_csv('../tests/data/pecgs_TN_wgs_bam/input.bammap', sep='\\t')\n",
    "bammap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "f99dd1cd",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C3N-05929': {'wgs': {'tumor': ('/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/c5d518a3-fc69-4d7b-a719-0aad38abe43a/23fa88f9-9293-406b-bc21-2928ff12bc8b_wgs_gdc_realn.bam',\n",
       "    'd3ddd33b-c7a3-4caf-bdd7-c05e98a19d83'),\n",
       "   'normal': ('/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/823dda5b-ff15-4a66-b28b-de0632850623/e3bf416b-3d08-4d1b-bec4-808e55c7788c_wgs_gdc_realn.bam',\n",
       "    '7d8f4ede-1911-456e-9eee-2fafe7132f50')}}}"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cases = sorted(set(bammap['case']))\n",
    "inputs = {}\n",
    "for case in cases:\n",
    "    data = {\n",
    "        'wgs': {\n",
    "        }\n",
    "    }\n",
    "    filtered = bammap[bammap['case']==case]\n",
    "    # check for wxs tumor bam\n",
    "    f = filtered[[True if e=='WGS' and df=='BAM' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wgs tumor bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wgs']['tumor'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "    \n",
    "    # check for wxs normal bam\n",
    "    f = filtered[[True if e=='WGS' and df=='BAM' and 'normal' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wgs normal bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wgs']['normal'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "            \n",
    "    # if everything is present then accept sample\n",
    "    if 'tumor' in data['wgs'] and 'normal' in data['wgs']:\n",
    "        inputs[case] = data\n",
    "inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "102214a5",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C3N-05929': {'wgs_tumor_bam.filepath': '/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/c5d518a3-fc69-4d7b-a719-0aad38abe43a/23fa88f9-9293-406b-bc21-2928ff12bc8b_wgs_gdc_realn.bam',\n",
       "  'wgs_normal_bam.filepath': '/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/823dda5b-ff15-4a66-b28b-de0632850623/e3bf416b-3d08-4d1b-bec4-808e55c7788c_wgs_gdc_realn.bam',\n",
       "  'wgs_tumor_bam.uuid': 'd3ddd33b-c7a3-4caf-bdd7-c05e98a19d83',\n",
       "  'wgs_normal_bam.uuid': '7d8f4ede-1911-456e-9eee-2fafe7132f50'}}"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = {}\n",
    "for case, d in inputs.items():\n",
    "    run_list[case] = {f'{dt}_{st}_bam.filepath': fp\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()}\n",
    "    run_list[case].update({f'{dt}_{st}_bam.uuid': uid\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()})\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "c30d35a6",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wgs_normal_bam.filepath</th>\n",
       "      <th>wgs_normal_bam.uuid</th>\n",
       "      <th>wgs_tumor_bam.filepath</th>\n",
       "      <th>wgs_tumor_bam.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1</th>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>1fd2054c-fc78-433d-8f99-32f8e6909db1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                          wgs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                 wgs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  7d8f4ede-1911-456e-9eee-2fafe7132f50   \n",
       "\n",
       "                                                                           wgs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                  wgs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83   \n",
       "\n",
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  C3N-05929   \n",
       "\n",
       "                                                                            run_uuid  \n",
       "run_id                                                                                \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  1fd2054c-fc78-433d-8f99-32f8e6909db1  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_table = pd.DataFrame.from_dict(run_list).transpose()\n",
    "run_table['case_id'] = run_table.index.to_list()\n",
    "run_table['run_uuid'] = [str(uuid.uuid4()) for i in range(run_table.shape[0])]\n",
    "run_table.index = [f'{c}_{u}' for c, u in zip(run_table.index, run_table['run_uuid'])]\n",
    "run_table.index.name = 'run_id'\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "ac3c14f2",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'wgs_normal_bam.filepath', 'wgs_normal_bam.uuid', 'wgs_tumor_bam.filepath', 'wgs_tumor_bam.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_TN_wgs_bam/run_list.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74a9c1b0",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b50f9e6b",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### MMRF_1250 test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "5b6dc7e5",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>case</th>\n",
       "      <th>disease</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>data_format</th>\n",
       "      <th>reference</th>\n",
       "      <th>UUID</th>\n",
       "      <th>system</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00677.WXS.T</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WXS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>6e0bde4a-22c1-4e1f-a38a-dff275b61472</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00677.WXS.N</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WXS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>8919accb-6973-4ec5-b29f-9d895a1713d4</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R1.T</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2983df85-958b-483c-95e5-2375426879d1</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R2.T</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>f840b94b-eac9-4f87-98b6-51b26b606e1a</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     sample_name       case disease experimental_strategy  \\\n",
       "0                C3L-00677.WXS.T  C3L-00677     GBM                   WXS   \n",
       "1                C3L-00677.WXS.N  C3L-00677     GBM                   WXS   \n",
       "2  HT191P1-S1H1A3Y3.RNA-seq.R1.T  C3L-00677     GBM               RNA-seq   \n",
       "3  HT191P1-S1H1A3Y3.RNA-seq.R2.T  C3L-00677     GBM               RNA-seq   \n",
       "\n",
       "    sample_type                                          data_path  filesize  \\\n",
       "0         tumor  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "1  blood_normal  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "2         tumor  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "3         tumor  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "\n",
       "  data_format reference                                  UUID    system  \n",
       "0         BAM      hg38  6e0bde4a-22c1-4e1f-a38a-dff275b61472  storage1  \n",
       "1         BAM      hg38  8919accb-6973-4ec5-b29f-9d895a1713d4  storage1  \n",
       "2       FASTQ       NaN  2983df85-958b-483c-95e5-2375426879d1  storage1  \n",
       "3       FASTQ       NaN  f840b94b-eac9-4f87-98b6-51b26b606e1a  storage1  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bammap = pd.read_csv('../tests/data/pecgs_TN_wxs_bam/input.bammap', sep='\\t')\n",
    "bammap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "922d94bb",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C3L-00677': {'wxs': {'tumor': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/1d6c4f0f-09c4-4472-a19e-30ee7e0ecf64_gdc_realn.bam',\n",
       "    '6e0bde4a-22c1-4e1f-a38a-dff275b61472'),\n",
       "   'normal': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/5cee818c-61e8-4441-bd9a-5c984cde9c6b_gdc_realn.bam',\n",
       "    '8919accb-6973-4ec5-b29f-9d895a1713d4')}}}"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cases = sorted(set(bammap['case']))\n",
    "inputs = {}\n",
    "for case in cases:\n",
    "    data = {\n",
    "        'wxs': {\n",
    "        }\n",
    "    }\n",
    "    filtered = bammap[bammap['case']==case]\n",
    "    # check for wxs tumor bam\n",
    "    f = filtered[[True if e=='WXS' and df=='BAM' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wxs tumor bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wxs']['tumor'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "    \n",
    "    # check for wxs normal bam\n",
    "    f = filtered[[True if e=='WXS' and df=='BAM' and 'normal' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wxs normal bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wxs']['normal'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "            \n",
    "    # if everything is present then accept sample\n",
    "    if 'tumor' in data['wxs'] and 'normal' in data['wxs']:\n",
    "        inputs[case] = data\n",
    "inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "116a51f9",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C3L-00677': {'wxs_tumor_bam.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/1d6c4f0f-09c4-4472-a19e-30ee7e0ecf64_gdc_realn.bam',\n",
       "  'wxs_normal_bam.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/5cee818c-61e8-4441-bd9a-5c984cde9c6b_gdc_realn.bam',\n",
       "  'wxs_tumor_bam.uuid': '6e0bde4a-22c1-4e1f-a38a-dff275b61472',\n",
       "  'wxs_normal_bam.uuid': '8919accb-6973-4ec5-b29f-9d895a1713d4'}}"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = {}\n",
    "for case, d in inputs.items():\n",
    "    run_list[case] = {f'{dt}_{st}_bam.filepath': fp\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()}\n",
    "    run_list[case].update({f'{dt}_{st}_bam.uuid': uid\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()})\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "deaceac1",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wxs_normal_bam.filepath</th>\n",
       "      <th>wxs_normal_bam.uuid</th>\n",
       "      <th>wxs_tumor_bam.filepath</th>\n",
       "      <th>wxs_tumor_bam.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621</th>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>8919accb-6973-4ec5-b29f-9d895a1713d4</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>6e0bde4a-22c1-4e1f-a38a-dff275b61472</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>2a9edf1e-76df-4595-9125-fb9b9d919621</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                          wxs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                 wxs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  8919accb-6973-4ec5-b29f-9d895a1713d4   \n",
       "\n",
       "                                                                           wxs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                  wxs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  6e0bde4a-22c1-4e1f-a38a-dff275b61472   \n",
       "\n",
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  C3L-00677   \n",
       "\n",
       "                                                                            run_uuid  \n",
       "run_id                                                                                \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  2a9edf1e-76df-4595-9125-fb9b9d919621  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_table = pd.DataFrame.from_dict(run_list).transpose()\n",
    "run_table['case_id'] = run_table.index.to_list()\n",
    "run_table['run_uuid'] = [str(uuid.uuid4()) for i in range(run_table.shape[0])]\n",
    "run_table.index = [f'{c}_{u}' for c, u in zip(run_table.index, run_table['run_uuid'])]\n",
    "run_table.index.name = 'run_id'\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "a86410cc",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'wxs_normal_bam.filepath', 'wxs_normal_bam.uuid', 'wxs_tumor_bam.filepath', 'wxs_tumor_bam.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_TN_wxs_bam/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffd48545",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c55eb482",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### HT191P1 test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9aa00f98",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>case</th>\n",
       "      <th>disease</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>data_format</th>\n",
       "      <th>reference</th>\n",
       "      <th>UUID</th>\n",
       "      <th>system</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R1.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>78777884-505c-4b2f-9170-107662c1effa</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R2.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2699158f-25fe-4f19-a7cf-2a4e1cf687b3</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R1.N</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eafb3178-e459-44c0-a374-8e17a8107fce</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R2.N</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9923349f-dc3c-40d4-a097-803c4828f732</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R1.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fb1f4b8a-a56b-4bc6-b059-864c2b066dc8</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R2.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80ce8102-e8a4-4529-85b0-701a4dc022ea</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     sample_name              case disease  \\\n",
       "0      HT191P1-S1H1A3Y3.WXS.R1.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "1      HT191P1-S1H1A3Y3.WXS.R2.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "2      HT191P1-S1H1A3Y3.WXS.R1.N  HT191P1-S1H1A3Y3    PDAC   \n",
       "3      HT191P1-S1H1A3Y3.WXS.R2.N  HT191P1-S1H1A3Y3    PDAC   \n",
       "4  HT191P1-S1H1A3Y3.RNA-seq.R1.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "5  HT191P1-S1H1A3Y3.RNA-seq.R2.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "\n",
       "  experimental_strategy   sample_type  \\\n",
       "0                   WXS         tumor   \n",
       "1                   WXS         tumor   \n",
       "2                   WXS  blood_normal   \n",
       "3                   WXS  blood_normal   \n",
       "4               RNA-seq         tumor   \n",
       "5               RNA-seq         tumor   \n",
       "\n",
       "                                           data_path  filesize data_format  \\\n",
       "0  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...  12345678       FASTQ   \n",
       "1  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...  12345678       FASTQ   \n",
       "2  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...  12345678       FASTQ   \n",
       "3  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...  12345678       FASTQ   \n",
       "4  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...  12345678       FASTQ   \n",
       "5  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...  12345678       FASTQ   \n",
       "\n",
       "   reference                                  UUID    system  \n",
       "0        NaN  78777884-505c-4b2f-9170-107662c1effa  storage1  \n",
       "1        NaN  2699158f-25fe-4f19-a7cf-2a4e1cf687b3  storage1  \n",
       "2        NaN  eafb3178-e459-44c0-a374-8e17a8107fce  storage1  \n",
       "3        NaN  9923349f-dc3c-40d4-a097-803c4828f732  storage1  \n",
       "4        NaN  fb1f4b8a-a56b-4bc6-b059-864c2b066dc8  storage1  \n",
       "5        NaN  80ce8102-e8a4-4529-85b0-701a4dc022ea  storage1  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bammap = pd.read_csv('../tests/data/pecgs_TN_wxs_fq/input.bammap', sep='\\t')\n",
    "bammap"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "548e58a8",
   "metadata": {
    "hidden": true
   },
   "source": [
    "looking for all samples in bammap with all input files present (in this case paired WXS fastqs and rna-seq tumor fastqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "f64ff8e3",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HT191P1-S1H1A3Y3': {'wxs': {'tumor': {'R1': ('/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz',\n",
       "     '78777884-505c-4b2f-9170-107662c1effa'),\n",
       "    'R2': ('/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz',\n",
       "     '2699158f-25fe-4f19-a7cf-2a4e1cf687b3')},\n",
       "   'normal': {'R1': ('/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R1_001.fastq.gz',\n",
       "     'eafb3178-e459-44c0-a374-8e17a8107fce'),\n",
       "    'R2': ('/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R2_001.fastq.gz',\n",
       "     '9923349f-dc3c-40d4-a097-803c4828f732')}},\n",
       "  'rna-seq': {'tumor': {'R1': ('/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "     'fb1f4b8a-a56b-4bc6-b059-864c2b066dc8'),\n",
       "    'R2': ('/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R2.fastq.gz',\n",
       "     '80ce8102-e8a4-4529-85b0-701a4dc022ea')}}}}"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cases = sorted(set(bammap['case']))\n",
    "inputs = {}\n",
    "for case in cases:\n",
    "    data = {\n",
    "        'wxs': {\n",
    "            'tumor': {},\n",
    "            'normal': {}\n",
    "        },\n",
    "        'rna-seq': {\n",
    "            'tumor': {}\n",
    "        }\n",
    "    }\n",
    "    filtered = bammap[bammap['case']==case]\n",
    "    \n",
    "    # check for wxs tumor fastqs\n",
    "    f = filtered[[True if e=='WXS' and df=='FASTQ' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 2:\n",
    "        print(f'Warning: more than 2 wxs tumor fastqs found for case {case}')\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r1.' in row['sample_name'].lower():\n",
    "            data['wxs']['tumor']['R1'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r2.' in row['sample_name'].lower():\n",
    "            data['wxs']['tumor']['R2'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "            \n",
    "    # check for wxs normal fastqs\n",
    "    f = filtered[[True if e=='WXS' and df=='FASTQ' and 'normal' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 2:\n",
    "        print(f'Warning: more than 2 wxs normal fastqs found for case {case}')\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r1.' in row['sample_name'].lower():\n",
    "            data['wxs']['normal']['R1'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r2.' in row['sample_name'].lower():\n",
    "            data['wxs']['normal']['R2'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "            \n",
    "    # check for rna-seq tumor fastqs\n",
    "    f = filtered[[True if e=='RNA-seq' and df=='FASTQ' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 2:\n",
    "        print(f'Warning: more than 2 rna-seq tumor fastqs found for case {case}')\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r1.' in row['sample_name'].lower():\n",
    "            data['rna-seq']['tumor']['R1'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r2.' in row['sample_name'].lower():\n",
    "            data['rna-seq']['tumor']['R2'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "            \n",
    "    \n",
    "    # if everything is present then accept sample\n",
    "    # dont do in real life bc pipeline is split by input, but easier this way for test case\n",
    "    if len(data['wxs']['tumor']) == 2 and len(data['wxs']['normal']) == 2 and len(data['rna-seq']['tumor']) == 2:\n",
    "        inputs[case] = data\n",
    "inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "82b19743",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HT191P1-S1H1A3Y3': {'wxs_tumor_R1.filepath': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz',\n",
       "  'wxs_tumor_R2.filepath': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz',\n",
       "  'wxs_normal_R1.filepath': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R1_001.fastq.gz',\n",
       "  'wxs_normal_R2.filepath': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R2_001.fastq.gz',\n",
       "  'rna-seq_tumor_R1.filepath': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "  'rna-seq_tumor_R2.filepath': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R2.fastq.gz',\n",
       "  'wxs_tumor_R1.uuid': '78777884-505c-4b2f-9170-107662c1effa',\n",
       "  'wxs_tumor_R2.uuid': '2699158f-25fe-4f19-a7cf-2a4e1cf687b3',\n",
       "  'wxs_normal_R1.uuid': 'eafb3178-e459-44c0-a374-8e17a8107fce',\n",
       "  'wxs_normal_R2.uuid': '9923349f-dc3c-40d4-a097-803c4828f732',\n",
       "  'rna-seq_tumor_R1.uuid': 'fb1f4b8a-a56b-4bc6-b059-864c2b066dc8',\n",
       "  'rna-seq_tumor_R2.uuid': '80ce8102-e8a4-4529-85b0-701a4dc022ea'}}"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = {}\n",
    "for case, d in inputs.items():\n",
    "    run_list[case] = {f'{dt}_{st}_{r}.filepath': fp\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, d2 in d1.items()\n",
    "                     for r, (fp, uid) in d2.items()}\n",
    "    run_list[case].update({f'{dt}_{st}_{r}.uuid': uid\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, d2 in d1.items()\n",
    "                     for r, (fp, uid) in d2.items()})\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "302919f6",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5bb5519</th>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>fb1f4b8a-a56b-4bc6-b059-864c2b066dc8</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>80ce8102-e8a4-4529-85b0-701a4dc022ea</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>eafb3178-e459-44c0-a374-8e17a8107fce</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>9923349f-dc3c-40d4-a097-803c4828f732</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>78777884-505c-4b2f-9170-107662c1effa</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>2699158f-25fe-4f19-a7cf-2a4e1cf687b3</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>be6b68e6-365d-4451-8195-f1b3d5bb5519</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  fb1f4b8a-a56b-4bc6-b059-864c2b066dc8   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  80ce8102-e8a4-4529-85b0-701a4dc022ea   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  eafb3178-e459-44c0-a374-8e17a8107fce   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  9923349f-dc3c-40d4-a097-803c4828f732   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  78777884-505c-4b2f-9170-107662c1effa   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  2699158f-25fe-4f19-a7cf-2a4e1cf687b3   \n",
       "\n",
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  be6b68e6-365d-4451-8195-f1b3d5bb5519  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_table = pd.DataFrame.from_dict(run_list).transpose()\n",
    "run_table['case_id'] = run_table.index.to_list()\n",
    "run_table['run_uuid'] = [str(uuid.uuid4()) for i in range(run_table.shape[0])]\n",
    "run_table.index = [f'{c}_{u}' for c, u in zip(run_table.index, run_table['run_uuid'])]\n",
    "run_table.index.name = 'run_id'\n",
    "run_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bd95a01",
   "metadata": {
    "hidden": true
   },
   "source": [
    "wxs fastq runlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "511f0205",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'wxs_normal_R1.filepath', 'wxs_normal_R1.uuid', 'wxs_normal_R2.filepath', 'wxs_normal_R2.uuid', 'wxs_tumor_R1.filepath', 'wxs_tumor_R1.uuid', 'wxs_tumor_R2.filepath', 'wxs_tumor_R2.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_TN_wxs_fq/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0f0aa1a",
   "metadata": {
    "hidden": true
   },
   "source": [
    "rnaseq fq runlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "1933f042",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'rna-seq_tumor_R1.filepath', 'rna-seq_tumor_R1.uuid', 'rna-seq_tumor_R2.filepath', 'rna-seq_tumor_R2.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_T_rna_fq/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "893c438b",
   "metadata": {
    "hidden": true
   },
   "source": [
    "if we want the .bams to have correct readgroup ids, etc. then we need to pass some sequencing info. this info is available in the Samplemap.csv from MGI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8de72505",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Instrument Data ID</th>\n",
       "      <th>Flow Cell ID</th>\n",
       "      <th>Lane</th>\n",
       "      <th>Index Sequence</th>\n",
       "      <th>Status</th>\n",
       "      <th>Library Summary</th>\n",
       "      <th>Library Name</th>\n",
       "      <th>Library Type</th>\n",
       "      <th>Sample Full Name</th>\n",
       "      <th>Species Name</th>\n",
       "      <th>Creation Event ID</th>\n",
       "      <th>Creation Date</th>\n",
       "      <th>Completion Date</th>\n",
       "      <th>Gerald Bam Path</th>\n",
       "      <th>Full Path</th>\n",
       "      <th>File1</th>\n",
       "      <th>File2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2903151174</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>AAGAAGTCAC-ACGAATAAGG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011195</td>\n",
       "      <td>TWCE-HT110B1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT110B1-S1H1A3Y3D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153983</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>AAGAAGTCAC-ACGAATAAGG_S4_L002_R1_001.fastq.gz</td>\n",
       "      <td>AAGAAGTCAC-ACGAATAAGG_S4_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2903151239</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>AAGAGCTCGC-AGTAAGAGTC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011186</td>\n",
       "      <td>TWCE-HT110B1-S1H4A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT110B1-S1H4A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153994</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>AAGAGCTCGC-AGTAAGAGTC_S5_L002_R2_001.fastq.gz</td>\n",
       "      <td>AAGAGCTCGC-AGTAAGAGTC_S5_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2903151057</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>AAGATCCGTC-TTACACCAAC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011207</td>\n",
       "      <td>TWCE-HT088B1-S1H4A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT088B1-S1H4A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153965</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>AAGATCCGTC-TTACACCAAC_S6_L002_R2_001.fastq.gz</td>\n",
       "      <td>AAGATCCGTC-TTACACCAAC_S6_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2903151099</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>AAGCTAGCTG-CTTCCGTGAA</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011176</td>\n",
       "      <td>TWCE-HT067B1-S1H2A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT067B1-S1H2A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153968</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>AAGCTAGCTG-CTTCCGTGAA_S7_L002_R1_001.fastq.gz</td>\n",
       "      <td>AAGCTAGCTG-CTTCCGTGAA_S7_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2903151285</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>AATTAAGCGG-ATGGTCAGTA</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011173</td>\n",
       "      <td>TWCE-HT163B1-S1H6A3Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT163B1-S1H6A3Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154009</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>AATTAAGCGG-ATGGTCAGTA_S11_L002_R2_001.fastq.gz</td>\n",
       "      <td>AATTAAGCGG-ATGGTCAGTA_S11_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2903151242</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>ACTTGCATAC-ACATGAATGC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011194</td>\n",
       "      <td>TWCE-HT069B1-S1H4A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT069B1-S1H4A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154000</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>ACTTGCATAC-ACATGAATGC_S15_L002_R1_001.fastq.gz</td>\n",
       "      <td>ACTTGCATAC-ACATGAATGC_S15_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2903151337</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>ACTTGGCTTC-GCAGAGCACA</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011188</td>\n",
       "      <td>TWCE-HT154B1-S1H1A3Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT154B1-S1H1A3Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154022</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>ACTTGGCTTC-GCAGAGCACA_S16_L002_R2_001.fastq.gz</td>\n",
       "      <td>ACTTGGCTTC-GCAGAGCACA_S16_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2903151190</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>AGCTGAACTA-TGTGTCCAAC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011210</td>\n",
       "      <td>TWCE-HT190P1-S1H9A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT190P1-S1H9A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153989</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>AGCTGAACTA-TGTGTCCAAC_S28_L002_R2_001.fastq.gz</td>\n",
       "      <td>AGCTGAACTA-TGTGTCCAAC_S28_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2903150959</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CAATAGAGGC-CTTACGTATC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011204</td>\n",
       "      <td>TWCE-HT191P1-S1H2A3Y3D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT191P1-S1H2A3Y3D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153936</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CAATAGAGGC-CTTACGTATC_S32_L002_R1_001.fastq.gz</td>\n",
       "      <td>CAATAGAGGC-CTTACGTATC_S32_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2903151064</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CACATCAAGC-TGTATAGGCT</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011198</td>\n",
       "      <td>TWCE-HT154B1-S1H3A3Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT154B1-S1H3A3Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153966</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CACATCAAGC-TGTATAGGCT_S35_L002_R2_001.fastq.gz</td>\n",
       "      <td>CACATCAAGC-TGTATAGGCT_S35_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2903151310</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CACGGAATTG-TTACAGCCAG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011208</td>\n",
       "      <td>TWCE-HT068B1-S1H1A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT068B1-S1H1A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154014</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CACGGAATTG-TTACAGCCAG_S37_L002_R2_001.fastq.gz</td>\n",
       "      <td>CACGGAATTG-TTACAGCCAG_S37_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2903151312</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CAGTCACTTG-ACAGCCAGGT</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011229</td>\n",
       "      <td>TWCE-HT191P1-S1H4A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT191P1-S1H4A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154016</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CAGTCACTTG-ACAGCCAGGT_S43_L002_R1_001.fastq.gz</td>\n",
       "      <td>CAGTCACTTG-ACAGCCAGGT_S43_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2903150950</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CATTCCTAAC-TGTGGCGATT</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011220</td>\n",
       "      <td>TWCE-HT185P1-S1H3A3Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT185P1-S1H3A3Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153935</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CATTCCTAAC-TGTGGCGATT_S47_L002_R2_001.fastq.gz</td>\n",
       "      <td>CATTCCTAAC-TGTGGCGATT_S47_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2903151257</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAACTCTTA-GAAGTCCTTG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011212</td>\n",
       "      <td>TWCE-HT185P1-S1H6A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT185P1-S1H6A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154006</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CCAACTCTTA-GAAGTCCTTG_S50_L002_R2_001.fastq.gz</td>\n",
       "      <td>CCAACTCTTA-GAAGTCCTTG_S50_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2903151005</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCACAATTGC-GATGTGAAGT</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011222</td>\n",
       "      <td>TWCE-HT105B1-S1H3A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT105B1-S1H3A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153948</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CCACAATTGC-GATGTGAAGT_S51_L002_R1_001.fastq.gz</td>\n",
       "      <td>CCACAATTGC-GATGTGAAGT_S51_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2903151238</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011183</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153995</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2903151309</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCGATTATGA-ACCGGAGTAA</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011189</td>\n",
       "      <td>TWCE-HT163B1-S1H2A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT163B1-S1H2A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154013</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CCGATTATGA-ACCGGAGTAA_S57_L002_R1_001.fastq.gz</td>\n",
       "      <td>CCGATTATGA-ACCGGAGTAA_S57_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2903151253</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CGGATAACGG-TATCCTTCGC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011232</td>\n",
       "      <td>TWCE-HT190P1-S1H7A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT190P1-S1H7A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154004</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>CGGATAACGG-TATCCTTCGC_S70_L002_R1_001.fastq.gz</td>\n",
       "      <td>CGGATAACGG-TATCCTTCGC_S70_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2903151109</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>GTAAGCGAGG-AGTTCACCGG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011215</td>\n",
       "      <td>TWCE-HT144B1-S1H9A3Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT144B1-S1H9A3Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153974</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>GTAAGCGAGG-AGTTCACCGG_S75_L002_R2_001.fastq.gz</td>\n",
       "      <td>GTAAGCGAGG-AGTTCACCGG_S75_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2903154675</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGCATTAG-TACGATAAGC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011218</td>\n",
       "      <td>TWCE-HT144B1-S1H1A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT144B1-S1H1A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153969</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 22:45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17716...</td>\n",
       "      <td>GTCGCATTAG-TACGATAAGC_S80_L002_R2_001.fastq.gz</td>\n",
       "      <td>GTCGCATTAG-TACGATAAGC_S80_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2903151196</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGTTACCT-TTACTTCGTG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011192</td>\n",
       "      <td>TWCE-HT105B1-1679N1K2Y2D2_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT105B1-1679N1K2Y2D2_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153991</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>GTCGTTACCT-TTACTTCGTG_S81_L002_R2_001.fastq.gz</td>\n",
       "      <td>GTCGTTACCT-TTACTTCGTG_S81_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2903150930</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCTGAATAG-CTTGTCTTGC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011236</td>\n",
       "      <td>TWCE-HT171B1-S1H8A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT171B1-S1H8A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153930</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>GTCTGAATAG-CTTGTCTTGC_S82_L002_R2_001.fastq.gz</td>\n",
       "      <td>GTCTGAATAG-CTTGTCTTGC_S82_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2903151113</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>GTGAAGGCTG-GCACGATGCA</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011221</td>\n",
       "      <td>TWCE-HT137B1-S1H7A2K2Y3D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT137B1-S1H7A2K2Y3D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153978</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>GTGAAGGCTG-GCACGATGCA_S84_L002_R1_001.fastq.gz</td>\n",
       "      <td>GTGAAGGCTG-GCACGATGCA_S84_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2903151188</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TCAGCTGTCA-ATACAGGACG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011206</td>\n",
       "      <td>TWCE-HT088B1-S1H1A2K2Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT088B1-S1H1A2K2Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153985</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TCAGCTGTCA-ATACAGGACG_S85_L002_R2_001.fastq.gz</td>\n",
       "      <td>TCAGCTGTCA-ATACAGGACG_S85_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2903151302</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TCCTACAGCG-TGTGACCAGT</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011178</td>\n",
       "      <td>TWCE-HT035B1-PAH2Y2D2_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT035B1-PAH2Y2D2_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177154010</td>\n",
       "      <td>9/13/20 18:58</td>\n",
       "      <td>9/13/20 20:21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TCCTACAGCG-TGTGACCAGT_S91_L002_R2_001.fastq.gz</td>\n",
       "      <td>TCCTACAGCG-TGTGACCAGT_S91_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>2903151055</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TCCTGATGCG-TTCAGATTGC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011223</td>\n",
       "      <td>TWCE-HT171B1-S1H1A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT171B1-S1H1A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153961</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TCCTGATGCG-TTCAGATTGC_S92_L002_R2_001.fastq.gz</td>\n",
       "      <td>TCCTGATGCG-TTCAGATTGC_S92_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2903151067</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TCTCGCAGGA-GACGCAAGGT</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011201</td>\n",
       "      <td>TWCE-HT185P1-S1H2A3Y3D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT185P1-S1H2A3Y3D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153967</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TCTCGCAGGA-GACGCAAGGT_S103_L002_R2_001.fastq.gz</td>\n",
       "      <td>TCTCGCAGGA-GACGCAAGGT_S103_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2903151122</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TGAAGTCAAC-GAGGTAGTTC</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011228</td>\n",
       "      <td>TWCE-HT190P1-S1H3A3Y1D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT190P1-S1H3A3Y1D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153980</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 20:03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TGAAGTCAAC-GAGGTAGTTC_S104_L002_R1_001.fastq.gz</td>\n",
       "      <td>TGAAGTCAAC-GAGGTAGTTC_S104_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2903150994</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TGAGACTCGC-TAGGAAGCGG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011225</td>\n",
       "      <td>TWCE-HT110B1-1693N1K2Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT110B1-1693N1K2Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153943</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TGAGACTCGC-TAGGAAGCGG_S107_L002_R1_001.fastq.gz</td>\n",
       "      <td>TGAGACTCGC-TAGGAAGCGG_S107_L002_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>2903150971</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>TGATTGTCTG-TAAGCTGTAG</td>\n",
       "      <td>completed</td>\n",
       "      <td>2903011216</td>\n",
       "      <td>TWCE-HT154B1-S1H5A3Y2D1_1-lib1</td>\n",
       "      <td>genomic library</td>\n",
       "      <td>TWCE-HT154B1-S1H5A3Y2D1_1</td>\n",
       "      <td>human</td>\n",
       "      <td>177153940</td>\n",
       "      <td>9/13/20 18:57</td>\n",
       "      <td>9/13/20 19:49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gscmnt/gc13037/lims/2857177/2020-09/csf_17715...</td>\n",
       "      <td>TGATTGTCTG-TAAGCTGTAG_S108_L002_R2_001.fastq.gz</td>\n",
       "      <td>TGATTGTCTG-TAAGCTGTAG_S108_L002_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Instrument Data ID Flow Cell ID  Lane         Index Sequence     Status  \\\n",
       "0           2903151174    HFMFWDSXY     2  AAGAAGTCAC-ACGAATAAGG  completed   \n",
       "1           2903151239    HFMFWDSXY     2  AAGAGCTCGC-AGTAAGAGTC  completed   \n",
       "2           2903151057    HFMFWDSXY     2  AAGATCCGTC-TTACACCAAC  completed   \n",
       "3           2903151099    HFMFWDSXY     2  AAGCTAGCTG-CTTCCGTGAA  completed   \n",
       "4           2903151285    HFMFWDSXY     2  AATTAAGCGG-ATGGTCAGTA  completed   \n",
       "5           2903151242    HFMFWDSXY     2  ACTTGCATAC-ACATGAATGC  completed   \n",
       "6           2903151337    HFMFWDSXY     2  ACTTGGCTTC-GCAGAGCACA  completed   \n",
       "7           2903151190    HFMFWDSXY     2  AGCTGAACTA-TGTGTCCAAC  completed   \n",
       "8           2903150959    HFMFWDSXY     2  CAATAGAGGC-CTTACGTATC  completed   \n",
       "9           2903151064    HFMFWDSXY     2  CACATCAAGC-TGTATAGGCT  completed   \n",
       "10          2903151310    HFMFWDSXY     2  CACGGAATTG-TTACAGCCAG  completed   \n",
       "11          2903151312    HFMFWDSXY     2  CAGTCACTTG-ACAGCCAGGT  completed   \n",
       "12          2903150950    HFMFWDSXY     2  CATTCCTAAC-TGTGGCGATT  completed   \n",
       "13          2903151257    HFMFWDSXY     2  CCAACTCTTA-GAAGTCCTTG  completed   \n",
       "14          2903151005    HFMFWDSXY     2  CCACAATTGC-GATGTGAAGT  completed   \n",
       "15          2903151238    HFMFWDSXY     2  CCAGTAGCGT-ATGTATTGGC  completed   \n",
       "16          2903151309    HFMFWDSXY     2  CCGATTATGA-ACCGGAGTAA  completed   \n",
       "17          2903151253    HFMFWDSXY     2  CGGATAACGG-TATCCTTCGC  completed   \n",
       "18          2903151109    HFMFWDSXY     2  GTAAGCGAGG-AGTTCACCGG  completed   \n",
       "19          2903154675    HFMFWDSXY     2  GTCGCATTAG-TACGATAAGC  completed   \n",
       "20          2903151196    HFMFWDSXY     2  GTCGTTACCT-TTACTTCGTG  completed   \n",
       "21          2903150930    HFMFWDSXY     2  GTCTGAATAG-CTTGTCTTGC  completed   \n",
       "22          2903151113    HFMFWDSXY     2  GTGAAGGCTG-GCACGATGCA  completed   \n",
       "23          2903151188    HFMFWDSXY     2  TCAGCTGTCA-ATACAGGACG  completed   \n",
       "24          2903151302    HFMFWDSXY     2  TCCTACAGCG-TGTGACCAGT  completed   \n",
       "25          2903151055    HFMFWDSXY     2  TCCTGATGCG-TTCAGATTGC  completed   \n",
       "26          2903151067    HFMFWDSXY     2  TCTCGCAGGA-GACGCAAGGT  completed   \n",
       "27          2903151122    HFMFWDSXY     2  TGAAGTCAAC-GAGGTAGTTC  completed   \n",
       "28          2903150994    HFMFWDSXY     2  TGAGACTCGC-TAGGAAGCGG  completed   \n",
       "29          2903150971    HFMFWDSXY     2  TGATTGTCTG-TAAGCTGTAG  completed   \n",
       "\n",
       "    Library Summary                      Library Name     Library Type  \\\n",
       "0        2903011195    TWCE-HT110B1-S1H1A3Y3D1_1-lib1  genomic library   \n",
       "1        2903011186    TWCE-HT110B1-S1H4A3Y1D1_1-lib1  genomic library   \n",
       "2        2903011207    TWCE-HT088B1-S1H4A3Y1D1_1-lib1  genomic library   \n",
       "3        2903011176    TWCE-HT067B1-S1H2A3Y1D1_1-lib1  genomic library   \n",
       "4        2903011173    TWCE-HT163B1-S1H6A3Y2D1_1-lib1  genomic library   \n",
       "5        2903011194    TWCE-HT069B1-S1H4A3Y1D1_1-lib1  genomic library   \n",
       "6        2903011188    TWCE-HT154B1-S1H1A3Y2D1_1-lib1  genomic library   \n",
       "7        2903011210    TWCE-HT190P1-S1H9A3Y1D1_1-lib1  genomic library   \n",
       "8        2903011204    TWCE-HT191P1-S1H2A3Y3D1_1-lib1  genomic library   \n",
       "9        2903011198    TWCE-HT154B1-S1H3A3Y2D1_1-lib1  genomic library   \n",
       "10       2903011208    TWCE-HT068B1-S1H1A3Y1D1_1-lib1  genomic library   \n",
       "11       2903011229    TWCE-HT191P1-S1H4A3Y1D1_1-lib1  genomic library   \n",
       "12       2903011220    TWCE-HT185P1-S1H3A3Y2D1_1-lib1  genomic library   \n",
       "13       2903011212    TWCE-HT185P1-S1H6A3Y1D1_1-lib1  genomic library   \n",
       "14       2903011222    TWCE-HT105B1-S1H3A3Y1D1_1-lib1  genomic library   \n",
       "15       2903011183    TWCE-HT191P1-S1H1A3Y3D1_1-lib1  genomic library   \n",
       "16       2903011189    TWCE-HT163B1-S1H2A3Y1D1_1-lib1  genomic library   \n",
       "17       2903011232    TWCE-HT190P1-S1H7A3Y1D1_1-lib1  genomic library   \n",
       "18       2903011215    TWCE-HT144B1-S1H9A3Y2D1_1-lib1  genomic library   \n",
       "19       2903011218    TWCE-HT144B1-S1H1A3Y1D1_1-lib1  genomic library   \n",
       "20       2903011192  TWCE-HT105B1-1679N1K2Y2D2_1-lib1  genomic library   \n",
       "21       2903011236    TWCE-HT171B1-S1H8A3Y1D1_1-lib1  genomic library   \n",
       "22       2903011221  TWCE-HT137B1-S1H7A2K2Y3D1_1-lib1  genomic library   \n",
       "23       2903011206  TWCE-HT088B1-S1H1A2K2Y1D1_1-lib1  genomic library   \n",
       "24       2903011178      TWCE-HT035B1-PAH2Y2D2_1-lib1  genomic library   \n",
       "25       2903011223    TWCE-HT171B1-S1H1A3Y1D1_1-lib1  genomic library   \n",
       "26       2903011201    TWCE-HT185P1-S1H2A3Y3D1_1-lib1  genomic library   \n",
       "27       2903011228    TWCE-HT190P1-S1H3A3Y1D1_1-lib1  genomic library   \n",
       "28       2903011225  TWCE-HT110B1-1693N1K2Y2D1_1-lib1  genomic library   \n",
       "29       2903011216    TWCE-HT154B1-S1H5A3Y2D1_1-lib1  genomic library   \n",
       "\n",
       "               Sample Full Name Species Name  Creation Event ID  \\\n",
       "0     TWCE-HT110B1-S1H1A3Y3D1_1        human          177153983   \n",
       "1     TWCE-HT110B1-S1H4A3Y1D1_1        human          177153994   \n",
       "2     TWCE-HT088B1-S1H4A3Y1D1_1        human          177153965   \n",
       "3     TWCE-HT067B1-S1H2A3Y1D1_1        human          177153968   \n",
       "4     TWCE-HT163B1-S1H6A3Y2D1_1        human          177154009   \n",
       "5     TWCE-HT069B1-S1H4A3Y1D1_1        human          177154000   \n",
       "6     TWCE-HT154B1-S1H1A3Y2D1_1        human          177154022   \n",
       "7     TWCE-HT190P1-S1H9A3Y1D1_1        human          177153989   \n",
       "8     TWCE-HT191P1-S1H2A3Y3D1_1        human          177153936   \n",
       "9     TWCE-HT154B1-S1H3A3Y2D1_1        human          177153966   \n",
       "10    TWCE-HT068B1-S1H1A3Y1D1_1        human          177154014   \n",
       "11    TWCE-HT191P1-S1H4A3Y1D1_1        human          177154016   \n",
       "12    TWCE-HT185P1-S1H3A3Y2D1_1        human          177153935   \n",
       "13    TWCE-HT185P1-S1H6A3Y1D1_1        human          177154006   \n",
       "14    TWCE-HT105B1-S1H3A3Y1D1_1        human          177153948   \n",
       "15    TWCE-HT191P1-S1H1A3Y3D1_1        human          177153995   \n",
       "16    TWCE-HT163B1-S1H2A3Y1D1_1        human          177154013   \n",
       "17    TWCE-HT190P1-S1H7A3Y1D1_1        human          177154004   \n",
       "18    TWCE-HT144B1-S1H9A3Y2D1_1        human          177153974   \n",
       "19    TWCE-HT144B1-S1H1A3Y1D1_1        human          177153969   \n",
       "20  TWCE-HT105B1-1679N1K2Y2D2_1        human          177153991   \n",
       "21    TWCE-HT171B1-S1H8A3Y1D1_1        human          177153930   \n",
       "22  TWCE-HT137B1-S1H7A2K2Y3D1_1        human          177153978   \n",
       "23  TWCE-HT088B1-S1H1A2K2Y1D1_1        human          177153985   \n",
       "24      TWCE-HT035B1-PAH2Y2D2_1        human          177154010   \n",
       "25    TWCE-HT171B1-S1H1A3Y1D1_1        human          177153961   \n",
       "26    TWCE-HT185P1-S1H2A3Y3D1_1        human          177153967   \n",
       "27    TWCE-HT190P1-S1H3A3Y1D1_1        human          177153980   \n",
       "28  TWCE-HT110B1-1693N1K2Y2D1_1        human          177153943   \n",
       "29    TWCE-HT154B1-S1H5A3Y2D1_1        human          177153940   \n",
       "\n",
       "    Creation Date Completion Date  Gerald Bam Path  \\\n",
       "0   9/13/20 18:58   9/13/20 20:02              NaN   \n",
       "1   9/13/20 18:58   9/13/20 20:15              NaN   \n",
       "2   9/13/20 18:57   9/13/20 20:01              NaN   \n",
       "3   9/13/20 18:57   9/13/20 20:01              NaN   \n",
       "4   9/13/20 18:58   9/13/20 20:16              NaN   \n",
       "5   9/13/20 18:58   9/13/20 20:16              NaN   \n",
       "6   9/13/20 18:58   9/13/20 20:21              NaN   \n",
       "7   9/13/20 18:58   9/13/20 20:03              NaN   \n",
       "8   9/13/20 18:57   9/13/20 19:49              NaN   \n",
       "9   9/13/20 18:57   9/13/20 19:59              NaN   \n",
       "10  9/13/20 18:58   9/13/20 20:18              NaN   \n",
       "11  9/13/20 18:58   9/13/20 20:20              NaN   \n",
       "12  9/13/20 18:57   9/13/20 19:48              NaN   \n",
       "13  9/13/20 18:58   9/13/20 20:16              NaN   \n",
       "14  9/13/20 18:57   9/13/20 19:52              NaN   \n",
       "15  9/13/20 18:58   9/13/20 20:16              NaN   \n",
       "16  9/13/20 18:58   9/13/20 20:20              NaN   \n",
       "17  9/13/20 18:58   9/13/20 20:15              NaN   \n",
       "18  9/13/20 18:57   9/13/20 20:02              NaN   \n",
       "19  9/13/20 18:57   9/13/20 22:45              NaN   \n",
       "20  9/13/20 18:58   9/13/20 20:16              NaN   \n",
       "21  9/13/20 18:57   9/13/20 19:49              NaN   \n",
       "22  9/13/20 18:57   9/13/20 20:00              NaN   \n",
       "23  9/13/20 18:58   9/13/20 20:09              NaN   \n",
       "24  9/13/20 18:58   9/13/20 20:21              NaN   \n",
       "25  9/13/20 18:57   9/13/20 20:00              NaN   \n",
       "26  9/13/20 18:57   9/13/20 20:01              NaN   \n",
       "27  9/13/20 18:57   9/13/20 20:03              NaN   \n",
       "28  9/13/20 18:57   9/13/20 19:48              NaN   \n",
       "29  9/13/20 18:57   9/13/20 19:49              NaN   \n",
       "\n",
       "                                            Full Path  \\\n",
       "0   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "1   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "2   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "3   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "4   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "5   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "6   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "7   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "8   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "9   /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "10  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "11  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "12  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "13  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "14  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "15  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "16  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "17  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "18  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "19  /gscmnt/gc13037/lims/2857177/2020-09/csf_17716...   \n",
       "20  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "21  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "22  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "23  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "24  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "25  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "26  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "27  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "28  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "29  /gscmnt/gc13037/lims/2857177/2020-09/csf_17715...   \n",
       "\n",
       "                                              File1  \\\n",
       "0     AAGAAGTCAC-ACGAATAAGG_S4_L002_R1_001.fastq.gz   \n",
       "1     AAGAGCTCGC-AGTAAGAGTC_S5_L002_R2_001.fastq.gz   \n",
       "2     AAGATCCGTC-TTACACCAAC_S6_L002_R2_001.fastq.gz   \n",
       "3     AAGCTAGCTG-CTTCCGTGAA_S7_L002_R1_001.fastq.gz   \n",
       "4    AATTAAGCGG-ATGGTCAGTA_S11_L002_R2_001.fastq.gz   \n",
       "5    ACTTGCATAC-ACATGAATGC_S15_L002_R1_001.fastq.gz   \n",
       "6    ACTTGGCTTC-GCAGAGCACA_S16_L002_R2_001.fastq.gz   \n",
       "7    AGCTGAACTA-TGTGTCCAAC_S28_L002_R2_001.fastq.gz   \n",
       "8    CAATAGAGGC-CTTACGTATC_S32_L002_R1_001.fastq.gz   \n",
       "9    CACATCAAGC-TGTATAGGCT_S35_L002_R2_001.fastq.gz   \n",
       "10   CACGGAATTG-TTACAGCCAG_S37_L002_R2_001.fastq.gz   \n",
       "11   CAGTCACTTG-ACAGCCAGGT_S43_L002_R1_001.fastq.gz   \n",
       "12   CATTCCTAAC-TGTGGCGATT_S47_L002_R2_001.fastq.gz   \n",
       "13   CCAACTCTTA-GAAGTCCTTG_S50_L002_R2_001.fastq.gz   \n",
       "14   CCACAATTGC-GATGTGAAGT_S51_L002_R1_001.fastq.gz   \n",
       "15   CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz   \n",
       "16   CCGATTATGA-ACCGGAGTAA_S57_L002_R1_001.fastq.gz   \n",
       "17   CGGATAACGG-TATCCTTCGC_S70_L002_R1_001.fastq.gz   \n",
       "18   GTAAGCGAGG-AGTTCACCGG_S75_L002_R2_001.fastq.gz   \n",
       "19   GTCGCATTAG-TACGATAAGC_S80_L002_R2_001.fastq.gz   \n",
       "20   GTCGTTACCT-TTACTTCGTG_S81_L002_R2_001.fastq.gz   \n",
       "21   GTCTGAATAG-CTTGTCTTGC_S82_L002_R2_001.fastq.gz   \n",
       "22   GTGAAGGCTG-GCACGATGCA_S84_L002_R1_001.fastq.gz   \n",
       "23   TCAGCTGTCA-ATACAGGACG_S85_L002_R2_001.fastq.gz   \n",
       "24   TCCTACAGCG-TGTGACCAGT_S91_L002_R2_001.fastq.gz   \n",
       "25   TCCTGATGCG-TTCAGATTGC_S92_L002_R2_001.fastq.gz   \n",
       "26  TCTCGCAGGA-GACGCAAGGT_S103_L002_R2_001.fastq.gz   \n",
       "27  TGAAGTCAAC-GAGGTAGTTC_S104_L002_R1_001.fastq.gz   \n",
       "28  TGAGACTCGC-TAGGAAGCGG_S107_L002_R1_001.fastq.gz   \n",
       "29  TGATTGTCTG-TAAGCTGTAG_S108_L002_R2_001.fastq.gz   \n",
       "\n",
       "                                              File2  \n",
       "0     AAGAAGTCAC-ACGAATAAGG_S4_L002_R2_001.fastq.gz  \n",
       "1     AAGAGCTCGC-AGTAAGAGTC_S5_L002_R1_001.fastq.gz  \n",
       "2     AAGATCCGTC-TTACACCAAC_S6_L002_R1_001.fastq.gz  \n",
       "3     AAGCTAGCTG-CTTCCGTGAA_S7_L002_R2_001.fastq.gz  \n",
       "4    AATTAAGCGG-ATGGTCAGTA_S11_L002_R1_001.fastq.gz  \n",
       "5    ACTTGCATAC-ACATGAATGC_S15_L002_R2_001.fastq.gz  \n",
       "6    ACTTGGCTTC-GCAGAGCACA_S16_L002_R1_001.fastq.gz  \n",
       "7    AGCTGAACTA-TGTGTCCAAC_S28_L002_R1_001.fastq.gz  \n",
       "8    CAATAGAGGC-CTTACGTATC_S32_L002_R2_001.fastq.gz  \n",
       "9    CACATCAAGC-TGTATAGGCT_S35_L002_R1_001.fastq.gz  \n",
       "10   CACGGAATTG-TTACAGCCAG_S37_L002_R1_001.fastq.gz  \n",
       "11   CAGTCACTTG-ACAGCCAGGT_S43_L002_R2_001.fastq.gz  \n",
       "12   CATTCCTAAC-TGTGGCGATT_S47_L002_R1_001.fastq.gz  \n",
       "13   CCAACTCTTA-GAAGTCCTTG_S50_L002_R1_001.fastq.gz  \n",
       "14   CCACAATTGC-GATGTGAAGT_S51_L002_R2_001.fastq.gz  \n",
       "15   CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz  \n",
       "16   CCGATTATGA-ACCGGAGTAA_S57_L002_R2_001.fastq.gz  \n",
       "17   CGGATAACGG-TATCCTTCGC_S70_L002_R2_001.fastq.gz  \n",
       "18   GTAAGCGAGG-AGTTCACCGG_S75_L002_R1_001.fastq.gz  \n",
       "19   GTCGCATTAG-TACGATAAGC_S80_L002_R1_001.fastq.gz  \n",
       "20   GTCGTTACCT-TTACTTCGTG_S81_L002_R1_001.fastq.gz  \n",
       "21   GTCTGAATAG-CTTGTCTTGC_S82_L002_R1_001.fastq.gz  \n",
       "22   GTGAAGGCTG-GCACGATGCA_S84_L002_R2_001.fastq.gz  \n",
       "23   TCAGCTGTCA-ATACAGGACG_S85_L002_R1_001.fastq.gz  \n",
       "24   TCCTACAGCG-TGTGACCAGT_S91_L002_R1_001.fastq.gz  \n",
       "25   TCCTGATGCG-TTCAGATTGC_S92_L002_R1_001.fastq.gz  \n",
       "26  TCTCGCAGGA-GACGCAAGGT_S103_L002_R1_001.fastq.gz  \n",
       "27  TGAAGTCAAC-GAGGTAGTTC_S104_L002_R2_001.fastq.gz  \n",
       "28  TGAGACTCGC-TAGGAAGCGG_S107_L002_R2_001.fastq.gz  \n",
       "29  TGATTGTCTG-TAAGCTGTAG_S108_L002_R1_001.fastq.gz  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/data/pecgs/test_sample/ht191p1-s1h1a3/Samplemap.csv', sep=',')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "273668b3",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>lane</th>\n",
       "      <th>index_sequencer</th>\n",
       "      <th>library_preparation</th>\n",
       "      <th>platform</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>tumor</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>normal</td>\n",
       "      <td>HH7KNDSXY</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTATCGCT-CTTGAAGGTT</td>\n",
       "      <td>TWCE-HT191P1-JM1D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           sample_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "\n",
       "                                                   experimental_strategy  \\\n",
       "run_id                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "\n",
       "                                                   sample_type   flowcell  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       tumor  HFMFWDSXY   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...      normal  HH7KNDSXY   \n",
       "\n",
       "                                                   lane  \\\n",
       "run_id                                                    \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...    2   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...    1   \n",
       "\n",
       "                                                          index_sequencer  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CCAGTAGCGT-ATGTATTGGC   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CATTATCGCT-CTTGAAGGTT   \n",
       "\n",
       "                                                               library_preparation  \\\n",
       "run_id                                                                               \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  TWCE-HT191P1-S1H1A3Y3D1_1-lib1   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       TWCE-HT191P1-JM1D1_1-lib1   \n",
       "\n",
       "                                                    platform  \n",
       "run_id                                                        \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = [\n",
    "    ['HT191P1-S1H1A3Y3', '419b8098-e4b2-4318-883d-c233cb7e06c8', 'wxs', 'tumor', 'HFMFWDSXY', '2', 'CCAGTAGCGT-ATGTATTGGC', 'TWCE-HT191P1-S1H1A3Y3D1_1-lib1', 'ILLUMINA'],\n",
    "    ['HT191P1-S1H1A3Y3', '419b8098-e4b2-4318-883d-c233cb7e06c8', 'wxs', 'normal', 'HH7KNDSXY', '1', 'CATTATCGCT-CTTGAAGGTT', 'TWCE-HT191P1-JM1D1_1-lib1', 'ILLUMINA']\n",
    "]\n",
    "sq_df = pd.DataFrame(data=data, columns=['sample_id', 'run_uuid', 'experimental_strategy', 'sample_type',\n",
    "                                        'flowcell', 'lane', 'index_sequencer', 'library_preparation', 'platform'])\n",
    "sq_df.index = [f'{s}_{u}' for s, u in zip(sq_df['sample_id'], sq_df['run_uuid'])]\n",
    "sq_df.index.name = 'run_id'\n",
    "sq_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7f5e6cad",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "sq_df.to_csv('../tests/data/pecgs_TN_wxs_fq/sequencing_info.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85690a09",
   "metadata": {},
   "source": [
    "###### save defaults"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "9bf067e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "resources_dir = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "21872288",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "    'cpu': 40,\n",
    "    \n",
    "    # align dna-seq\n",
    "    'known_sites': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/dnaseq_alignment/dbsnp/00-All.chr.vcf.gz',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': f'{resources_dir}/dnaseq_alignment/dbsnp/00-All.chr.vcf.gz.tbi'}\n",
    "        ]\n",
    "    },\n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    'platform': 'ILLUMINA',\n",
    "    \n",
    "    # cnv\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/snp_database/af-only-gnomad.hg38.common_biallelic.chr1-22XY.vcf'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/pon/create_pon/gatk4scnaPON.Normal.hdf5'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/remaking_protein-coding/gencode.v34.annotation.gene_filterd.need_gene_symbol.no_sym.filtered_to_hgnc_protein-coding_genes.bed'\n",
    "    },\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/IDT_xGen_Exome_Research_Panel_v1.hg38.removed_alt_chr.bed.target.preprocessed.exome.interval_list'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/PECGS/ref_genome/hg38.microsatellite'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 1,\n",
    "    'minimal_microsatellite_size': 1,\n",
    "    \n",
    "    # tindaisy\n",
    "    'clinvar_annotation': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/clinvar/GRCh38/clinvar_20200706.vcf.gz'\n",
    "    },\n",
    "    'rescue_clinvar': True,\n",
    "    'rescue_cosmic': True,\n",
    "    'call_regions': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed.gz'\n",
    "    },\n",
    "    'canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'pindel_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/pindel.WES.ini'\n",
    "    },\n",
    "    'strelka_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/strelka.WES.ini'\n",
    "    },\n",
    "    'varscan_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/varscan.WES.ini'\n",
    "    },\n",
    "    'classification_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/classification_filter_config.ini'\n",
    "    },\n",
    "    'af_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/af_filter_config.ini'\n",
    "    },\n",
    "    'centromere_bed': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tindaisy_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/v99/vep-cache.99_GRCh38.tar.gz'\n",
    "    },\n",
    "    'tindaisy_vep_cache_version': 99,\n",
    "    'assembly': 'GRCh38',\n",
    "    'tindaisy_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.d1.vd1.chrlist.txt'\n",
    "    },\n",
    "    \n",
    "    # TinJasmine\n",
    "    'centromere': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tinjasmine_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/chrlist/GRCh38.d1.vd1.chrlist-reordered.txt'\n",
    "    },\n",
    "    'pindel_config_template': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_config_template/pindel_germline_filter_config.ini'\n",
    "    },\n",
    "    'Canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/Canonical_BED/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'ROI_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/ROI_BED/Homo_sapiens.GRCh38.95.allCDS.2bpFlanks.biomart.withCHR.bed'\n",
    "    },\n",
    "#     'varscan_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/varscan_filter_config/VLD_FilterVCF-varscan.config.ini'\n",
    "#     },\n",
    "#     'pindel_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_filter_config/VLD_FilterVCF-pindel.config.ini'\n",
    "#     },\n",
    "#     'gatk_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/gatk_filter_config/VLD_FilterVCF-GATK.config.ini'\n",
    "#     },\n",
    "    'tinjasmine_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/20220525/vep-cache.100_GRCh38.tar.gz'\n",
    "    },\n",
    "    \n",
    "    # neoscan\n",
    "    'neoscan_ref_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29'\n",
    "    },\n",
    "    'neoscan_bed': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29/proteome.bed'\n",
    "    },\n",
    "    \n",
    "    # charger\n",
    "    'charger_inheritance_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cancer_pred_genes_160genes_011321_curated_forCharGer.txt'\n",
    "    },\n",
    "    'charger_pp2_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/160cpgs.txt'\n",
    "    },\n",
    "    'charger_pathogenic_variants': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/emptyRemoved_20160428_pathogenic_variants_HGVSg_VEP_grch38lifOver.vcf'\n",
    "    },\n",
    "    'charger_hotspot3d_clusters': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cptac_mc3_combined_noHypers_sorted.maf.3D_Proximity.pairwise.recurrence.l0.ad10.r10.clusters'\n",
    "    },\n",
    "    'charger_clinvar_alleles': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/clinvar_alleles.single.b38.tsv.gz'\n",
    "    },\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "07023c90",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows/defaults.pecgs_TN_wxs_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "fedbeeb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_TN_wxs_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "0caca783",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "    'cpu': 40,\n",
    "    \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    \n",
    "    # cnv\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/snp_database/af-only-gnomad.hg38.common_biallelic.chr1-22XY.vcf'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/pon/create_pon/gatk4scnaPON.Normal.hdf5'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/remaking_protein-coding/gencode.v34.annotation.gene_filterd.need_gene_symbol.no_sym.filtered_to_hgnc_protein-coding_genes.bed'\n",
    "    },\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/IDT_xGen_Exome_Research_Panel_v1.hg38.removed_alt_chr.bed.target.preprocessed.exome.interval_list'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/PECGS/ref_genome/hg38.microsatellite'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 1,\n",
    "    'minimal_microsatellite_size': 1,\n",
    "    \n",
    "    # tindaisy\n",
    "    'clinvar_annotation': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/clinvar/GRCh38/clinvar_20200706.vcf.gz'\n",
    "    },\n",
    "    'rescue_clinvar': True,\n",
    "    'rescue_cosmic': True,\n",
    "    'call_regions': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed.gz'\n",
    "    },\n",
    "    'canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'pindel_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/pindel.WES.ini'\n",
    "    },\n",
    "    'strelka_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/strelka.WES.ini'\n",
    "    },\n",
    "    'varscan_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/varscan.WES.ini'\n",
    "    },\n",
    "    'classification_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/classification_filter_config.ini'\n",
    "    },\n",
    "    'af_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/af_filter_config.ini'\n",
    "    },\n",
    "    'centromere_bed': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tindaisy_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/v99/vep-cache.99_GRCh38.tar.gz'\n",
    "    },\n",
    "    'tindaisy_vep_cache_version': 99,\n",
    "    'assembly': 'GRCh38',\n",
    "    'tindaisy_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.d1.vd1.chrlist.txt'\n",
    "    },\n",
    "    \n",
    "    # TinJasmine\n",
    "    'centromere': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tinjasmine_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/chrlist/GRCh38.d1.vd1.chrlist-reordered.txt'\n",
    "    },\n",
    "    'pindel_config_template': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_config_template/pindel_germline_filter_config.ini'\n",
    "    },\n",
    "    'Canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/Canonical_BED/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'ROI_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/ROI_BED/Homo_sapiens.GRCh38.95.allCDS.2bpFlanks.biomart.withCHR.bed'\n",
    "    },\n",
    "#     'varscan_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/varscan_filter_config/VLD_FilterVCF-varscan.config.ini'\n",
    "#     },\n",
    "#     'pindel_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_filter_config/VLD_FilterVCF-pindel.config.ini'\n",
    "#     },\n",
    "#     'gatk_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/gatk_filter_config/VLD_FilterVCF-GATK.config.ini'\n",
    "#     },\n",
    "    'tinjasmine_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/20220525/vep-cache.100_GRCh38.tar.gz'\n",
    "    },\n",
    "    \n",
    "    # neoscan\n",
    "    'neoscan_ref_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29'\n",
    "    },\n",
    "    'neoscan_bed': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29/proteome.bed'\n",
    "    },\n",
    "    \n",
    "    # charger\n",
    "    'charger_inheritance_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cancer_pred_genes_160genes_011321_curated_forCharGer.txt'\n",
    "    },\n",
    "    'charger_pp2_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/160cpgs.txt'\n",
    "    },\n",
    "    'charger_pathogenic_variants': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/emptyRemoved_20160428_pathogenic_variants_HGVSg_VEP_grch38lifOver.vcf'\n",
    "    },\n",
    "    'charger_hotspot3d_clusters': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cptac_mc3_combined_noHypers_sorted.maf.3D_Proximity.pairwise.recurrence.l0.ad10.r10.clusters'\n",
    "    },\n",
    "    'charger_clinvar_alleles': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/clinvar_alleles.single.b38.tsv.gz'\n",
    "    },\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "71c90817",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows/defaults.pecgs_TN_wxs_bam.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0af40fef",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_TN_wxs_bam.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "dbe514f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "    'cpu': 40,\n",
    "    \n",
    "    # fusion\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/Integrate_dependencies/bwts'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/FilterDatabase'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/FusionAnnotator'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/ericscript_dependencies/ericscript_db_homosapiens_ensembl84'\n",
    "    },\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/STAR-Fusion_dependencies/GRCh38_gencode_v37_CTAT_lib_Mar012021.plug-n-play/ctat_genome_lib_build_dir'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/fusion/Integrate_dependencies/annot.ensembl.GRCh38.txt'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/fusion/INTEGRATE_0_2_6/INTEGRATE-build/bin/Integrate'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/fusion/Integrate_dependencies/STAR/hg38.fa'\n",
    "    },\n",
    "    \n",
    "    # bulk expression\n",
    "    'star_index': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/bulk_expression/STAR_GRCh38_d1_vd1_d1_GENCODE_v34'\n",
    "    },\n",
    "    'gtf': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/bulk_expression/STAR_GRCh38_d1_vd1_d1_GENCODE_v34/gencode.v34.annotation.gtf'\n",
    "    },\n",
    "    'gene_info': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/bulk_expression/STAR_GRCh38_d1_vd1_d1_GENCODE_v34/gencode.gene.info.v34.tsv'\n",
    "    },\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "42947ee4",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows/defaults.pecgs_T_rna_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b7602d3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_T_rna_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "81c73d7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "#     'cpu': 40,\n",
    "    \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa',\n",
    "    },\n",
    "    \n",
    "    # somatic sv\n",
    "    'generate_evidence_bam': True,\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e8e42887",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_TN_wgs_bam.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03ec54fb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
