{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7f760e82",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import pandas as pd\n",
    "import yaml\n",
    "import uuid"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6eb10c2",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### C3N-05929 test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "643351b1",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>case</th>\n",
       "      <th>disease</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>data_format</th>\n",
       "      <th>reference</th>\n",
       "      <th>UUID</th>\n",
       "      <th>system</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3N-05929.WGS.T</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>WGS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3N-05929.WGS.N</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>WGS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample_name       case disease experimental_strategy   sample_type  \\\n",
       "0  C3N-05929.WGS.T  C3N-05929    LSCC                   WGS         tumor   \n",
       "1  C3N-05929.WGS.N  C3N-05929    LSCC                   WGS  blood_normal   \n",
       "\n",
       "                                           data_path  filesize data_format  \\\n",
       "0  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...  12345678         BAM   \n",
       "1  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...  12345678         BAM   \n",
       "\n",
       "  reference                                  UUID    system  \n",
       "0      hg38  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83  storage1  \n",
       "1      hg38  7d8f4ede-1911-456e-9eee-2fafe7132f50  storage1  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bammap = pd.read_csv('../tests/data/pecgs_TN_wgs_bam/input.bammap', sep='\\t')\n",
    "bammap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f99dd1cd",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'C3N-05929': {'wgs': {'tumor': ('/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/c5d518a3-fc69-4d7b-a719-0aad38abe43a/23fa88f9-9293-406b-bc21-2928ff12bc8b_wgs_gdc_realn.bam',\n",
       "     'd3ddd33b-c7a3-4caf-bdd7-c05e98a19d83'),\n",
       "    'normal': ('/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/823dda5b-ff15-4a66-b28b-de0632850623/e3bf416b-3d08-4d1b-bec4-808e55c7788c_wgs_gdc_realn.bam',\n",
       "     '7d8f4ede-1911-456e-9eee-2fafe7132f50')}}},\n",
       " {'C3N-05929': {'disease': 'LSCC'}})"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cases = sorted(set(bammap['case']))\n",
    "inputs = {}\n",
    "metadata = {}\n",
    "for case in cases:\n",
    "    data = {\n",
    "        'wgs': {\n",
    "        }\n",
    "    }\n",
    "    filtered = bammap[bammap['case']==case]\n",
    "    # check for wxs tumor bam\n",
    "    f = filtered[[True if e=='WGS' and df=='BAM' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wgs tumor bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wgs']['tumor'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "    \n",
    "    # check for wxs normal bam\n",
    "    f = filtered[[True if e=='WGS' and df=='BAM' and 'normal' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wgs normal bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wgs']['normal'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "            \n",
    "    # if everything is present then accept sample\n",
    "    if 'tumor' in data['wgs'] and 'normal' in data['wgs']:\n",
    "        inputs[case] = data\n",
    "        metadata[case] = {'disease': f['disease'].to_list()[0]}\n",
    "inputs, metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "102214a5",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C3N-05929': {'wgs_tumor_bam.filepath': '/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/c5d518a3-fc69-4d7b-a719-0aad38abe43a/23fa88f9-9293-406b-bc21-2928ff12bc8b_wgs_gdc_realn.bam',\n",
       "  'wgs_normal_bam.filepath': '/storage1/fs1/m.wyczalkowski/Active/Primary/CPTAC3.share/CPTAC3-GDC/GDC_import/data/823dda5b-ff15-4a66-b28b-de0632850623/e3bf416b-3d08-4d1b-bec4-808e55c7788c_wgs_gdc_realn.bam',\n",
       "  'wgs_tumor_bam.uuid': 'd3ddd33b-c7a3-4caf-bdd7-c05e98a19d83',\n",
       "  'wgs_normal_bam.uuid': '7d8f4ede-1911-456e-9eee-2fafe7132f50'}}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = {}\n",
    "for case, d in inputs.items():\n",
    "    run_list[case] = {f'{dt}_{st}_bam.filepath': fp\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()}\n",
    "    run_list[case].update({f'{dt}_{st}_bam.uuid': uid\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()})\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c30d35a6",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wgs_normal_bam.filepath</th>\n",
       "      <th>wgs_normal_bam.uuid</th>\n",
       "      <th>wgs_tumor_bam.filepath</th>\n",
       "      <th>wgs_tumor_bam.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90</th>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>ad046693-8a91-4c29-a97e-3f13da5e7c90</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                          wgs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                 wgs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  7d8f4ede-1911-456e-9eee-2fafe7132f50   \n",
       "\n",
       "                                                                           wgs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                  wgs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83   \n",
       "\n",
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  C3N-05929   \n",
       "\n",
       "                                                                            run_uuid  \n",
       "run_id                                                                                \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  ad046693-8a91-4c29-a97e-3f13da5e7c90  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_table = pd.DataFrame.from_dict(run_list).transpose()\n",
    "run_table['case_id'] = run_table.index.to_list()\n",
    "run_table['run_uuid'] = [str(uuid.uuid4()) for i in range(run_table.shape[0])]\n",
    "run_table.index = [f'{c}_{u}' for c, u in zip(run_table.index, run_table['run_uuid'])]\n",
    "run_table.index.name = 'run_id'\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0161e6d2",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-05929</th>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          disease\n",
       "C3N-05929    LSCC"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_table = pd.DataFrame.from_dict(metadata).transpose()\n",
    "metadata_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c37ab304",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wgs_normal_bam.filepath</th>\n",
       "      <th>wgs_normal_bam.uuid</th>\n",
       "      <th>wgs_tumor_bam.filepath</th>\n",
       "      <th>wgs_tumor_bam.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90</th>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>ad046693-8a91-4c29-a97e-3f13da5e7c90</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                          wgs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                 wgs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  7d8f4ede-1911-456e-9eee-2fafe7132f50   \n",
       "\n",
       "                                                                           wgs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                  wgs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83   \n",
       "\n",
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  C3N-05929   \n",
       "\n",
       "                                                                            run_uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  ad046693-8a91-4c29-a97e-3f13da5e7c90   \n",
       "\n",
       "                                               disease  \n",
       "run_id                                                  \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90    LSCC  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for c in metadata_table.columns:\n",
    "    run_table[c] = [metadata_table.loc[x, c] for x in run_table['case_id']]\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b67c0ba7",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "run_table['project'] = 'TCGA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ac3c14f2",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'project', 'disease', 'wgs_normal_bam.filepath', 'wgs_normal_bam.uuid', 'wgs_tumor_bam.filepath', 'wgs_tumor_bam.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_TN_wgs_bam/run_list.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "74a9c1b0",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>disease</th>\n",
       "      <th>wgs_normal_bam.filepath</th>\n",
       "      <th>wgs_normal_bam.uuid</th>\n",
       "      <th>wgs_tumor_bam.filepath</th>\n",
       "      <th>wgs_tumor_bam.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90</th>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>ad046693-8a91-4c29-a97e-3f13da5e7c90</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  C3N-05929   \n",
       "\n",
       "                                                                            run_uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  ad046693-8a91-4c29-a97e-3f13da5e7c90   \n",
       "\n",
       "                                               disease  \\\n",
       "run_id                                                   \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90    LSCC   \n",
       "\n",
       "                                                                          wgs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                 wgs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  7d8f4ede-1911-456e-9eee-2fafe7132f50   \n",
       "\n",
       "                                                                           wgs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                  wgs_tumor_bam.uuid  \n",
       "run_id                                                                                \n",
       "C3N-05929_ad046693-8a91-4c29-a97e-3f13da5e7c90  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df = pd.read_csv('../tests/data/pecgs_TN_wgs_bam/run_list.txt', sep='\\t', index_col=0)\n",
    "# df['project'] = 'TCGA'\n",
    "# df = df[['case_id', 'run_uuid', 'project', 'disease', 'wgs_normal_bam.filepath', 'wgs_normal_bam.uuid', 'wgs_tumor_bam.filepath', 'wgs_tumor_bam.uuid']]\n",
    "# df.to_csv('../tests/data/pecgs_TN_wgs_bam/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b50f9e6b",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### MMRF_1250 test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "5b6dc7e5",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>case</th>\n",
       "      <th>disease</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>data_format</th>\n",
       "      <th>reference</th>\n",
       "      <th>UUID</th>\n",
       "      <th>system</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00677.WXS.T</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WXS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>6e0bde4a-22c1-4e1f-a38a-dff275b61472</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00677.WXS.N</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WXS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>BAM</td>\n",
       "      <td>hg38</td>\n",
       "      <td>8919accb-6973-4ec5-b29f-9d895a1713d4</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R1.T</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2983df85-958b-483c-95e5-2375426879d1</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R2.T</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>GBM</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>f840b94b-eac9-4f87-98b6-51b26b606e1a</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     sample_name       case disease experimental_strategy  \\\n",
       "0                C3L-00677.WXS.T  C3L-00677     GBM                   WXS   \n",
       "1                C3L-00677.WXS.N  C3L-00677     GBM                   WXS   \n",
       "2  HT191P1-S1H1A3Y3.RNA-seq.R1.T  C3L-00677     GBM               RNA-seq   \n",
       "3  HT191P1-S1H1A3Y3.RNA-seq.R2.T  C3L-00677     GBM               RNA-seq   \n",
       "\n",
       "    sample_type                                          data_path  filesize  \\\n",
       "0         tumor  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "1  blood_normal  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "2         tumor  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "3         tumor  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678   \n",
       "\n",
       "  data_format reference                                  UUID    system  \n",
       "0         BAM      hg38  6e0bde4a-22c1-4e1f-a38a-dff275b61472  storage1  \n",
       "1         BAM      hg38  8919accb-6973-4ec5-b29f-9d895a1713d4  storage1  \n",
       "2       FASTQ       NaN  2983df85-958b-483c-95e5-2375426879d1  storage1  \n",
       "3       FASTQ       NaN  f840b94b-eac9-4f87-98b6-51b26b606e1a  storage1  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bammap = pd.read_csv('../tests/data/pecgs_TN_wxs_bam/input.bammap', sep='\\t')\n",
    "bammap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "922d94bb",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'C3L-00677': {'wxs': {'tumor': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/1d6c4f0f-09c4-4472-a19e-30ee7e0ecf64_gdc_realn.bam',\n",
       "     '6e0bde4a-22c1-4e1f-a38a-dff275b61472'),\n",
       "    'normal': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/5cee818c-61e8-4441-bd9a-5c984cde9c6b_gdc_realn.bam',\n",
       "     '8919accb-6973-4ec5-b29f-9d895a1713d4')}}},\n",
       " {'C3L-00677': {'disease': 'GBM'}})"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cases = sorted(set(bammap['case']))\n",
    "inputs = {}\n",
    "metadata = {}\n",
    "for case in cases:\n",
    "    data = {\n",
    "        'wxs': {\n",
    "        }\n",
    "    }\n",
    "    filtered = bammap[bammap['case']==case]\n",
    "    # check for wxs tumor bam\n",
    "    f = filtered[[True if e=='WXS' and df=='BAM' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wxs tumor bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wxs']['tumor'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "    \n",
    "    # check for wxs normal bam\n",
    "    f = filtered[[True if e=='WXS' and df=='BAM' and 'normal' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 1:\n",
    "        print(f'Warning: more than 1 wxs normal bams found for case {case}')\n",
    "    if f.shape[0]:\n",
    "        data['wxs']['normal'] = (f['data_path'].to_list()[0], f['UUID'].to_list()[0])\n",
    "            \n",
    "    # if everything is present then accept sample\n",
    "    if 'tumor' in data['wxs'] and 'normal' in data['wxs']:\n",
    "        inputs[case] = data\n",
    "        metadata[case] = {'disease': f['disease'].to_list()[0]}\n",
    "inputs, metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "116a51f9",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'C3L-00677': {'wxs_tumor_bam.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/1d6c4f0f-09c4-4472-a19e-30ee7e0ecf64_gdc_realn.bam',\n",
       "  'wxs_normal_bam.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/C3L-00677/wxs/5cee818c-61e8-4441-bd9a-5c984cde9c6b_gdc_realn.bam',\n",
       "  'wxs_tumor_bam.uuid': '6e0bde4a-22c1-4e1f-a38a-dff275b61472',\n",
       "  'wxs_normal_bam.uuid': '8919accb-6973-4ec5-b29f-9d895a1713d4'}}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = {}\n",
    "for case, d in inputs.items():\n",
    "    run_list[case] = {f'{dt}_{st}_bam.filepath': fp\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()}\n",
    "    run_list[case].update({f'{dt}_{st}_bam.uuid': uid\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, (fp, uid) in d1.items()})\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "deaceac1",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wxs_normal_bam.filepath</th>\n",
       "      <th>wxs_normal_bam.uuid</th>\n",
       "      <th>wxs_tumor_bam.filepath</th>\n",
       "      <th>wxs_tumor_bam.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9</th>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>8919accb-6973-4ec5-b29f-9d895a1713d4</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>6e0bde4a-22c1-4e1f-a38a-dff275b61472</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>9d492e62-5b46-4b9a-a4af-cd40c74548b9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                          wxs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                 wxs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  8919accb-6973-4ec5-b29f-9d895a1713d4   \n",
       "\n",
       "                                                                           wxs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                  wxs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  6e0bde4a-22c1-4e1f-a38a-dff275b61472   \n",
       "\n",
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  C3L-00677   \n",
       "\n",
       "                                                                            run_uuid  \n",
       "run_id                                                                                \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  9d492e62-5b46-4b9a-a4af-cd40c74548b9  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_table = pd.DataFrame.from_dict(run_list).transpose()\n",
    "run_table['case_id'] = run_table.index.to_list()\n",
    "run_table['run_uuid'] = [str(uuid.uuid4()) for i in range(run_table.shape[0])]\n",
    "run_table.index = [f'{c}_{u}' for c, u in zip(run_table.index, run_table['run_uuid'])]\n",
    "run_table.index.name = 'run_id'\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e7d153aa",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          disease\n",
       "C3L-00677     GBM"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_table = pd.DataFrame.from_dict(metadata).transpose()\n",
    "metadata_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "54d4d38e",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wxs_normal_bam.filepath</th>\n",
       "      <th>wxs_normal_bam.uuid</th>\n",
       "      <th>wxs_tumor_bam.filepath</th>\n",
       "      <th>wxs_tumor_bam.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9</th>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>8919accb-6973-4ec5-b29f-9d895a1713d4</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>6e0bde4a-22c1-4e1f-a38a-dff275b61472</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>9d492e62-5b46-4b9a-a4af-cd40c74548b9</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                          wxs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                 wxs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  8919accb-6973-4ec5-b29f-9d895a1713d4   \n",
       "\n",
       "                                                                           wxs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                  wxs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  6e0bde4a-22c1-4e1f-a38a-dff275b61472   \n",
       "\n",
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  C3L-00677   \n",
       "\n",
       "                                                                            run_uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9  9d492e62-5b46-4b9a-a4af-cd40c74548b9   \n",
       "\n",
       "                                               disease  \n",
       "run_id                                                  \n",
       "C3L-00677_9d492e62-5b46-4b9a-a4af-cd40c74548b9     GBM  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for c in metadata_table.columns:\n",
    "    run_table[c] = [metadata_table.loc[x, c] for x in run_table['case_id']]\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "a86410cc",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'disease', 'wxs_normal_bam.filepath', 'wxs_normal_bam.uuid', 'wxs_tumor_bam.filepath', 'wxs_tumor_bam.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_TN_wxs_bam/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "460fa99d",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# df = pd.read_csv('../tests/data/pecgs_TN_wxs_bam/run_list.txt', sep='\\t', index_col=0)\n",
    "# df['project'] = 'TCGA'\n",
    "# df = df[['case_id', 'run_uuid', 'project', 'disease', 'wxs_normal_bam.filepath', 'wxs_normal_bam.uuid', 'wxs_tumor_bam.filepath', 'wxs_tumor_bam.uuid']]\n",
    "# df.to_csv('../tests/data/pecgs_TN_wxs_bam/run_list.txt', sep='\\t')\n",
    "# # df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ffd48545",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# run_map = filtered.transpose().to_dict()\n",
    "# run_map = {k: {c.replace('.filepath', ''): val\n",
    "#                for c, val in v.items() if 'filepath' in c}\n",
    "#            for k, v in run_map.items()}\n",
    "# run_map"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c55eb482",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### HT191P1 test case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "9aa00f98",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>case</th>\n",
       "      <th>disease</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>data_format</th>\n",
       "      <th>reference</th>\n",
       "      <th>UUID</th>\n",
       "      <th>system</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R1.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9f3664ec-db31-46ae-b9f3-c06af39bda0e</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R2.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>b9df3743-3d7a-4ebd-89e3-85b22bb930ac</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R1.N</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5932ffee-c02e-4816-8e2f-1dd42d15ee31</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HT191P1-S1H1A3Y3.WXS.R2.N</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>WXS</td>\n",
       "      <td>blood_normal</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7bbe5119-4d51-4810-893c-0e0797c8219b</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R1.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06955efb-3963-416e-8a61-24c298f2f108</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HT191P1-S1H1A3Y3.RNA-seq.R2.T</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>tumor</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>12345678</td>\n",
       "      <td>FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ddd71943-6ad8-4c96-8935-0caba920da00</td>\n",
       "      <td>storage1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     sample_name              case disease  \\\n",
       "0      HT191P1-S1H1A3Y3.WXS.R1.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "1      HT191P1-S1H1A3Y3.WXS.R2.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "2      HT191P1-S1H1A3Y3.WXS.R1.N  HT191P1-S1H1A3Y3    PDAC   \n",
       "3      HT191P1-S1H1A3Y3.WXS.R2.N  HT191P1-S1H1A3Y3    PDAC   \n",
       "4  HT191P1-S1H1A3Y3.RNA-seq.R1.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "5  HT191P1-S1H1A3Y3.RNA-seq.R2.T  HT191P1-S1H1A3Y3    PDAC   \n",
       "\n",
       "  experimental_strategy   sample_type  \\\n",
       "0                   WXS         tumor   \n",
       "1                   WXS         tumor   \n",
       "2                   WXS  blood_normal   \n",
       "3                   WXS  blood_normal   \n",
       "4               RNA-seq         tumor   \n",
       "5               RNA-seq         tumor   \n",
       "\n",
       "                                           data_path  filesize data_format  \\\n",
       "0  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678       FASTQ   \n",
       "1  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678       FASTQ   \n",
       "2  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678       FASTQ   \n",
       "3  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678       FASTQ   \n",
       "4  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678       FASTQ   \n",
       "5  /storage1/fs1/dinglab/Active/Projects/estorrs/...  12345678       FASTQ   \n",
       "\n",
       "   reference                                  UUID    system  \n",
       "0        NaN  9f3664ec-db31-46ae-b9f3-c06af39bda0e  storage1  \n",
       "1        NaN  b9df3743-3d7a-4ebd-89e3-85b22bb930ac  storage1  \n",
       "2        NaN  5932ffee-c02e-4816-8e2f-1dd42d15ee31  storage1  \n",
       "3        NaN  7bbe5119-4d51-4810-893c-0e0797c8219b  storage1  \n",
       "4        NaN  06955efb-3963-416e-8a61-24c298f2f108  storage1  \n",
       "5        NaN  ddd71943-6ad8-4c96-8935-0caba920da00  storage1  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bammap = pd.read_csv('../tests/data/pecgs_TN_wxs_fq/input.bammap', sep='\\t')\n",
    "bammap"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "548e58a8",
   "metadata": {
    "hidden": true
   },
   "source": [
    "looking for all samples in bammap with all input files present (in this case paired WXS fastqs and rna-seq tumor fastqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "f64ff8e3",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'HT191P1-S1H1A3Y3': {'wxs': {'tumor': {'R1': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz',\n",
       "      '9f3664ec-db31-46ae-b9f3-c06af39bda0e'),\n",
       "     'R2': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz',\n",
       "      'b9df3743-3d7a-4ebd-89e3-85b22bb930ac')},\n",
       "    'normal': {'R1': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R1_001.fastq.gz',\n",
       "      '5932ffee-c02e-4816-8e2f-1dd42d15ee31'),\n",
       "     'R2': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R2_001.fastq.gz',\n",
       "      '7bbe5119-4d51-4810-893c-0e0797c8219b')}},\n",
       "   'rna-seq': {'tumor': {'R1': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "      '06955efb-3963-416e-8a61-24c298f2f108'),\n",
       "     'R2': ('/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R2.fastq.gz',\n",
       "      'ddd71943-6ad8-4c96-8935-0caba920da00')}}}},\n",
       " {'HT191P1-S1H1A3Y3': {'disease': 'PDAC'}})"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cases = sorted(set(bammap['case']))\n",
    "inputs = {}\n",
    "metadata = {}\n",
    "for case in cases:\n",
    "    data = {\n",
    "        'wxs': {\n",
    "            'tumor': {},\n",
    "            'normal': {}\n",
    "        },\n",
    "        'rna-seq': {\n",
    "            'tumor': {}\n",
    "        }\n",
    "    }\n",
    "    filtered = bammap[bammap['case']==case]\n",
    "    \n",
    "    # check for wxs tumor fastqs\n",
    "    f = filtered[[True if e=='WXS' and df=='FASTQ' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 2:\n",
    "        print(f'Warning: more than 2 wxs tumor fastqs found for case {case}')\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r1.' in row['sample_name'].lower():\n",
    "            data['wxs']['tumor']['R1'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r2.' in row['sample_name'].lower():\n",
    "            data['wxs']['tumor']['R2'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "            \n",
    "    # check for wxs normal fastqs\n",
    "    f = filtered[[True if e=='WXS' and df=='FASTQ' and 'normal' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 2:\n",
    "        print(f'Warning: more than 2 wxs normal fastqs found for case {case}')\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r1.' in row['sample_name'].lower():\n",
    "            data['wxs']['normal']['R1'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r2.' in row['sample_name'].lower():\n",
    "            data['wxs']['normal']['R2'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "            \n",
    "    # check for rna-seq tumor fastqs\n",
    "    f = filtered[[True if e=='RNA-seq' and df=='FASTQ' and 'tumor' in st.lower() else False\n",
    "                 for e, st, df in zip(filtered['experimental_strategy'],\n",
    "                                      filtered['sample_type'], filtered['data_format'])]]\n",
    "    if f.shape[0] > 2:\n",
    "        print(f'Warning: more than 2 rna-seq tumor fastqs found for case {case}')\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r1.' in row['sample_name'].lower():\n",
    "            data['rna-seq']['tumor']['R1'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "    for i, row in f.iterrows():\n",
    "        if '.r2.' in row['sample_name'].lower():\n",
    "            data['rna-seq']['tumor']['R2'] = (row['data_path'], row['UUID'])\n",
    "            break\n",
    "            \n",
    "    \n",
    "    # if everything is present then accept sample\n",
    "    # dont do in real life bc pipeline is split by input, but easier this way for test case\n",
    "    if len(data['wxs']['tumor']) == 2 and len(data['wxs']['normal']) == 2 and len(data['rna-seq']['tumor']) == 2:\n",
    "        inputs[case] = data\n",
    "        metadata[case] = {'disease': f['disease'].to_list()[0]}\n",
    "inputs, metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "82b19743",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HT191P1-S1H1A3Y3': {'wxs_tumor_R1.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz',\n",
       "  'wxs_tumor_R2.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz',\n",
       "  'wxs_normal_R1.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R1_001.fastq.gz',\n",
       "  'wxs_normal_R2.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R2_001.fastq.gz',\n",
       "  'rna-seq_tumor_R1.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "  'rna-seq_tumor_R2.filepath': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R2.fastq.gz',\n",
       "  'wxs_tumor_R1.uuid': '9f3664ec-db31-46ae-b9f3-c06af39bda0e',\n",
       "  'wxs_tumor_R2.uuid': 'b9df3743-3d7a-4ebd-89e3-85b22bb930ac',\n",
       "  'wxs_normal_R1.uuid': '5932ffee-c02e-4816-8e2f-1dd42d15ee31',\n",
       "  'wxs_normal_R2.uuid': '7bbe5119-4d51-4810-893c-0e0797c8219b',\n",
       "  'rna-seq_tumor_R1.uuid': '06955efb-3963-416e-8a61-24c298f2f108',\n",
       "  'rna-seq_tumor_R2.uuid': 'ddd71943-6ad8-4c96-8935-0caba920da00'}}"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = {}\n",
    "for case, d in inputs.items():\n",
    "    run_list[case] = {f'{dt}_{st}_{r}.filepath': fp\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, d2 in d1.items()\n",
    "                     for r, (fp, uid) in d2.items()}\n",
    "    run_list[case].update({f'{dt}_{st}_{r}.uuid': uid\n",
    "                     for dt, d1 in d.items()\n",
    "                     for st, d2 in d1.items()\n",
    "                     for r, (fp, uid) in d2.items()})\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "302919f6",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97628d17</th>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>06955efb-3963-416e-8a61-24c298f2f108</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>ddd71943-6ad8-4c96-8935-0caba920da00</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>5932ffee-c02e-4816-8e2f-1dd42d15ee31</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>7bbe5119-4d51-4810-893c-0e0797c8219b</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>9f3664ec-db31-46ae-b9f3-c06af39bda0e</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>b9df3743-3d7a-4ebd-89e3-85b22bb930ac</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>b089ade7-05a6-4dbf-97c2-b9eb97628d17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  06955efb-3963-416e-8a61-24c298f2f108   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  ddd71943-6ad8-4c96-8935-0caba920da00   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  5932ffee-c02e-4816-8e2f-1dd42d15ee31   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  7bbe5119-4d51-4810-893c-0e0797c8219b   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  9f3664ec-db31-46ae-b9f3-c06af39bda0e   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b9df3743-3d7a-4ebd-89e3-85b22bb930ac   \n",
       "\n",
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b089ade7-05a6-4dbf-97c2-b9eb97628d17  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_table = pd.DataFrame.from_dict(run_list).transpose()\n",
    "run_table['case_id'] = run_table.index.to_list()\n",
    "run_table['run_uuid'] = [str(uuid.uuid4()) for i in range(run_table.shape[0])]\n",
    "run_table.index = [f'{c}_{u}' for c, u in zip(run_table.index, run_table['run_uuid'])]\n",
    "run_table.index.name = 'run_id'\n",
    "run_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "aa95d0eb",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3</th>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 disease\n",
       "HT191P1-S1H1A3Y3    PDAC"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_table = pd.DataFrame.from_dict(metadata).transpose()\n",
    "metadata_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "a270d871",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97628d17</th>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>06955efb-3963-416e-8a61-24c298f2f108</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>ddd71943-6ad8-4c96-8935-0caba920da00</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>5932ffee-c02e-4816-8e2f-1dd42d15ee31</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>7bbe5119-4d51-4810-893c-0e0797c8219b</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>9f3664ec-db31-46ae-b9f3-c06af39bda0e</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>b9df3743-3d7a-4ebd-89e3-85b22bb930ac</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>b089ade7-05a6-4dbf-97c2-b9eb97628d17</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  06955efb-3963-416e-8a61-24c298f2f108   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  ddd71943-6ad8-4c96-8935-0caba920da00   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  5932ffee-c02e-4816-8e2f-1dd42d15ee31   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  7bbe5119-4d51-4810-893c-0e0797c8219b   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  9f3664ec-db31-46ae-b9f3-c06af39bda0e   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b9df3743-3d7a-4ebd-89e3-85b22bb930ac   \n",
       "\n",
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b089ade7-05a6-4dbf-97c2-b9eb97628d17   \n",
       "\n",
       "                                                   disease  \n",
       "run_id                                                      \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...    PDAC  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for c in metadata_table.columns:\n",
    "    run_table[c] = [metadata_table.loc[x, c] for x in run_table['case_id']]\n",
    "run_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bd95a01",
   "metadata": {
    "hidden": true
   },
   "source": [
    "wxs fastq runlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "511f0205",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'disease', 'wxs_normal_R1.filepath', 'wxs_normal_R1.uuid', 'wxs_normal_R2.filepath', 'wxs_normal_R2.uuid', 'wxs_tumor_R1.filepath', 'wxs_tumor_R1.uuid', 'wxs_tumor_R2.filepath', 'wxs_tumor_R2.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_TN_wxs_fq/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0f0aa1a",
   "metadata": {
    "hidden": true
   },
   "source": [
    "rnaseq fq runlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1933f042",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "filtered = run_table[['case_id', 'run_uuid', 'disease', 'rna-seq_tumor_R1.filepath', 'rna-seq_tumor_R1.uuid', 'rna-seq_tumor_R2.filepath', 'rna-seq_tumor_R2.uuid']]\n",
    "filtered.to_csv('../tests/data/pecgs_T_rna_fq/run_list.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "ab4f6a69",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>project</th>\n",
       "      <th>disease</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97628d17</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>b089ade7-05a6-4dbf-97c2-b9eb97628d17</td>\n",
       "      <td>TCGA</td>\n",
       "      <td>PAAD</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>5932ffee-c02e-4816-8e2f-1dd42d15ee31</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>7bbe5119-4d51-4810-893c-0e0797c8219b</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>9f3664ec-db31-46ae-b9f3-c06af39bda0e</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>b9df3743-3d7a-4ebd-89e3-85b22bb930ac</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b089ade7-05a6-4dbf-97c2-b9eb97628d17   \n",
       "\n",
       "                                                   project disease  \\\n",
       "run_id                                                               \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...    TCGA    PAAD   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  5932ffee-c02e-4816-8e2f-1dd42d15ee31   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  7bbe5119-4d51-4810-893c-0e0797c8219b   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  9f3664ec-db31-46ae-b9f3-c06af39bda0e   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b9df3743-3d7a-4ebd-89e3-85b22bb930ac  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('../tests/data/pecgs_TN_wxs_fq/run_list.txt', sep='\\t', index_col=0)\n",
    "# df['project'] = 'TCGA'\n",
    "# df['disease'] = 'PAAD'\n",
    "# df = df[['case_id', 'run_uuid', 'project', 'disease', 'wxs_normal_R1.filepath', 'wxs_normal_R1.uuid', 'wxs_normal_R2.filepath', 'wxs_normal_R2.uuid', 'wxs_tumor_R1.filepath', 'wxs_tumor_R1.uuid', 'wxs_tumor_R2.filepath', 'wxs_tumor_R2.uuid']]\n",
    "# df.to_csv('../tests/data/pecgs_TN_wxs_fq/run_list.txt', sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "75e89f5f",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>project</th>\n",
       "      <th>disease</th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97628d17</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>b089ade7-05a6-4dbf-97c2-b9eb97628d17</td>\n",
       "      <td>TCGA</td>\n",
       "      <td>PAAD</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>06955efb-3963-416e-8a61-24c298f2f108</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>ddd71943-6ad8-4c96-8935-0caba920da00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b089ade7-05a6-4dbf-97c2-b9eb97628d17   \n",
       "\n",
       "                                                   project disease  \\\n",
       "run_id                                                               \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...    TCGA    PAAD   \n",
       "\n",
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  06955efb-3963-416e-8a61-24c298f2f108   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  ddd71943-6ad8-4c96-8935-0caba920da00  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('../tests/data/pecgs_T_rna_fq/run_list.txt', sep='\\t', index_col=0)\n",
    "# df['project'] = 'TCGA'\n",
    "# df['disease'] = 'PAAD'\n",
    "# df = df[['case_id', 'run_uuid', 'project', 'disease', 'rna-seq_tumor_R1.filepath', 'rna-seq_tumor_R1.uuid', 'rna-seq_tumor_R2.filepath', 'rna-seq_tumor_R2.uuid']]\n",
    "# df.to_csv('../tests/data/pecgs_T_rna_fq/run_list.txt', sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "893c438b",
   "metadata": {
    "hidden": true
   },
   "source": [
    "if we want the .bams to have correct readgroup ids, etc. then we need to pass some sequencing info. this info is available in the Samplemap.csv from MGI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8de72505",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/data/pecgs/test_sample/ht191p1-s1h1a3/Samplemap.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[0;32mIn [27]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m/data/pecgs/test_sample/ht191p1-s1h1a3/Samplemap.csv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m,\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      2\u001b[0m df\n",
      "File \u001b[0;32m/diskmnt/Projects/Users/estorrs/miniconda3/envs/pecgs/lib/python3.8/site-packages/pandas/util/_decorators.py:311\u001b[0m, in \u001b[0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    305\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(args) \u001b[38;5;241m>\u001b[39m num_allow_args:\n\u001b[1;32m    306\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m    307\u001b[0m         msg\u001b[38;5;241m.\u001b[39mformat(arguments\u001b[38;5;241m=\u001b[39marguments),\n\u001b[1;32m    308\u001b[0m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m,\n\u001b[1;32m    309\u001b[0m         stacklevel\u001b[38;5;241m=\u001b[39mstacklevel,\n\u001b[1;32m    310\u001b[0m     )\n\u001b[0;32m--> 311\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/diskmnt/Projects/Users/estorrs/miniconda3/envs/pecgs/lib/python3.8/site-packages/pandas/io/parsers/readers.py:680\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m    665\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    666\u001b[0m     dialect,\n\u001b[1;32m    667\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    676\u001b[0m     defaults\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdelimiter\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m},\n\u001b[1;32m    677\u001b[0m )\n\u001b[1;32m    678\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 680\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/diskmnt/Projects/Users/estorrs/miniconda3/envs/pecgs/lib/python3.8/site-packages/pandas/io/parsers/readers.py:575\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    572\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    574\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 575\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    577\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    578\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m/diskmnt/Projects/Users/estorrs/miniconda3/envs/pecgs/lib/python3.8/site-packages/pandas/io/parsers/readers.py:933\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    930\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m    932\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m--> 933\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/diskmnt/Projects/Users/estorrs/miniconda3/envs/pecgs/lib/python3.8/site-packages/pandas/io/parsers/readers.py:1217\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1213\u001b[0m     mode \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1214\u001b[0m \u001b[38;5;66;03m# error: No overload variant of \"get_handle\" matches argument types\u001b[39;00m\n\u001b[1;32m   1215\u001b[0m \u001b[38;5;66;03m# \"Union[str, PathLike[str], ReadCsvBuffer[bytes], ReadCsvBuffer[str]]\"\u001b[39;00m\n\u001b[1;32m   1216\u001b[0m \u001b[38;5;66;03m# , \"str\", \"bool\", \"Any\", \"Any\", \"Any\", \"Any\", \"Any\"\u001b[39;00m\n\u001b[0;32m-> 1217\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\u001b[43m  \u001b[49m\u001b[38;5;66;43;03m# type: ignore[call-overload]\u001b[39;49;00m\n\u001b[1;32m   1218\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1219\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1220\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1221\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1222\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1223\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1224\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1225\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1226\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1227\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1228\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m/diskmnt/Projects/Users/estorrs/miniconda3/envs/pecgs/lib/python3.8/site-packages/pandas/io/common.py:789\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    784\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    785\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    786\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    787\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    788\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 789\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    796\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    797\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    798\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/data/pecgs/test_sample/ht191p1-s1h1a3/Samplemap.csv'"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('/data/pecgs/test_sample/ht191p1-s1h1a3/Samplemap.csv', sep=',')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "273668b3",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>lane</th>\n",
       "      <th>index_sequencer</th>\n",
       "      <th>library_preparation</th>\n",
       "      <th>platform</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>tumor</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>normal</td>\n",
       "      <td>HH7KNDSXY</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTATCGCT-CTTGAAGGTT</td>\n",
       "      <td>TWCE-HT191P1-JM1D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           sample_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "\n",
       "                                                   experimental_strategy  \\\n",
       "run_id                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "\n",
       "                                                   sample_type   flowcell  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       tumor  HFMFWDSXY   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...      normal  HH7KNDSXY   \n",
       "\n",
       "                                                   lane  \\\n",
       "run_id                                                    \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...    2   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...    1   \n",
       "\n",
       "                                                          index_sequencer  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CCAGTAGCGT-ATGTATTGGC   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CATTATCGCT-CTTGAAGGTT   \n",
       "\n",
       "                                                               library_preparation  \\\n",
       "run_id                                                                               \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  TWCE-HT191P1-S1H1A3Y3D1_1-lib1   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       TWCE-HT191P1-JM1D1_1-lib1   \n",
       "\n",
       "                                                    platform  \n",
       "run_id                                                        \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = [\n",
    "    ['HT191P1-S1H1A3Y3', '419b8098-e4b2-4318-883d-c233cb7e06c8', 'wxs', 'tumor', 'HFMFWDSXY', '2', 'CCAGTAGCGT-ATGTATTGGC', 'TWCE-HT191P1-S1H1A3Y3D1_1-lib1', 'ILLUMINA'],\n",
    "    ['HT191P1-S1H1A3Y3', '419b8098-e4b2-4318-883d-c233cb7e06c8', 'wxs', 'normal', 'HH7KNDSXY', '1', 'CATTATCGCT-CTTGAAGGTT', 'TWCE-HT191P1-JM1D1_1-lib1', 'ILLUMINA']\n",
    "]\n",
    "sq_df = pd.DataFrame(data=data, columns=['sample_id', 'run_uuid', 'experimental_strategy', 'sample_type',\n",
    "                                        'flowcell', 'lane', 'index_sequencer', 'library_preparation', 'platform'])\n",
    "sq_df.index = [f'{s}_{u}' for s, u in zip(sq_df['sample_id'], sq_df['run_uuid'])]\n",
    "sq_df.index.name = 'run_id'\n",
    "sq_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7f5e6cad",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "sq_df.to_csv('../tests/data/pecgs_TN_wxs_fq/sequencing_info.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "ec24a820",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>lane</th>\n",
       "      <th>index_sequencer</th>\n",
       "      <th>library_preparation</th>\n",
       "      <th>platform</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97628d17</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>b089ade7-05a6-4dbf-97c2-b9eb97628d17</td>\n",
       "      <td>wxs</td>\n",
       "      <td>tumor</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97628d17</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>b089ade7-05a6-4dbf-97c2-b9eb97628d17</td>\n",
       "      <td>wxs</td>\n",
       "      <td>normal</td>\n",
       "      <td>HH7KNDSXY</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTATCGCT-CTTGAAGGTT</td>\n",
       "      <td>TWCE-HT191P1-JM1D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           sample_id  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  HT191P1-S1H1A3Y3   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b089ade7-05a6-4dbf-97c2-b9eb97628d17   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  b089ade7-05a6-4dbf-97c2-b9eb97628d17   \n",
       "\n",
       "                                                   experimental_strategy  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...                   wxs   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...                   wxs   \n",
       "\n",
       "                                                   sample_type   flowcell  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...       tumor  HFMFWDSXY   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...      normal  HH7KNDSXY   \n",
       "\n",
       "                                                    lane  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...     2   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...     1   \n",
       "\n",
       "                                                          index_sequencer  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  CCAGTAGCGT-ATGTATTGGC   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  CATTATCGCT-CTTGAAGGTT   \n",
       "\n",
       "                                                               library_preparation  \\\n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  TWCE-HT191P1-S1H1A3Y3D1_1-lib1   \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...       TWCE-HT191P1-JM1D1_1-lib1   \n",
       "\n",
       "                                                    platform  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  ILLUMINA  \n",
       "HT191P1-S1H1A3Y3_b089ade7-05a6-4dbf-97c2-b9eb97...  ILLUMINA  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('../tests/data/pecgs_TN_wxs_fq/sequencing_info.txt', sep='\\t', index_col=0)\n",
    "df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85690a09",
   "metadata": {},
   "source": [
    "###### save defaults"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9bf067e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "resources_dir = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs_resources'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "21872288",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "    'cpu': 40,\n",
    "    \n",
    "    # align dna-seq\n",
    "    'known_sites': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/dnaseq_alignment/dbsnp/00-All.chr.vcf.gz',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': f'{resources_dir}/dnaseq_alignment/dbsnp/00-All.chr.vcf.gz.tbi'}\n",
    "        ]\n",
    "    },\n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    'platform': 'ILLUMINA',\n",
    "    \n",
    "    # cnv\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/snp_database/af-only-gnomad.hg38.common_biallelic.chr1-22XY.vcf'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/pon/pecgs_pon_v1/create_pon/gatk4scnaPON.Normal.hdf5'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/gencode.v34.annotation.gene_filtered.protein_coding.ensembl_ID_no_version.protein-coding_hgnc_filtered.duplicates_removed.ensembl_ID_removed.txt'\n",
    "    },\n",
    "    'cytoband': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/cytoBand.txt'\n",
    "    },\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/IDT_xGen_Exome_Research_Panel_v1.hg38.removed_alt_chr.autosome_only.bed.preprocessed.exome.interval_list'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/PECGS/ref_genome/hg38.microsatellite'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 1,\n",
    "    'minimal_microsatellite_size': 1,\n",
    "    \n",
    "    # tindaisy\n",
    "    'clinvar_annotation': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/clinvar/GRCh38/clinvar_20200706.vcf.gz'\n",
    "    },\n",
    "    'rescue_clinvar': True,\n",
    "    'rescue_cosmic': True,\n",
    "    'call_regions': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed.gz'\n",
    "    },\n",
    "    'canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'pindel_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/pindel.WES.ini'\n",
    "    },\n",
    "    'strelka_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/strelka.WES.ini'\n",
    "    },\n",
    "    'varscan_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/varscan.WES.ini'\n",
    "    },\n",
    "    'classification_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/classification_filter_config.ini'\n",
    "    },\n",
    "    'af_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/af_filter_config.ini'\n",
    "    },\n",
    "    'centromere_bed': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tindaisy_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/20220525/vep-cache.102_GRCh38.tar.gz'\n",
    "    },\n",
    "    'tindaisy_vep_cache_version': 102,\n",
    "    'assembly': 'GRCh38',\n",
    "    'tindaisy_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.d1.vd1.chrlist.txt'\n",
    "    },\n",
    "    'tindaisy_rescue_bed': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/bed/tindaisy_vaf_rescue/299_drivers.bed'\n",
    "    },\n",
    "    \n",
    "    # TinJasmine\n",
    "    'centromere': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tinjasmine_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/chrlist/GRCh38.d1.vd1.chrlist-reordered.txt'\n",
    "    },\n",
    "    'pindel_config_template': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_config_template/pindel_germline_filter_config.ini'\n",
    "    },\n",
    "    'Canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/Canonical_BED/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'ROI_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/ROI_BED/Homo_sapiens.GRCh38.95.allCDS.2bpFlanks.biomart.withCHR.bed'\n",
    "    },\n",
    "#     'varscan_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/varscan_filter_config/VLD_FilterVCF-varscan.config.ini'\n",
    "#     },\n",
    "#     'pindel_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_filter_config/VLD_FilterVCF-pindel.config.ini'\n",
    "#     },\n",
    "#     'gatk_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/gatk_filter_config/VLD_FilterVCF-GATK.config.ini'\n",
    "#     },\n",
    "    'tinjasmine_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/20220525/vep-cache.102_GRCh38.tar.gz'\n",
    "    },\n",
    "    \n",
    "    # neoscan\n",
    "    'neoscan_ref_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29'\n",
    "    },\n",
    "    'neoscan_bed': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29/proteome.bed'\n",
    "    },\n",
    "    \n",
    "    # charger\n",
    "    'charger_inheritance_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cancer_pred_genes_160genes_011321_curated_forCharGer.txt'\n",
    "    },\n",
    "    'charger_pp2_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/160cpgs.txt'\n",
    "    },\n",
    "    'charger_pathogenic_variants': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/emptyRemoved_20160428_pathogenic_variants_HGVSg_VEP_grch38lifOver.vcf'\n",
    "    },\n",
    "    'charger_hotspot3d_clusters': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cptac_mc3_combined_noHypers_sorted.maf.3D_Proximity.pairwise.recurrence.l0.ad10.r10.clusters'\n",
    "    },\n",
    "    'charger_clinvar_alleles': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/clinvar_alleles.single.b38.tsv.gz'\n",
    "    },\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "07023c90",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('/diskmnt/Projects/Users/estorrs/pecgs-pipeline/cwl/pecgs_workflows/defaults.pecgs_TN_wxs_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fedbeeb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_TN_wxs_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0caca783",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "    'cpu': 40,\n",
    "    \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    \n",
    "    # cnv\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/snp_database/af-only-gnomad.hg38.common_biallelic.chr1-22XY.vcf'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/pon/pecgs_pon_v1/create_pon/gatk4scnaPON.Normal.hdf5'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/gencode.v34.annotation.gene_filtered.protein_coding.ensembl_ID_no_version.protein-coding_hgnc_filtered.duplicates_removed.ensembl_ID_removed.txt'\n",
    "    },\n",
    "    'cytoband': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/cytoBand.txt'\n",
    "    },\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/db/IDT_xGen_Exome_Research_Panel_v1.hg38.removed_alt_chr.autosome_only.bed.preprocessed.exome.interval_list'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/PECGS/ref_genome/hg38.microsatellite'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 1,\n",
    "    'minimal_microsatellite_size': 1,\n",
    "    \n",
    "    # tindaisy\n",
    "    'clinvar_annotation': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/clinvar/GRCh38/clinvar_20200706.vcf.gz'\n",
    "    },\n",
    "    'rescue_clinvar': True,\n",
    "    'rescue_cosmic': True,\n",
    "    'call_regions': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed.gz'\n",
    "    },\n",
    "    'canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'pindel_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/pindel.WES.ini'\n",
    "    },\n",
    "    'strelka_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/strelka.WES.ini'\n",
    "    },\n",
    "    'varscan_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/caller_config/varscan.WES.ini'\n",
    "    },\n",
    "    'classification_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/classification_filter_config.ini'\n",
    "    },\n",
    "    'af_config': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/filter_config/af_filter_config.ini'\n",
    "    },\n",
    "    'centromere_bed': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tindaisy_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/20220525/vep-cache.102_GRCh38.tar.gz'\n",
    "    },\n",
    "    'tindaisy_vep_cache_version': 102,\n",
    "    'assembly': 'GRCh38',\n",
    "    'tindaisy_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/submodules/TinDaisy/params/chrlist/GRCh38.d1.vd1.chrlist.txt'\n",
    "    },\n",
    "    'tindaisy_rescue_bed': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/bed/tindaisy_vaf_rescue/299_drivers.bed'\n",
    "    },\n",
    "    \n",
    "    # TinJasmine\n",
    "    'centromere': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/centromere/ucsc-centromere.GRCh38.bed'\n",
    "    },\n",
    "    'tinjasmine_chrlist': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/chrlist/GRCh38.d1.vd1.chrlist-reordered.txt'\n",
    "    },\n",
    "    'pindel_config_template': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_config_template/pindel_germline_filter_config.ini'\n",
    "    },\n",
    "    'Canonical_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/Canonical_BED/GRCh38.callRegions.bed'\n",
    "    },\n",
    "    'ROI_BED': {\n",
    "        'class': 'File',\n",
    "        'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/ROI_BED/Homo_sapiens.GRCh38.95.allCDS.2bpFlanks.biomart.withCHR.bed'\n",
    "    },\n",
    "#     'varscan_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/varscan_filter_config/VLD_FilterVCF-varscan.config.ini'\n",
    "#     },\n",
    "#     'pindel_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/pindel_filter_config/VLD_FilterVCF-pindel.config.ini'\n",
    "#     },\n",
    "#     'gatk_filter_config': {\n",
    "#         'class': 'File',\n",
    "#         'path': '/storage1/fs1/dinglab/Active/Resources/Pipelines/TinJasmine/TinJasmine_etc/gatk_filter_config/VLD_FilterVCF-GATK.config.ini'\n",
    "#     },\n",
    "    'tinjasmine_vep_cache_gz': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/vep/20220525/vep-cache.102_GRCh38.tar.gz'\n",
    "    },\n",
    "    \n",
    "    # neoscan\n",
    "    'neoscan_ref_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29'\n",
    "    },\n",
    "    'neoscan_bed': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/neoscan/refseq_hg38_june29/proteome.bed'\n",
    "    },\n",
    "    \n",
    "    # charger\n",
    "    'charger_inheritance_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cancer_pred_genes_160genes_011321_curated_forCharGer.txt'\n",
    "    },\n",
    "    'charger_pp2_gene_list': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/160cpgs.txt'\n",
    "    },\n",
    "    'charger_pathogenic_variants': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/emptyRemoved_20160428_pathogenic_variants_HGVSg_VEP_grch38lifOver.vcf'\n",
    "    },\n",
    "    'charger_hotspot3d_clusters': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/PanCan/cptac_mc3_combined_noHypers_sorted.maf.3D_Proximity.pairwise.recurrence.l0.ad10.r10.clusters'\n",
    "    },\n",
    "    'charger_clinvar_alleles': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/charger/CharGer_dependencyFiles/clinvar_alleles.single.b38.tsv.gz'\n",
    "    },\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "71c90817",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('/diskmnt/Projects/Users/estorrs/pecgs-pipeline/cwl/pecgs_workflows/defaults.pecgs_TN_wxs_bam.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0af40fef",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_TN_wxs_bam.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "dbe514f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "    'cpu': 40,\n",
    "    \n",
    "    # fusion\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/Integrate_dependencies/bwts'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/FilterDatabase'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/FusionAnnotator'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/ericscript_dependencies/ericscript_db_homosapiens_ensembl84'\n",
    "    },\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/fusion/STAR-Fusion_dependencies/GRCh38_gencode_v37_CTAT_lib_Mar012021.plug-n-play/ctat_genome_lib_build_dir'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/fusion/Integrate_dependencies/annot.ensembl.GRCh38.txt'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/fusion/INTEGRATE_0_2_6/INTEGRATE-build/bin/Integrate'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/fusion/Integrate_dependencies/STAR/hg38.fa'\n",
    "    },\n",
    "    \n",
    "    # bulk expression\n",
    "    'star_index': {\n",
    "        'class': 'Directory',\n",
    "        'path': f'{resources_dir}/bulk_expression/STAR_GRCh38_d1_vd1_d1_GENCODE_v34'\n",
    "    },\n",
    "    'gtf': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/bulk_expression/STAR_GRCh38_d1_vd1_d1_GENCODE_v34/gencode.v34.annotation.gtf'\n",
    "    },\n",
    "    'gene_info': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/bulk_expression/STAR_GRCh38_d1_vd1_d1_GENCODE_v34/gencode.gene.info.v34.tsv'\n",
    "    },\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "42947ee4",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('/diskmnt/Projects/Users/estorrs/pecgs-pipeline/cwl/pecgs_workflows/defaults.pecgs_T_rna_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b7602d3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_T_rna_fq.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "81c73d7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "pecgs_storage1_defaults = {\n",
    "    # general\n",
    "#     'cpu': 40,\n",
    "    \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': f'{resources_dir}/cnv/references/GRCh38.d1.vd1/GRCh38.d1.vd1.fa',\n",
    "    },\n",
    "    \n",
    "    # somatic sv\n",
    "    'generate_evidence_bam': True,\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e8e42887",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(pecgs_storage1_defaults, open('../wombat/templates/compute1.defaults.pecgs_TN_wgs_bam.yaml', 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03ec54fb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
