{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "50f1c34f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pathlib import Path\n",
    "\n",
    "import pandas as pd\n",
    "import mygene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a952fb1a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c1867a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_dir = '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files'\n",
    "Path(out_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "00e65d7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def result_to_df(result):\n",
    "    data = []\n",
    "    for r in result:\n",
    "        genomic_pos = r['genomic_pos']\n",
    "        if isinstance(genomic_pos, list):\n",
    "            genomic_pos = genomic_pos[0]\n",
    "        data.append([genomic_pos['chr'], genomic_pos['start'], genomic_pos['end'], r['query']])\n",
    "    df = pd.DataFrame(data=data, columns=['chr', 'start', 'stop', 'gene'])\n",
    "    df = df.sort_values(['chr', 'start'])\n",
    "    df['chr'] = [f'chr{x}' for x in df['chr']]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1397284",
   "metadata": {},
   "source": [
    "## CHOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d4b1dd61",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>SMG</th>\n",
       "      <th>Match type</th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Location</th>\n",
       "      <th>driver_gene_299</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS proto-oncogene, GTPase</td>\n",
       "      <td>HGNC:6407</td>\n",
       "      <td>12p12.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>isocitrate dehydrogenase (NADP(+)) 1</td>\n",
       "      <td>HGNC:5382</td>\n",
       "      <td>2q34</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>IDH2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>IDH2</td>\n",
       "      <td>isocitrate dehydrogenase (NADP(+)) 2</td>\n",
       "      <td>HGNC:5383</td>\n",
       "      <td>15q26.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>FGFR2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>FGFR2</td>\n",
       "      <td>fibroblast growth factor receptor 2</td>\n",
       "      <td>HGNC:3689</td>\n",
       "      <td>10q26.13</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>BRCA1 associated protein 1</td>\n",
       "      <td>HGNC:950</td>\n",
       "      <td>3p21.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TP53</td>\n",
       "      <td>tumor protein p53</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>17p13.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>AT-rich interaction domain 1A</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>PBRM1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>PBRM1</td>\n",
       "      <td>polybromo 1</td>\n",
       "      <td>HGNC:30064</td>\n",
       "      <td>3p21.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>EPHA2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>EPHA2</td>\n",
       "      <td>EPH receptor A2</td>\n",
       "      <td>HGNC:3386</td>\n",
       "      <td>1p36.13</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>RASA1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>RASA1</td>\n",
       "      <td>RAS p21 protein activator 1</td>\n",
       "      <td>HGNC:9871</td>\n",
       "      <td>5q14.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>RB1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>RB1</td>\n",
       "      <td>RB transcriptional corepressor 1</td>\n",
       "      <td>HGNC:9884</td>\n",
       "      <td>13q14.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>SMAD4</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SMAD4</td>\n",
       "      <td>SMAD family member 4</td>\n",
       "      <td>HGNC:6770</td>\n",
       "      <td>18q21.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>HKDC1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>HKDC1</td>\n",
       "      <td>hexokinase domain containing 1</td>\n",
       "      <td>HGNC:23302</td>\n",
       "      <td>10q22.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>SLC16A3</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SLC16A3</td>\n",
       "      <td>solute carrier family 16 member 3</td>\n",
       "      <td>HGNC:10924</td>\n",
       "      <td>17q25.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>phosphatase and tensin homolog</td>\n",
       "      <td>HGNC:9588</td>\n",
       "      <td>10q23.31</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>ECE2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ECE2</td>\n",
       "      <td>endothelin converting enzyme 2</td>\n",
       "      <td>HGNC:13275</td>\n",
       "      <td>3q27.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>EPPK1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>EPPK1</td>\n",
       "      <td>epiplakin 1</td>\n",
       "      <td>HGNC:15577</td>\n",
       "      <td>8q24.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>FYN</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>FYN</td>\n",
       "      <td>FYN proto-oncogene, Src family tyrosine kinase</td>\n",
       "      <td>HGNC:4037</td>\n",
       "      <td>6q21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>STK11</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>STK11</td>\n",
       "      <td>serine/threonine kinase 11</td>\n",
       "      <td>HGNC:11389</td>\n",
       "      <td>19p13.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>TERT</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TERT</td>\n",
       "      <td>telomerase reverse transcriptase</td>\n",
       "      <td>HGNC:11730</td>\n",
       "      <td>5p15.33</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CHOL</td>\n",
       "      <td>APC</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>APC</td>\n",
       "      <td>APC regulator of WNT signaling pathway</td>\n",
       "      <td>HGNC:583</td>\n",
       "      <td>5q22.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Cancer      SMG       Match type Approved symbol  \\\n",
       "0    CHOL     KRAS  Approved symbol            KRAS   \n",
       "1    CHOL     IDH1  Approved symbol            IDH1   \n",
       "2    CHOL     IDH2  Approved symbol            IDH2   \n",
       "3    CHOL    FGFR2  Approved symbol           FGFR2   \n",
       "4    CHOL     BAP1  Approved symbol            BAP1   \n",
       "5    CHOL     TP53  Approved symbol            TP53   \n",
       "6    CHOL   ARID1A  Approved symbol          ARID1A   \n",
       "7    CHOL    PBRM1  Approved symbol           PBRM1   \n",
       "8    CHOL    EPHA2  Approved symbol           EPHA2   \n",
       "9    CHOL    RASA1  Approved symbol           RASA1   \n",
       "10   CHOL      RB1  Approved symbol             RB1   \n",
       "11   CHOL    SMAD4  Approved symbol           SMAD4   \n",
       "12   CHOL    HKDC1  Approved symbol           HKDC1   \n",
       "13   CHOL  SLC16A3  Approved symbol         SLC16A3   \n",
       "14   CHOL     PTEN  Approved symbol            PTEN   \n",
       "15   CHOL     ECE2  Approved symbol            ECE2   \n",
       "16   CHOL    EPPK1  Approved symbol           EPPK1   \n",
       "17   CHOL      FYN  Approved symbol             FYN   \n",
       "18   CHOL    STK11  Approved symbol           STK11   \n",
       "19   CHOL     TERT  Approved symbol            TERT   \n",
       "20   CHOL      APC  Approved symbol             APC   \n",
       "\n",
       "                                     Approved name     HGNC ID  Location  \\\n",
       "0                      KRAS proto-oncogene, GTPase   HGNC:6407   12p12.1   \n",
       "1             isocitrate dehydrogenase (NADP(+)) 1   HGNC:5382      2q34   \n",
       "2             isocitrate dehydrogenase (NADP(+)) 2   HGNC:5383   15q26.1   \n",
       "3              fibroblast growth factor receptor 2   HGNC:3689  10q26.13   \n",
       "4                       BRCA1 associated protein 1    HGNC:950    3p21.1   \n",
       "5                                tumor protein p53  HGNC:11998   17p13.1   \n",
       "6                    AT-rich interaction domain 1A  HGNC:11110   1p36.11   \n",
       "7                                      polybromo 1  HGNC:30064    3p21.1   \n",
       "8                                  EPH receptor A2   HGNC:3386   1p36.13   \n",
       "9                      RAS p21 protein activator 1   HGNC:9871    5q14.3   \n",
       "10                RB transcriptional corepressor 1   HGNC:9884   13q14.2   \n",
       "11                            SMAD family member 4   HGNC:6770   18q21.2   \n",
       "12                  hexokinase domain containing 1  HGNC:23302   10q22.1   \n",
       "13               solute carrier family 16 member 3  HGNC:10924   17q25.3   \n",
       "14                  phosphatase and tensin homolog   HGNC:9588  10q23.31   \n",
       "15                  endothelin converting enzyme 2  HGNC:13275    3q27.1   \n",
       "16                                     epiplakin 1  HGNC:15577    8q24.3   \n",
       "17  FYN proto-oncogene, Src family tyrosine kinase   HGNC:4037      6q21   \n",
       "18                      serine/threonine kinase 11  HGNC:11389   19p13.3   \n",
       "19                telomerase reverse transcriptase  HGNC:11730   5p15.33   \n",
       "20          APC regulator of WNT signaling pathway    HGNC:583    5q22.2   \n",
       "\n",
       "   driver_gene_299  \n",
       "0              yes  \n",
       "1              yes  \n",
       "2              yes  \n",
       "3              yes  \n",
       "4              yes  \n",
       "5              yes  \n",
       "6              yes  \n",
       "7              yes  \n",
       "8              yes  \n",
       "9              yes  \n",
       "10             yes  \n",
       "11             yes  \n",
       "12             NaN  \n",
       "13             NaN  \n",
       "14             yes  \n",
       "15             NaN  \n",
       "16             NaN  \n",
       "17             NaN  \n",
       "18             yes  \n",
       "19             NaN  \n",
       "20             yes  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = pd.read_csv(\n",
    "#     '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/chol_somatic_SELECTED_DRIVER_genes.tsv',\n",
    "    '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/chol_somatic_smg_list_yizhe_03072023.txt', # slacked from yizhe\n",
    "    sep='\\t'\n",
    ")\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8832a8d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# col = 'HGNC_Gene_Symbol'\n",
    "col = 'Approved symbol'\n",
    "genes = sorted(set([x for x in meta[col] if not pd.isnull(x)]))\n",
    "len(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "74f4d2c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:biothings.client:querying 1-21...\n",
      "INFO:biothings.client:done.\n",
      "INFO:biothings.client:Finished.\n"
     ]
    }
   ],
   "source": [
    "mg = mygene.MyGeneInfo()\n",
    "result = mg.querymany(genes, scopes='symbol', fields='all', species='human')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "94f45ad4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16124337</td>\n",
       "      <td>16156069</td>\n",
       "      <td>EPHA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26693236</td>\n",
       "      <td>26782104</td>\n",
       "      <td>ARID1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10</td>\n",
       "      <td>69220332</td>\n",
       "      <td>69267552</td>\n",
       "      <td>HKDC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10</td>\n",
       "      <td>87862563</td>\n",
       "      <td>87971930</td>\n",
       "      <td>PTEN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10</td>\n",
       "      <td>121478332</td>\n",
       "      <td>121598458</td>\n",
       "      <td>FGFR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr12</td>\n",
       "      <td>25205246</td>\n",
       "      <td>25250936</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48303744</td>\n",
       "      <td>48599436</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90083045</td>\n",
       "      <td>90102477</td>\n",
       "      <td>IDH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7661779</td>\n",
       "      <td>7687538</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr17</td>\n",
       "      <td>82217934</td>\n",
       "      <td>82261129</td>\n",
       "      <td>SLC16A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr18</td>\n",
       "      <td>51028394</td>\n",
       "      <td>51085045</td>\n",
       "      <td>SMAD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1177558</td>\n",
       "      <td>1228431</td>\n",
       "      <td>STK11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>208236229</td>\n",
       "      <td>208266074</td>\n",
       "      <td>IDH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3</td>\n",
       "      <td>52401008</td>\n",
       "      <td>52410008</td>\n",
       "      <td>BAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3</td>\n",
       "      <td>52545352</td>\n",
       "      <td>52685917</td>\n",
       "      <td>PBRM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3</td>\n",
       "      <td>184276011</td>\n",
       "      <td>184293031</td>\n",
       "      <td>ECE2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1253147</td>\n",
       "      <td>1295068</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr5</td>\n",
       "      <td>87267883</td>\n",
       "      <td>87391931</td>\n",
       "      <td>RASA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5</td>\n",
       "      <td>112707498</td>\n",
       "      <td>112846239</td>\n",
       "      <td>APC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr6</td>\n",
       "      <td>111660332</td>\n",
       "      <td>111873452</td>\n",
       "      <td>FYN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>143857324</td>\n",
       "      <td>143878467</td>\n",
       "      <td>EPPK1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr      start       stop     gene\n",
       "4    chr1   16124337   16156069    EPHA2\n",
       "1    chr1   26693236   26782104   ARID1A\n",
       "8   chr10   69220332   69267552    HKDC1\n",
       "13  chr10   87862563   87971930     PTEN\n",
       "6   chr10  121478332  121598458    FGFR2\n",
       "11  chr12   25205246   25250936     KRAS\n",
       "15  chr13   48303744   48599436      RB1\n",
       "10  chr15   90083045   90102477     IDH2\n",
       "20  chr17    7661779    7687538     TP53\n",
       "16  chr17   82217934   82261129  SLC16A3\n",
       "17  chr18   51028394   51085045    SMAD4\n",
       "18  chr19    1177558    1228431    STK11\n",
       "9    chr2  208236229  208266074     IDH1\n",
       "2    chr3   52401008   52410008     BAP1\n",
       "12   chr3   52545352   52685917    PBRM1\n",
       "3    chr3  184276011  184293031     ECE2\n",
       "19   chr5    1253147    1295068     TERT\n",
       "14   chr5   87267883   87391931    RASA1\n",
       "0    chr5  112707498  112846239      APC\n",
       "7    chr6  111660332  111873452      FYN\n",
       "5    chr8  143857324  143878467    EPPK1"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = result_to_df(result)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "04d495c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(os.path.join(out_dir, 'formatted_beds', 'pecgs_chol.bed'), index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5e8545d",
   "metadata": {},
   "source": [
    "## MM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "64df88b5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>SMG</th>\n",
       "      <th>Match.type</th>\n",
       "      <th>Approved.symbol</th>\n",
       "      <th>Approved.name</th>\n",
       "      <th>HGNC.ID</th>\n",
       "      <th>Location</th>\n",
       "      <th>driver_gene_299</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MM</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS proto-oncogene, GTPase</td>\n",
       "      <td>HGNC:6407</td>\n",
       "      <td>12p12.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MM</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>NRAS proto-oncogene, GTPase</td>\n",
       "      <td>HGNC:7989</td>\n",
       "      <td>1p13.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MM</td>\n",
       "      <td>FAM46C</td>\n",
       "      <td>Previous symbol</td>\n",
       "      <td>TENT5C</td>\n",
       "      <td>terminal nucleotidyltransferase 5C</td>\n",
       "      <td>HGNC:24712</td>\n",
       "      <td>1p12</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MM</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>B-Raf proto-oncogene, serine/threonine kinase</td>\n",
       "      <td>HGNC:1097</td>\n",
       "      <td>7q34</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MM</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TP53</td>\n",
       "      <td>tumor protein p53</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>17p13.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MM</td>\n",
       "      <td>DIS3</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>DIS3</td>\n",
       "      <td>DIS3 homolog, exosome endoribonuclease and 3'-...</td>\n",
       "      <td>HGNC:20604</td>\n",
       "      <td>13q21.33</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MM</td>\n",
       "      <td>PRDM1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>PRDM1</td>\n",
       "      <td>PR/SET domain 1</td>\n",
       "      <td>HGNC:9346</td>\n",
       "      <td>6q21</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MM</td>\n",
       "      <td>SP140</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SP140</td>\n",
       "      <td>SP140 nuclear body protein</td>\n",
       "      <td>HGNC:17133</td>\n",
       "      <td>2q37.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MM</td>\n",
       "      <td>EGR1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>EGR1</td>\n",
       "      <td>early growth response 1</td>\n",
       "      <td>HGNC:3238</td>\n",
       "      <td>5q31.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MM</td>\n",
       "      <td>TRAF3</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TRAF3</td>\n",
       "      <td>TNF receptor associated factor 3</td>\n",
       "      <td>HGNC:12033</td>\n",
       "      <td>14q32.32</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MM</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ATM</td>\n",
       "      <td>ATM serine/threonine kinase</td>\n",
       "      <td>HGNC:795</td>\n",
       "      <td>11q22.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MM</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>cyclin D1</td>\n",
       "      <td>HGNC:1582</td>\n",
       "      <td>11q13.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MM</td>\n",
       "      <td>LTB</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>LTB</td>\n",
       "      <td>lymphotoxin beta</td>\n",
       "      <td>HGNC:6711</td>\n",
       "      <td>6p21.33</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MM</td>\n",
       "      <td>IRF4</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>IRF4</td>\n",
       "      <td>interferon regulatory factor 4</td>\n",
       "      <td>HGNC:6119</td>\n",
       "      <td>6p25.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MM</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>fibroblast growth factor receptor 3</td>\n",
       "      <td>HGNC:3690</td>\n",
       "      <td>4p16.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MM</td>\n",
       "      <td>RB1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>RB1</td>\n",
       "      <td>RB transcriptional corepressor 1</td>\n",
       "      <td>HGNC:9884</td>\n",
       "      <td>13q14.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MM</td>\n",
       "      <td>ACTG1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ACTG1</td>\n",
       "      <td>actin gamma 1</td>\n",
       "      <td>HGNC:144</td>\n",
       "      <td>17q25.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MM</td>\n",
       "      <td>CYLD</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>CYLD</td>\n",
       "      <td>CYLD lysine 63 deubiquitinase</td>\n",
       "      <td>HGNC:2584</td>\n",
       "      <td>16q12.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MM</td>\n",
       "      <td>MAX</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>MAX</td>\n",
       "      <td>MYC associated factor X</td>\n",
       "      <td>HGNC:6913</td>\n",
       "      <td>14q23.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MM</td>\n",
       "      <td>ATR</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ATR</td>\n",
       "      <td>ATR serine/threonine kinase</td>\n",
       "      <td>HGNC:882</td>\n",
       "      <td>3q23</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MM</td>\n",
       "      <td>HIST1H1E</td>\n",
       "      <td>Previous symbol</td>\n",
       "      <td>H1-4</td>\n",
       "      <td>H1.4 linker histone, cluster member</td>\n",
       "      <td>HGNC:4718</td>\n",
       "      <td>6p22.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Cancer       SMG       Match.type Approved.symbol  \\\n",
       "0      MM      KRAS  Approved symbol            KRAS   \n",
       "1      MM      NRAS  Approved symbol            NRAS   \n",
       "2      MM    FAM46C  Previous symbol          TENT5C   \n",
       "3      MM      BRAF  Approved symbol            BRAF   \n",
       "4      MM      TP53  Approved symbol            TP53   \n",
       "5      MM      DIS3  Approved symbol            DIS3   \n",
       "6      MM     PRDM1  Approved symbol           PRDM1   \n",
       "7      MM     SP140  Approved symbol           SP140   \n",
       "8      MM      EGR1  Approved symbol            EGR1   \n",
       "9      MM     TRAF3  Approved symbol           TRAF3   \n",
       "10     MM       ATM  Approved symbol             ATM   \n",
       "11     MM     CCND1  Approved symbol           CCND1   \n",
       "12     MM       LTB  Approved symbol             LTB   \n",
       "13     MM      IRF4  Approved symbol            IRF4   \n",
       "14     MM     FGFR3  Approved symbol           FGFR3   \n",
       "15     MM       RB1  Approved symbol             RB1   \n",
       "16     MM     ACTG1  Approved symbol           ACTG1   \n",
       "17     MM      CYLD  Approved symbol            CYLD   \n",
       "18     MM       MAX  Approved symbol             MAX   \n",
       "19     MM       ATR  Approved symbol             ATR   \n",
       "20     MM  HIST1H1E  Previous symbol            H1-4   \n",
       "\n",
       "                                        Approved.name     HGNC.ID  Location  \\\n",
       "0                         KRAS proto-oncogene, GTPase   HGNC:6407   12p12.1   \n",
       "1                         NRAS proto-oncogene, GTPase   HGNC:7989    1p13.2   \n",
       "2                  terminal nucleotidyltransferase 5C  HGNC:24712      1p12   \n",
       "3       B-Raf proto-oncogene, serine/threonine kinase   HGNC:1097      7q34   \n",
       "4                                   tumor protein p53  HGNC:11998   17p13.1   \n",
       "5   DIS3 homolog, exosome endoribonuclease and 3'-...  HGNC:20604  13q21.33   \n",
       "6                                     PR/SET domain 1   HGNC:9346      6q21   \n",
       "7                          SP140 nuclear body protein  HGNC:17133    2q37.1   \n",
       "8                             early growth response 1   HGNC:3238    5q31.2   \n",
       "9                    TNF receptor associated factor 3  HGNC:12033  14q32.32   \n",
       "10                        ATM serine/threonine kinase    HGNC:795   11q22.3   \n",
       "11                                          cyclin D1   HGNC:1582   11q13.3   \n",
       "12                                   lymphotoxin beta   HGNC:6711   6p21.33   \n",
       "13                     interferon regulatory factor 4   HGNC:6119    6p25.3   \n",
       "14                fibroblast growth factor receptor 3   HGNC:3690    4p16.3   \n",
       "15                   RB transcriptional corepressor 1   HGNC:9884   13q14.2   \n",
       "16                                      actin gamma 1    HGNC:144   17q25.3   \n",
       "17                      CYLD lysine 63 deubiquitinase   HGNC:2584   16q12.1   \n",
       "18                            MYC associated factor X   HGNC:6913   14q23.3   \n",
       "19                        ATR serine/threonine kinase    HGNC:882      3q23   \n",
       "20                H1.4 linker histone, cluster member   HGNC:4718    6p22.2   \n",
       "\n",
       "   driver_gene_299  \n",
       "0              yes  \n",
       "1              yes  \n",
       "2              NaN  \n",
       "3              yes  \n",
       "4              yes  \n",
       "5              NaN  \n",
       "6              NaN  \n",
       "7              NaN  \n",
       "8              NaN  \n",
       "9              yes  \n",
       "10             yes  \n",
       "11             yes  \n",
       "12             NaN  \n",
       "13             NaN  \n",
       "14             yes  \n",
       "15             yes  \n",
       "16             NaN  \n",
       "17             yes  \n",
       "18             yes  \n",
       "19             yes  \n",
       "20             yes  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = pd.read_csv(\n",
    "#     '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/mm_somatic_SELECTED_DRIVER_genes.tsv',\n",
    "    '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/mm_somatic_smg_list_yizhe_03072023.txt', # slacked from yizhe\n",
    "    sep='\\t'\n",
    ")\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1db8cd6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# col = 'HGNC_Gene_Symbol'\n",
    "col = 'Approved.symbol'\n",
    "genes = sorted(set([x for x in meta[col] if not pd.isnull(x)]))\n",
    "len(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "2c4058b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # some genes have incorrect symbol name\n",
    "# m = {\n",
    "#     'HIST1H1E': 'H1-4',\n",
    "# }\n",
    "# genes = [m.get(g, g) for g in genes]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "855f5dae",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:biothings.client:querying 1-21...\n",
      "INFO:biothings.client:done.\n",
      "INFO:biothings.client:Finished.\n"
     ]
    }
   ],
   "source": [
    "mg = mygene.MyGeneInfo()\n",
    "result = mg.querymany(genes, scopes='symbol', fields='all', species='human')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "cae3f8f4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>114704469</td>\n",
       "      <td>114716771</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>117606048</td>\n",
       "      <td>117628389</td>\n",
       "      <td>TENT5C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>69641156</td>\n",
       "      <td>69654474</td>\n",
       "      <td>CCND1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr11</td>\n",
       "      <td>108223044</td>\n",
       "      <td>108369102</td>\n",
       "      <td>ATM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr12</td>\n",
       "      <td>25205246</td>\n",
       "      <td>25250936</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr13</td>\n",
       "      <td>48303744</td>\n",
       "      <td>48599436</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr13</td>\n",
       "      <td>72752169</td>\n",
       "      <td>72782096</td>\n",
       "      <td>DIS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr14</td>\n",
       "      <td>65006174</td>\n",
       "      <td>65102695</td>\n",
       "      <td>MAX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr14</td>\n",
       "      <td>102777449</td>\n",
       "      <td>102911500</td>\n",
       "      <td>TRAF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr16</td>\n",
       "      <td>50742050</td>\n",
       "      <td>50801935</td>\n",
       "      <td>CYLD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7661779</td>\n",
       "      <td>7687538</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>81509413</td>\n",
       "      <td>81523847</td>\n",
       "      <td>ACTG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr2</td>\n",
       "      <td>230203110</td>\n",
       "      <td>230313215</td>\n",
       "      <td>SP140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3</td>\n",
       "      <td>142449007</td>\n",
       "      <td>142578733</td>\n",
       "      <td>ATR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr4</td>\n",
       "      <td>1793293</td>\n",
       "      <td>1808872</td>\n",
       "      <td>FGFR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5</td>\n",
       "      <td>138465479</td>\n",
       "      <td>138469303</td>\n",
       "      <td>EGR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr6</td>\n",
       "      <td>391739</td>\n",
       "      <td>411443</td>\n",
       "      <td>IRF4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr6</td>\n",
       "      <td>26156329</td>\n",
       "      <td>26157115</td>\n",
       "      <td>H1-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31580525</td>\n",
       "      <td>31582522</td>\n",
       "      <td>LTB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr6</td>\n",
       "      <td>105993463</td>\n",
       "      <td>106109939</td>\n",
       "      <td>PRDM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140719327</td>\n",
       "      <td>140924929</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr      start       stop    gene\n",
       "14   chr1  114704469  114716771    NRAS\n",
       "18   chr1  117606048  117628389  TENT5C\n",
       "4   chr11   69641156   69654474   CCND1\n",
       "1   chr11  108223044  108369102     ATM\n",
       "11  chr12   25205246   25250936    KRAS\n",
       "16  chr13   48303744   48599436     RB1\n",
       "6   chr13   72752169   72782096    DIS3\n",
       "13  chr14   65006174   65102695     MAX\n",
       "20  chr14  102777449  102911500   TRAF3\n",
       "5   chr16   50742050   50801935    CYLD\n",
       "19  chr17    7661779    7687538    TP53\n",
       "0   chr17   81509413   81523847   ACTG1\n",
       "17   chr2  230203110  230313215   SP140\n",
       "2    chr3  142449007  142578733     ATR\n",
       "8    chr4    1793293    1808872   FGFR3\n",
       "7    chr5  138465479  138469303    EGR1\n",
       "10   chr6     391739     411443    IRF4\n",
       "9    chr6   26156329   26157115    H1-4\n",
       "12   chr6   31580525   31582522     LTB\n",
       "15   chr6  105993463  106109939   PRDM1\n",
       "3    chr7  140719327  140924929    BRAF"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = result_to_df(result)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "db563ee7",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(os.path.join(out_dir, 'formatted_beds', 'pecgs_mm.bed'), index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb8f74f7",
   "metadata": {},
   "source": [
    "## CRC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e591c989",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>SMG</th>\n",
       "      <th>Match type</th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Location</th>\n",
       "      <th>driver_gene_299</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRC</td>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>activin A receptor type 2A</td>\n",
       "      <td>HGNC:173</td>\n",
       "      <td>2q22.3-q23.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRC</td>\n",
       "      <td>AMER1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>AMER1</td>\n",
       "      <td>APC membrane recruitment protein 1</td>\n",
       "      <td>HGNC:26837</td>\n",
       "      <td>Xq11.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRC</td>\n",
       "      <td>APC</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>APC</td>\n",
       "      <td>APC regulator of WNT signaling pathway</td>\n",
       "      <td>HGNC:583</td>\n",
       "      <td>5q22.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRC</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>AT-rich interaction domain 1A</td>\n",
       "      <td>HGNC:11110</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRC</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>B-Raf proto-oncogene, serine/threonine kinase</td>\n",
       "      <td>HGNC:1097</td>\n",
       "      <td>7q34</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CRC</td>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>catenin beta 1</td>\n",
       "      <td>HGNC:2514</td>\n",
       "      <td>3p22.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CRC</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>F-box and WD repeat domain containing 7</td>\n",
       "      <td>HGNC:16712</td>\n",
       "      <td>4q31.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRC</td>\n",
       "      <td>GNAS</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>GNAS</td>\n",
       "      <td>GNAS complex locus</td>\n",
       "      <td>HGNC:4392</td>\n",
       "      <td>20q13.32</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CRC</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRAS proto-oncogene, GTPase</td>\n",
       "      <td>HGNC:6407</td>\n",
       "      <td>12p12.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CRC</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>NRAS proto-oncogene, GTPase</td>\n",
       "      <td>HGNC:7989</td>\n",
       "      <td>1p13.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CRC</td>\n",
       "      <td>PCBP1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>PCBP1</td>\n",
       "      <td>poly(rC) binding protein 1</td>\n",
       "      <td>HGNC:8647</td>\n",
       "      <td>2p13.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CRC</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>phosphatidylinositol-4,5-bisphosphate 3-kinase...</td>\n",
       "      <td>HGNC:8975</td>\n",
       "      <td>3q26.32</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CRC</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>phosphatase and tensin homolog</td>\n",
       "      <td>HGNC:9588</td>\n",
       "      <td>10q23.31</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CRC</td>\n",
       "      <td>SMAD2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SMAD2</td>\n",
       "      <td>SMAD family member 2</td>\n",
       "      <td>HGNC:6768</td>\n",
       "      <td>18q21.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CRC</td>\n",
       "      <td>SMAD4</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SMAD4</td>\n",
       "      <td>SMAD family member 4</td>\n",
       "      <td>HGNC:6770</td>\n",
       "      <td>18q21.2</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CRC</td>\n",
       "      <td>SOX9</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SOX9</td>\n",
       "      <td>SRY-box transcription factor 9</td>\n",
       "      <td>HGNC:11204</td>\n",
       "      <td>17q24.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CRC</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>transcription factor 7 like 2</td>\n",
       "      <td>HGNC:11641</td>\n",
       "      <td>10q25.2-q25.3</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CRC</td>\n",
       "      <td>TGIF1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TGIF1</td>\n",
       "      <td>TGFB induced factor homeobox 1</td>\n",
       "      <td>HGNC:11776</td>\n",
       "      <td>18p11.31</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CRC</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TP53</td>\n",
       "      <td>tumor protein p53</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>17p13.1</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CRC</td>\n",
       "      <td>ZFP36L2</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ZFP36L2</td>\n",
       "      <td>ZFP36 ring finger protein like 2</td>\n",
       "      <td>HGNC:1108</td>\n",
       "      <td>2p21</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CRC</td>\n",
       "      <td>ADAMTS18</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ADAMTS18</td>\n",
       "      <td>ADAM metallopeptidase with thrombospondin type...</td>\n",
       "      <td>HGNC:17110</td>\n",
       "      <td>16q23.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CRC</td>\n",
       "      <td>CBX4</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>CBX4</td>\n",
       "      <td>chromobox 4</td>\n",
       "      <td>HGNC:1554</td>\n",
       "      <td>17q25.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>CRC</td>\n",
       "      <td>CSMD3</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>CSMD3</td>\n",
       "      <td>CUB and Sushi multiple domains 3</td>\n",
       "      <td>HGNC:19291</td>\n",
       "      <td>8q23.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRC</td>\n",
       "      <td>ITGB4</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>ITGB4</td>\n",
       "      <td>integrin subunit beta 4</td>\n",
       "      <td>HGNC:6158</td>\n",
       "      <td>17q25.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CRC</td>\n",
       "      <td>LRP1B</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>LRP1B</td>\n",
       "      <td>LDL receptor related protein 1B</td>\n",
       "      <td>HGNC:6693</td>\n",
       "      <td>2q22.1-q22.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>CRC</td>\n",
       "      <td>SYNE1</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>SYNE1</td>\n",
       "      <td>spectrin repeat containing nuclear envelope pr...</td>\n",
       "      <td>HGNC:17089</td>\n",
       "      <td>6q25.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRC</td>\n",
       "      <td>TAF1L</td>\n",
       "      <td>Approved symbol</td>\n",
       "      <td>TAF1L</td>\n",
       "      <td>TATA-box binding protein associated factor 1 like</td>\n",
       "      <td>HGNC:18056</td>\n",
       "      <td>9p21.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Cancer       SMG       Match type Approved symbol  \\\n",
       "0     CRC    ACVR2A  Approved symbol          ACVR2A   \n",
       "1     CRC     AMER1  Approved symbol           AMER1   \n",
       "2     CRC       APC  Approved symbol             APC   \n",
       "3     CRC    ARID1A  Approved symbol          ARID1A   \n",
       "4     CRC      BRAF  Approved symbol            BRAF   \n",
       "5     CRC    CTNNB1  Approved symbol          CTNNB1   \n",
       "6     CRC     FBXW7  Approved symbol           FBXW7   \n",
       "7     CRC      GNAS  Approved symbol            GNAS   \n",
       "8     CRC      KRAS  Approved symbol            KRAS   \n",
       "9     CRC      NRAS  Approved symbol            NRAS   \n",
       "10    CRC     PCBP1  Approved symbol           PCBP1   \n",
       "11    CRC    PIK3CA  Approved symbol          PIK3CA   \n",
       "12    CRC      PTEN  Approved symbol            PTEN   \n",
       "13    CRC     SMAD2  Approved symbol           SMAD2   \n",
       "14    CRC     SMAD4  Approved symbol           SMAD4   \n",
       "15    CRC      SOX9  Approved symbol            SOX9   \n",
       "16    CRC    TCF7L2  Approved symbol          TCF7L2   \n",
       "17    CRC     TGIF1  Approved symbol           TGIF1   \n",
       "18    CRC      TP53  Approved symbol            TP53   \n",
       "19    CRC   ZFP36L2  Approved symbol         ZFP36L2   \n",
       "20    CRC  ADAMTS18  Approved symbol        ADAMTS18   \n",
       "21    CRC      CBX4  Approved symbol            CBX4   \n",
       "22    CRC     CSMD3  Approved symbol           CSMD3   \n",
       "23    CRC     ITGB4  Approved symbol           ITGB4   \n",
       "24    CRC     LRP1B  Approved symbol           LRP1B   \n",
       "25    CRC     SYNE1  Approved symbol           SYNE1   \n",
       "26    CRC     TAF1L  Approved symbol           TAF1L   \n",
       "\n",
       "                                        Approved name     HGNC ID  \\\n",
       "0                          activin A receptor type 2A    HGNC:173   \n",
       "1                  APC membrane recruitment protein 1  HGNC:26837   \n",
       "2              APC regulator of WNT signaling pathway    HGNC:583   \n",
       "3                       AT-rich interaction domain 1A  HGNC:11110   \n",
       "4       B-Raf proto-oncogene, serine/threonine kinase   HGNC:1097   \n",
       "5                                      catenin beta 1   HGNC:2514   \n",
       "6             F-box and WD repeat domain containing 7  HGNC:16712   \n",
       "7                                  GNAS complex locus   HGNC:4392   \n",
       "8                         KRAS proto-oncogene, GTPase   HGNC:6407   \n",
       "9                         NRAS proto-oncogene, GTPase   HGNC:7989   \n",
       "10                         poly(rC) binding protein 1   HGNC:8647   \n",
       "11  phosphatidylinositol-4,5-bisphosphate 3-kinase...   HGNC:8975   \n",
       "12                     phosphatase and tensin homolog   HGNC:9588   \n",
       "13                               SMAD family member 2   HGNC:6768   \n",
       "14                               SMAD family member 4   HGNC:6770   \n",
       "15                     SRY-box transcription factor 9  HGNC:11204   \n",
       "16                      transcription factor 7 like 2  HGNC:11641   \n",
       "17                     TGFB induced factor homeobox 1  HGNC:11776   \n",
       "18                                  tumor protein p53  HGNC:11998   \n",
       "19                   ZFP36 ring finger protein like 2   HGNC:1108   \n",
       "20  ADAM metallopeptidase with thrombospondin type...  HGNC:17110   \n",
       "21                                        chromobox 4   HGNC:1554   \n",
       "22                   CUB and Sushi multiple domains 3  HGNC:19291   \n",
       "23                            integrin subunit beta 4   HGNC:6158   \n",
       "24                    LDL receptor related protein 1B   HGNC:6693   \n",
       "25  spectrin repeat containing nuclear envelope pr...  HGNC:17089   \n",
       "26  TATA-box binding protein associated factor 1 like  HGNC:18056   \n",
       "\n",
       "         Location driver_gene_299  \n",
       "0    2q22.3-q23.1             yes  \n",
       "1          Xq11.2             yes  \n",
       "2          5q22.2             yes  \n",
       "3         1p36.11             yes  \n",
       "4            7q34             yes  \n",
       "5          3p22.1             yes  \n",
       "6          4q31.3             yes  \n",
       "7        20q13.32             yes  \n",
       "8         12p12.1             yes  \n",
       "9          1p13.2             yes  \n",
       "10         2p13.3             yes  \n",
       "11        3q26.32             yes  \n",
       "12       10q23.31             yes  \n",
       "13        18q21.1             yes  \n",
       "14        18q21.2             yes  \n",
       "15        17q24.3             yes  \n",
       "16  10q25.2-q25.3             yes  \n",
       "17       18p11.31             yes  \n",
       "18        17p13.1             yes  \n",
       "19           2p21             yes  \n",
       "20        16q23.1             NaN  \n",
       "21        17q25.3             NaN  \n",
       "22         8q23.3             NaN  \n",
       "23        17q25.1             NaN  \n",
       "24   2q22.1-q22.2             NaN  \n",
       "25         6q25.2             NaN  \n",
       "26         9p21.1             NaN  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = pd.read_csv(\n",
    "#     '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/crc_somatic_SELECTED_DRIVER_genes.tsv',\n",
    "    '/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/crc_somatic_smg_list_yizhe_03072023.txt', # slacked from yizhe\n",
    "    sep='\\t'\n",
    ")\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0fa10353",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# col = 'HGNC_Gene_Symbol'\n",
    "col = 'Approved symbol'\n",
    "genes = sorted(set([x for x in meta[col] if not pd.isnull(x)]))\n",
    "len(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9ba6ae28",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:biothings.client:querying 1-27...\n",
      "INFO:biothings.client:done.\n",
      "INFO:biothings.client:Finished.\n"
     ]
    }
   ],
   "source": [
    "mg = mygene.MyGeneInfo()\n",
    "result = mg.querymany(genes, scopes='symbol', fields='all', species='human')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4a8021f9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26693236</td>\n",
       "      <td>26782104</td>\n",
       "      <td>ARID1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>114704469</td>\n",
       "      <td>114716771</td>\n",
       "      <td>NRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr10</td>\n",
       "      <td>87862563</td>\n",
       "      <td>87971930</td>\n",
       "      <td>PTEN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr10</td>\n",
       "      <td>112950247</td>\n",
       "      <td>113167678</td>\n",
       "      <td>TCF7L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr12</td>\n",
       "      <td>25205246</td>\n",
       "      <td>25250936</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr16</td>\n",
       "      <td>77247813</td>\n",
       "      <td>77435034</td>\n",
       "      <td>ADAMTS18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7661779</td>\n",
       "      <td>7687538</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr17</td>\n",
       "      <td>72121020</td>\n",
       "      <td>72126416</td>\n",
       "      <td>SOX9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr17</td>\n",
       "      <td>75721328</td>\n",
       "      <td>75757818</td>\n",
       "      <td>ITGB4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr17</td>\n",
       "      <td>79833156</td>\n",
       "      <td>79839440</td>\n",
       "      <td>CBX4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr18</td>\n",
       "      <td>3411608</td>\n",
       "      <td>3459978</td>\n",
       "      <td>TGIF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr18</td>\n",
       "      <td>47808957</td>\n",
       "      <td>47931146</td>\n",
       "      <td>SMAD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr18</td>\n",
       "      <td>51028394</td>\n",
       "      <td>51085045</td>\n",
       "      <td>SMAD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr2</td>\n",
       "      <td>43222402</td>\n",
       "      <td>43226606</td>\n",
       "      <td>ZFP36L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr2</td>\n",
       "      <td>70087477</td>\n",
       "      <td>70089203</td>\n",
       "      <td>PCBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr2</td>\n",
       "      <td>140231423</td>\n",
       "      <td>142131016</td>\n",
       "      <td>LRP1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2</td>\n",
       "      <td>147844517</td>\n",
       "      <td>147930826</td>\n",
       "      <td>ACVR2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr20</td>\n",
       "      <td>58839718</td>\n",
       "      <td>58911192</td>\n",
       "      <td>GNAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3</td>\n",
       "      <td>41194741</td>\n",
       "      <td>41260096</td>\n",
       "      <td>CTNNB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3</td>\n",
       "      <td>179148114</td>\n",
       "      <td>179240093</td>\n",
       "      <td>PIK3CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr4</td>\n",
       "      <td>152320544</td>\n",
       "      <td>152536092</td>\n",
       "      <td>FBXW7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5</td>\n",
       "      <td>112707498</td>\n",
       "      <td>112846239</td>\n",
       "      <td>APC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr6</td>\n",
       "      <td>152121687</td>\n",
       "      <td>152637801</td>\n",
       "      <td>SYNE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140719327</td>\n",
       "      <td>140924929</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr8</td>\n",
       "      <td>112222928</td>\n",
       "      <td>113436939</td>\n",
       "      <td>CSMD3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr9</td>\n",
       "      <td>32629454</td>\n",
       "      <td>32635669</td>\n",
       "      <td>TAF1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>64185117</td>\n",
       "      <td>64205708</td>\n",
       "      <td>AMER1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr      start       stop      gene\n",
       "4    chr1   26693236   26782104    ARID1A\n",
       "14   chr1  114704469  114716771      NRAS\n",
       "17  chr10   87862563   87971930      PTEN\n",
       "23  chr10  112950247  113167678    TCF7L2\n",
       "12  chr12   25205246   25250936      KRAS\n",
       "1   chr16   77247813   77435034  ADAMTS18\n",
       "25  chr17    7661779    7687538      TP53\n",
       "20  chr17   72121020   72126416      SOX9\n",
       "11  chr17   75721328   75757818     ITGB4\n",
       "6   chr17   79833156   79839440      CBX4\n",
       "24  chr18    3411608    3459978     TGIF1\n",
       "18  chr18   47808957   47931146     SMAD2\n",
       "19  chr18   51028394   51085045     SMAD4\n",
       "26   chr2   43222402   43226606   ZFP36L2\n",
       "15   chr2   70087477   70089203     PCBP1\n",
       "13   chr2  140231423  142131016     LRP1B\n",
       "0    chr2  147844517  147930826    ACVR2A\n",
       "10  chr20   58839718   58911192      GNAS\n",
       "8    chr3   41194741   41260096    CTNNB1\n",
       "16   chr3  179148114  179240093    PIK3CA\n",
       "9    chr4  152320544  152536092     FBXW7\n",
       "3    chr5  112707498  112846239       APC\n",
       "21   chr6  152121687  152637801     SYNE1\n",
       "5    chr7  140719327  140924929      BRAF\n",
       "7    chr8  112222928  113436939     CSMD3\n",
       "22   chr9   32629454   32635669     TAF1L\n",
       "2    chrX   64185117   64205708     AMER1"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = result_to_df(result)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "09c9724e",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(os.path.join(out_dir, 'formatted_beds', 'pecgs_crc.bed'), index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59f64aca",
   "metadata": {},
   "source": [
    "## Driver paper cancer types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "bb10a0bd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>KEY</th>\n",
       "      <th>Tumor suppressor or oncogene prediction (by 20/20+)</th>\n",
       "      <th>Decision</th>\n",
       "      <th>Tissue Frequency</th>\n",
       "      <th>Pancan Frequency</th>\n",
       "      <th>Consensus Score</th>\n",
       "      <th>Correlation adusted score</th>\n",
       "      <th>Novel</th>\n",
       "      <th>Rescue Notes</th>\n",
       "      <th>Note about previous publication</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>PANCAN</td>\n",
       "      <td>ABL1_PANCAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rescued</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.17%</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>Evidence from OncoImpact/DriverNET overlap (SN...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACVR1</td>\n",
       "      <td>UCEC</td>\n",
       "      <td>ACVR1_UCEC</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>official</td>\n",
       "      <td>5.30%</td>\n",
       "      <td>0.75%</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACVR1B</td>\n",
       "      <td>PANCAN</td>\n",
       "      <td>ACVR1B_PANCAN</td>\n",
       "      <td>possible tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.09%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Found in 24132290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>COADREAD</td>\n",
       "      <td>ACVR2A_COADREAD</td>\n",
       "      <td>tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>2.85%</td>\n",
       "      <td>1.40%</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Found in 22810696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>LIHC</td>\n",
       "      <td>ACVR2A_LIHC</td>\n",
       "      <td>possible tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>3.11%</td>\n",
       "      <td>1.40%</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Found in private communication about integrati...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>734</th>\n",
       "      <td>ZMYM3</td>\n",
       "      <td>PRAD</td>\n",
       "      <td>ZMYM3_PRAD</td>\n",
       "      <td>tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>1.68%</td>\n",
       "      <td>1.85%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Found in 26544944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>ZNF133</td>\n",
       "      <td>OV</td>\n",
       "      <td>ZNF133_OV</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>official</td>\n",
       "      <td>2.94%</td>\n",
       "      <td>0.61%</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Found in 21720365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>736</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>PANCAN</td>\n",
       "      <td>ZNF750_PANCAN</td>\n",
       "      <td>possible tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.15%</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>ESCA</td>\n",
       "      <td>ZNF750_ESCA</td>\n",
       "      <td>possible tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>6.98%</td>\n",
       "      <td>1.15%</td>\n",
       "      <td>2.5</td>\n",
       "      <td>2.5</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Found in 28052061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>ZNF750</td>\n",
       "      <td>HNSC</td>\n",
       "      <td>ZNF750_HNSC</td>\n",
       "      <td>possible tsg</td>\n",
       "      <td>official</td>\n",
       "      <td>3.78%</td>\n",
       "      <td>1.15%</td>\n",
       "      <td>2.5</td>\n",
       "      <td>2.5</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>739 rows  12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene    Cancer              KEY  \\\n",
       "0      ABL1    PANCAN      ABL1_PANCAN   \n",
       "1     ACVR1      UCEC       ACVR1_UCEC   \n",
       "2    ACVR1B    PANCAN    ACVR1B_PANCAN   \n",
       "3    ACVR2A  COADREAD  ACVR2A_COADREAD   \n",
       "4    ACVR2A      LIHC      ACVR2A_LIHC   \n",
       "..      ...       ...              ...   \n",
       "734   ZMYM3      PRAD       ZMYM3_PRAD   \n",
       "735  ZNF133        OV        ZNF133_OV   \n",
       "736  ZNF750    PANCAN    ZNF750_PANCAN   \n",
       "737  ZNF750      ESCA      ZNF750_ESCA   \n",
       "738  ZNF750      HNSC      ZNF750_HNSC   \n",
       "\n",
       "    Tumor suppressor or oncogene prediction (by 20/20+)  Decision  \\\n",
       "0                                                  NaN    rescued   \n",
       "1                                             oncogene   official   \n",
       "2                                         possible tsg   official   \n",
       "3                                                  tsg   official   \n",
       "4                                         possible tsg   official   \n",
       "..                                                 ...        ...   \n",
       "734                                                tsg   official   \n",
       "735                                           oncogene   official   \n",
       "736                                       possible tsg   official   \n",
       "737                                       possible tsg   official   \n",
       "738                                       possible tsg   official   \n",
       "\n",
       "    Tissue Frequency Pancan Frequency  Consensus Score  \\\n",
       "0                NaN            1.17%              0.0   \n",
       "1              5.30%            0.75%              1.5   \n",
       "2                NaN            1.09%              1.0   \n",
       "3              2.85%            1.40%              1.5   \n",
       "4              3.11%            1.40%              1.5   \n",
       "..               ...              ...              ...   \n",
       "734            1.68%            1.85%              1.0   \n",
       "735            2.94%            0.61%              1.5   \n",
       "736              NaN            1.15%              1.5   \n",
       "737            6.98%            1.15%              2.5   \n",
       "738            3.78%            1.15%              2.5   \n",
       "\n",
       "     Correlation adusted score  Novel  \\\n",
       "0                          NaN      0   \n",
       "1                          1.5      0   \n",
       "2                          0.0      0   \n",
       "3                          1.5      0   \n",
       "4                          1.5      0   \n",
       "..                         ...    ...   \n",
       "734                        0.0      1   \n",
       "735                        1.5      1   \n",
       "736                        1.5      0   \n",
       "737                        2.5      0   \n",
       "738                        2.5      0   \n",
       "\n",
       "                                          Rescue Notes  \\\n",
       "0    Evidence from OncoImpact/DriverNET overlap (SN...   \n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3                                                  NaN   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "734                                                NaN   \n",
       "735                                                NaN   \n",
       "736                                                NaN   \n",
       "737                                                NaN   \n",
       "738                                                NaN   \n",
       "\n",
       "                       Note about previous publication  \n",
       "0                                                  NaN  \n",
       "1                                                    0  \n",
       "2                                    Found in 24132290  \n",
       "3                                    Found in 22810696  \n",
       "4    Found in private communication about integrati...  \n",
       "..                                                 ...  \n",
       "734                                  Found in 26544944  \n",
       "735                                  Found in 21720365  \n",
       "736                                                  0  \n",
       "737                                  Found in 28052061  \n",
       "738                                                  0  \n",
       "\n",
       "[739 rows x 12 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = pd.read_csv('/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/smg.bailey.cell.ct.tsv',\n",
    "                   sep='\\t')\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "e22c917b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACC',\n",
       " 'BLCA',\n",
       " 'BRCA',\n",
       " 'CESC',\n",
       " 'CHOL',\n",
       " 'COADREAD',\n",
       " 'DLBC',\n",
       " 'ESCA',\n",
       " 'GBM',\n",
       " 'HNSC',\n",
       " 'KICH',\n",
       " 'KIRC',\n",
       " 'KIRP',\n",
       " 'LAML',\n",
       " 'LGG',\n",
       " 'LIHC',\n",
       " 'LUAD',\n",
       " 'LUSC',\n",
       " 'MESO',\n",
       " 'OV',\n",
       " 'PAAD',\n",
       " 'PANCAN',\n",
       " 'PCPG',\n",
       " 'PRAD',\n",
       " 'SARC',\n",
       " 'SKCM',\n",
       " 'STAD',\n",
       " 'TGCT',\n",
       " 'THCA',\n",
       " 'THYM',\n",
       " 'UCEC',\n",
       " 'UCS',\n",
       " 'UVM'}"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(meta['Cancer'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "0984073d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# some genes have incorrect symbol name\n",
    "m = {\n",
    "    'CBWD3': 'ZNG1C',\n",
    "    'FAM46D': 'TENT5D',\n",
    "    'H3F3A': 'H3-3A',\n",
    "    'H3F3C': 'H3-5',\n",
    "    'HIST1H1C': 'H1-2',\n",
    "    'HIST1H1E': 'H1-4',\n",
    "    'RQCD1': 'CNOT9',\n",
    "    'TCEB1': 'ELOC',\n",
    "    'WHSC1': 'NSD2'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "182075b2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACC\n",
      "querying 1-5...done.\n",
      "Finished.\n",
      "BLCA\n",
      "querying 1-45...done.\n",
      "Finished.\n",
      "BRCA\n",
      "querying 1-29...done.\n",
      "Finished.\n",
      "CESC\n",
      "querying 1-26...done.\n",
      "Finished.\n",
      "CHOL\n",
      "querying 1-5...done.\n",
      "Finished.\n",
      "COADREAD\n",
      "querying 1-20...done.\n",
      "Finished.\n",
      "DLBC\n",
      "querying 1-13...done.\n",
      "Finished.\n",
      "ESCA\n",
      "querying 1-13...done.\n",
      "Finished.\n",
      "GBM\n",
      "querying 1-17...done.\n",
      "Finished.\n",
      "HNSC\n",
      "querying 1-33...done.\n",
      "Finished.\n",
      "KICH\n",
      "querying 1-2...done.\n",
      "Finished.\n",
      "KIRC\n",
      "querying 1-12...done.\n",
      "Finished.\n",
      "KIRP\n",
      "querying 1-9...done.\n",
      "Finished.\n",
      "LAML\n",
      "querying 1-22...done.\n",
      "Finished.\n",
      "LGG\n",
      "querying 1-24...done.\n",
      "Finished.\n",
      "LIHC\n",
      "querying 1-32...done.\n",
      "Finished.\n",
      "LUAD\n",
      "querying 1-20...done.\n",
      "Finished.\n",
      "LUSC\n",
      "querying 1-22...done.\n",
      "Finished.\n",
      "MESO\n",
      "querying 1-5...done.\n",
      "Finished.\n",
      "OV\n",
      "querying 1-8...done.\n",
      "Finished.\n",
      "PAAD\n",
      "querying 1-11...done.\n",
      "Finished.\n",
      "PANCAN\n",
      "querying 1-200...done.\n",
      "Finished.\n",
      "PCPG\n",
      "querying 1-5...done.\n",
      "Finished.\n",
      "PRAD\n",
      "querying 1-15...done.\n",
      "Finished.\n",
      "SARC\n",
      "querying 1-3...done.\n",
      "Finished.\n",
      "SKCM\n",
      "querying 1-24...done.\n",
      "Finished.\n",
      "STAD\n",
      "querying 1-24...done.\n",
      "Finished.\n",
      "TGCT\n",
      "querying 1-5...done.\n",
      "Finished.\n",
      "THCA\n",
      "querying 1-9...done.\n",
      "Finished.\n",
      "THYM\n",
      "querying 1-4...done.\n",
      "Finished.\n",
      "UCEC\n",
      "querying 1-55...done.\n",
      "Finished.\n",
      "UCS\n",
      "querying 1-14...done.\n",
      "Finished.\n",
      "UVM\n",
      "querying 1-8...done.\n",
      "Finished.\n"
     ]
    }
   ],
   "source": [
    "for disease in sorted(set(meta['Cancer'])):\n",
    "    print(disease)\n",
    "    genes = meta[meta['Cancer']==disease]['Gene'].to_list()\n",
    "    genes = [m.get(g, g) for g in genes]\n",
    "    genes = sorted(set(genes))\n",
    "    \n",
    "    mg = mygene.MyGeneInfo()\n",
    "    result = mg.querymany(genes, scopes='symbol', fields='all', species='human')\n",
    "    \n",
    "    df = result_to_df(result)\n",
    "    df.to_csv(os.path.join(out_dir, 'formatted_beds', f'driver_paper_{disease}.bed'),\n",
    "              index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "345818e0",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Drivers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "cc843296",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(299, 299)"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = pd.read_csv('/diskmnt/Projects/Users/estorrs/sandbox/pecgs_bed_files/gene_lists/cancer_gene_list_cell_uniq.tsv',\n",
    "                   sep='\\t', header=None)\n",
    "genes = genes[0]\n",
    "len(genes), len(set(genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "0b114ba5",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# some genes have incorrect symbol name\n",
    "m = {\n",
    "    'CBWD3': 'ZNG1C',\n",
    "    'FAM46D': 'TENT5D',\n",
    "    'H3F3A': 'H3-3A',\n",
    "    'H3F3C': 'H3-5',\n",
    "    'HIST1H1C': 'H1-2',\n",
    "    'HIST1H1E': 'H1-4',\n",
    "    'RQCD1': 'CNOT9',\n",
    "    'TCEB1': 'ELOC',\n",
    "    'WHSC1': 'NSD2'\n",
    "}\n",
    "genes = [m.get(g, g) for g in genes]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "9d30286c",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-299...done.\n",
      "Finished.\n"
     ]
    }
   ],
   "source": [
    "mg = mygene.MyGeneInfo()\n",
    "result = mg.querymany(genes, scopes='symbol', fields='all', species='human')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "a83be7d9",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6185020</td>\n",
       "      <td>6209389</td>\n",
       "      <td>RPL22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11106535</td>\n",
       "      <td>11262556</td>\n",
       "      <td>MTOR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16124337</td>\n",
       "      <td>16156069</td>\n",
       "      <td>EPHA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26693236</td>\n",
       "      <td>26782104</td>\n",
       "      <td>ARID1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>chr1</td>\n",
       "      <td>36224432</td>\n",
       "      <td>36305357</td>\n",
       "      <td>THRAP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>chrX</td>\n",
       "      <td>118823824</td>\n",
       "      <td>118826968</td>\n",
       "      <td>ZCCHC12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>chrX</td>\n",
       "      <td>123960212</td>\n",
       "      <td>124422664</td>\n",
       "      <td>STAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>chrX</td>\n",
       "      <td>129446501</td>\n",
       "      <td>129523500</td>\n",
       "      <td>SMARCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>chrX</td>\n",
       "      <td>134373288</td>\n",
       "      <td>134428791</td>\n",
       "      <td>PHF6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154348524</td>\n",
       "      <td>154374634</td>\n",
       "      <td>FLNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>299 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr      start       stop     gene\n",
       "234  chr1    6185020    6209389    RPL22\n",
       "168  chr1   11106535   11262556     MTOR\n",
       "85   chr1   16124337   16156069    EPHA2\n",
       "14   chr1   26693236   26782104   ARID1A\n",
       "272  chr1   36224432   36305357   THRAP3\n",
       "..    ...        ...        ...      ...\n",
       "291  chrX  118823824  118826968  ZCCHC12\n",
       "261  chrX  123960212  124422664    STAG2\n",
       "249  chrX  129446501  129523500  SMARCA1\n",
       "192  chrX  134373288  134428791     PHF6\n",
       "99   chrX  154348524  154374634     FLNA\n",
       "\n",
       "[299 rows x 4 columns]"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = result_to_df(result)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "c57ee899",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "df.to_csv(os.path.join(out_dir, 'formatted_beds', '299_drivers.bed'), index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa276b7f",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
