{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9d116b7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import yaml\n",
    "import json\n",
    "import pandas as pd\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f42a991a",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl_dir = '/home/estorrs/pecgs-pipeline/cwl/align_dnaseq'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63205254",
   "metadata": {},
   "source": [
    "looking at another template for reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6c51b278",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'bam_name': 'desired-output-name.bam',\n",
       " 'job_uuid': 'example-test-uuid',\n",
       " 'collect_wgs_metrics': True,\n",
       " 'amplicon_kit_set_file_list': [],\n",
       " 'capture_kit_set_file_list': [],\n",
       " 'readgroup_fastq_pe_file_list': [{'forward_fastq': {'class': 'File',\n",
       "    'location': 'input_R1.fastq.gz'},\n",
       "   'reverse_fastq': {'class': 'File', 'location': 'input_R2.fastq.gz'},\n",
       "   'readgroup_meta': {'ID': 'input_id',\n",
       "    'PL': 'ILLUMINA',\n",
       "    'SM': 'input_sample_id'}}],\n",
       " 'readgroup_fastq_se_file_list': [],\n",
       " 'readgroups_bam_file_list': [],\n",
       " 'common_biallelic_vcf': {'class': 'File',\n",
       "  'location': 'somatic-hg38_small_exac_common_3.hg38.vcf.gz',\n",
       "  'secondaryFiles': [{'class': 'File',\n",
       "    'location': 'somatic-hg38_small_exac_common_3.hg38.vcf.gz.tbi'}]},\n",
       " 'known_snp': {'class': 'File',\n",
       "  'location': 'dbsnp_144.hg38.vcf.gz',\n",
       "  'secondaryFiles': [{'class': 'File',\n",
       "    'location': 'dbsnp_144.hg38.vcf.gz.tbi'}]},\n",
       " 'run_markduplicates': True,\n",
       " 'reference_sequence': {'class': 'File',\n",
       "  'location': 'GRCh38.d1.vd1.fa',\n",
       "  'secondaryFiles': [{'class': 'File', 'location': 'GRCh38.d1.vd1.fa.amb'},\n",
       "   {'class': 'File', 'location': 'GRCh38.d1.vd1.fa.ann'},\n",
       "   {'class': 'File', 'location': 'GRCh38.d1.vd1.fa.bwt'},\n",
       "   {'class': 'File', 'location': 'GRCh38.d1.vd1.fa.fai'},\n",
       "   {'class': 'File', 'location': 'GRCh38.d1.vd1.fa.pac'},\n",
       "   {'class': 'File', 'location': 'GRCh38.d1.vd1.fa.sa'},\n",
       "   {'class': 'File', 'location': 'GRCh38.d1.vd1.dict'}]},\n",
       " 'thread_count': 8}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = yaml.safe_load(open('/home/estorrs/pecgs-pipeline/submodules/gdc-dnaseq-cwl/example/gdc_dnaseq_main_workflow_example_input.yaml'))\n",
    "t"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0303f60",
   "metadata": {},
   "source": [
    "test invocation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "fc44386a",
   "metadata": {},
   "outputs": [],
   "source": [
    "template = {\n",
    "    'bam_name': 'a_string',\n",
    "    'job_uuid': 'a_string',\n",
    "    'thread_count': 'a_int',\n",
    "    'known_snp': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/vcf.gz',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/vcf.gz.tbi'}\n",
    "        ]\n",
    "    },\n",
    "    'reference_sequence': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    'readgroup_fastq_pe_file_list': [\n",
    "        {\n",
    "            'forward_fastq': {\n",
    "                'class': 'File',\n",
    "                'path': 'path/to/R1/fastq'\n",
    "            },\n",
    "            'reverse_fastq': {\n",
    "                'class': 'File',\n",
    "                'path': 'path/to/R2/fastq'\n",
    "            },\n",
    "            'readgroup_meta': {\n",
    "                'ID': 'input_id',\n",
    "                'PL': 'ILLUMINA',\n",
    "                'SM': 'input_sample_id'\n",
    "            }\n",
    "        }\n",
    "    ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "03c4deeb",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.align_dnaseq_workflow.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d782150",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4358590e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "bf8e38fc",
   "metadata": {},
   "source": [
    "make cwl file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fb128d7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'cwlVersion': 'v1.0',\n",
       " 'class': 'Workflow',\n",
       " 'id': 'gdc_dnaseq_main_wf',\n",
       " 'requirements': [{'class': 'InlineJavascriptRequirement'},\n",
       "  {'class': 'MultipleInputFeatureRequirement'},\n",
       "  {'class': 'ScatterFeatureRequirement'},\n",
       "  {'class': 'SchemaDefRequirement',\n",
       "   'types': [{'$import': '../../tools/target_kit_schema.yml'},\n",
       "    {'$import': '../../tools/readgroup.yml'}]},\n",
       "  {'class': 'StepInputExpressionRequirement'},\n",
       "  {'class': 'SubworkflowFeatureRequirement'}],\n",
       " 'inputs': {'bam_name': 'string',\n",
       "  'job_uuid': 'string',\n",
       "  'collect_wgs_metrics': 'boolean',\n",
       "  'amplicon_kit_set_file_list': {'type': {'type': 'array',\n",
       "    'items': '../../tools/target_kit_schema.yml#amplicon_kit_set_file'}},\n",
       "  'capture_kit_set_file_list': {'type': {'type': 'array',\n",
       "    'items': '../../tools/target_kit_schema.yml#capture_kit_set_file'}},\n",
       "  'readgroup_fastq_pe_file_list': {'type': {'type': 'array',\n",
       "    'items': '../../tools/readgroup.yml#readgroup_fastq_file'}},\n",
       "  'readgroup_fastq_se_file_list': {'type': {'type': 'array',\n",
       "    'items': '../../tools/readgroup.yml#readgroup_fastq_file'}},\n",
       "  'readgroups_bam_file_list': {'type': {'type': 'array',\n",
       "    'items': '../../tools/readgroup.yml#readgroups_bam_file'}},\n",
       "  'common_biallelic_vcf': {'type': 'File', 'secondaryFiles': ['.tbi']},\n",
       "  'known_snp': {'type': 'File', 'secondaryFiles': ['.tbi']},\n",
       "  'run_markduplicates': 'boolean',\n",
       "  'reference_sequence': {'type': 'File',\n",
       "   'secondaryFiles': ['.amb',\n",
       "    '.ann',\n",
       "    '.bwt',\n",
       "    '.fai',\n",
       "    '.pac',\n",
       "    '.sa',\n",
       "    '^.dict']},\n",
       "  'thread_count': 'long'},\n",
       " 'outputs': {'output_bam': {'type': 'File',\n",
       "   'outputSource': 'gatk_applybqsr/output_bam'},\n",
       "  'sqlite': {'type': 'File',\n",
       "   'outputSource': 'merge_all_sqlite/destination_sqlite'}},\n",
       " 'steps': {'fastq_clean_pe': {'run': '../utils/fastq_clean.cwl',\n",
       "   'scatter': 'input',\n",
       "   'in': {'input': 'readgroup_fastq_pe_file_list', 'job_uuid': 'job_uuid'},\n",
       "   'out': ['output', 'sqlite']},\n",
       "  'fastq_clean_se': {'run': '../utils/fastq_clean.cwl',\n",
       "   'scatter': 'input',\n",
       "   'in': {'input': 'readgroup_fastq_se_file_list', 'job_uuid': 'job_uuid'},\n",
       "   'out': ['output', 'sqlite']},\n",
       "  'merge_sqlite_fastq_clean_pe': {'run': '../../tools/merge_sqlite.cwl',\n",
       "   'in': {'source_sqlite': 'fastq_clean_pe/sqlite', 'job_uuid': 'job_uuid'},\n",
       "   'out': ['destination_sqlite', 'log']},\n",
       "  'merge_sqlite_fastq_clean_se': {'run': '../../tools/merge_sqlite.cwl',\n",
       "   'in': {'source_sqlite': 'fastq_clean_se/sqlite', 'job_uuid': 'job_uuid'},\n",
       "   'out': ['destination_sqlite', 'log']},\n",
       "  'readgroups_bam_to_readgroups_fastq_lists': {'run': '../utils/readgroups_bam_to_readgroups_fastq_lists.cwl',\n",
       "   'scatter': 'readgroups_bam_file',\n",
       "   'in': {'readgroups_bam_file': 'readgroups_bam_file_list'},\n",
       "   'out': ['pe_file_list', 'se_file_list', 'o1_file_list', 'o2_file_list']},\n",
       "  'merge_fastq_arrays': {'run': '../utils/merge_fastq_array_workflow.cwl',\n",
       "   'in': {'bam_pe_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/pe_file_list',\n",
       "    'bam_se_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/se_file_list',\n",
       "    'bam_o1_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/o1_file_list',\n",
       "    'bam_o2_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/o2_file_list',\n",
       "    'fastqs_pe': 'fastq_clean_pe/output',\n",
       "    'fastqs_se': 'fastq_clean_se/output'},\n",
       "   'out': ['merged_pe_fastq_array', 'merged_se_fastq_array']},\n",
       "  'bwa_pe': {'run': '../align/bwa_pe.cwl',\n",
       "   'scatter': 'readgroup_fastq_pe',\n",
       "   'in': {'job_uuid': 'job_uuid',\n",
       "    'reference_sequence': 'reference_sequence',\n",
       "    'readgroup_fastq_pe': 'merge_fastq_arrays/merged_pe_fastq_array',\n",
       "    'thread_count': 'thread_count'},\n",
       "   'out': ['bam', 'sqlite']},\n",
       "  'bwa_se': {'run': '../align/bwa_se.cwl',\n",
       "   'scatter': 'readgroup_fastq_se',\n",
       "   'in': {'job_uuid': 'job_uuid',\n",
       "    'reference_sequence': 'reference_sequence',\n",
       "    'readgroup_fastq_se': 'merge_fastq_arrays/merged_se_fastq_array',\n",
       "    'thread_count': 'thread_count'},\n",
       "   'out': ['bam', 'sqlite']},\n",
       "  'merge_sqlite_bwa_pe': {'run': '../../tools/merge_sqlite.cwl',\n",
       "   'in': {'source_sqlite': 'bwa_pe/sqlite', 'job_uuid': 'job_uuid'},\n",
       "   'out': ['destination_sqlite', 'log']},\n",
       "  'merge_sqlite_bwa_se': {'run': '../../tools/merge_sqlite.cwl',\n",
       "   'in': {'source_sqlite': 'bwa_se/sqlite', 'job_uuid': 'job_uuid'},\n",
       "   'out': ['destination_sqlite', 'log']},\n",
       "  'dup_branch_decider': {'run': '../../tools/decider_markdup_input.cwl',\n",
       "   'in': {'run_markdups': 'run_markduplicates',\n",
       "    'bam': {'source': ['bwa_pe/bam', 'bwa_se/bam']}},\n",
       "   'out': ['do_markdup_workflow', 'skip_markdup_workflow', 'out_bam']},\n",
       "  'conditional_markduplicates': {'run': '../utils/conditional_markduplicates.cwl',\n",
       "   'scatter': 'run_markduplicates',\n",
       "   'in': {'bam': 'dup_branch_decider/out_bam',\n",
       "    'job_uuid': 'job_uuid',\n",
       "    'thread_count': 'thread_count',\n",
       "    'bam_name': 'bam_name',\n",
       "    'run_markduplicates': 'dup_branch_decider/do_markdup_workflow'},\n",
       "   'out': ['output', 'sqlite']},\n",
       "  'conditional_skip_markduplicates': {'run': '../utils/conditional_skip_markduplicates.cwl',\n",
       "   'scatter': 'skip_markduplicates',\n",
       "   'in': {'bam': 'dup_branch_decider/out_bam',\n",
       "    'job_uuid': 'job_uuid',\n",
       "    'thread_count': 'thread_count',\n",
       "    'bam_name': 'bam_name',\n",
       "    'skip_markduplicates': 'dup_branch_decider/skip_markdup_workflow'},\n",
       "   'out': ['output']},\n",
       "  'dup_outputs_decider': {'run': '../../tools/decider_markdup_output.cwl',\n",
       "   'in': {'markdup_bam': 'conditional_markduplicates/output',\n",
       "    'markdup_sqlite': 'conditional_markduplicates/sqlite',\n",
       "    'skip_markdup_bam': 'conditional_skip_markduplicates/output'},\n",
       "   'out': ['bam', 'sqlite']},\n",
       "  'gatk_baserecalibrator': {'run': '../../tools/gatk4_baserecalibrator.cwl',\n",
       "   'in': {'input': 'dup_outputs_decider/bam',\n",
       "    'known-sites': 'known_snp',\n",
       "    'reference': 'reference_sequence'},\n",
       "   'out': ['output_grp']},\n",
       "  'gatk_applybqsr': {'run': '../../tools/gatk4_applybqsr.cwl',\n",
       "   'in': {'input': 'dup_outputs_decider/bam',\n",
       "    'bqsr-recal-file': 'gatk_baserecalibrator/output_grp'},\n",
       "   'out': ['output_bam']},\n",
       "  'picard_validatesamfile_bqsr': {'run': '../../tools/picard_validatesamfile.cwl',\n",
       "   'in': {'INPUT': 'gatk_applybqsr/output_bam',\n",
       "    'VALIDATION_STRINGENCY': {'valueFrom': 'STRICT'}},\n",
       "   'out': ['OUTPUT']},\n",
       "  'picard_validatesamfile_bqsr_to_sqlite': {'run': '../../tools/picard_validatesamfile_to_sqlite.cwl',\n",
       "   'in': {'bam': {'source': 'gatk_applybqsr/output_bam',\n",
       "     'valueFrom': '$(self.basename)'},\n",
       "    'input_state': {'valueFrom': 'gatk_applybqsr_readgroups'},\n",
       "    'metric_path': 'picard_validatesamfile_bqsr/OUTPUT',\n",
       "    'job_uuid': 'job_uuid'},\n",
       "   'out': ['sqlite']},\n",
       "  'metrics': {'run': '../metrics/metrics.cwl',\n",
       "   'in': {'bam': 'gatk_applybqsr/output_bam',\n",
       "    'amplicon_kit_set_file_list': 'amplicon_kit_set_file_list',\n",
       "    'capture_kit_set_file_list': 'capture_kit_set_file_list',\n",
       "    'collect_wgs_metrics': {'source': 'collect_wgs_metrics',\n",
       "     'valueFrom': '${\\n   if (self) {\\n     return([1]);\\n   } else {\\n     return([]);\\n   }\\n }\\n'},\n",
       "    'common_biallelic_vcf': 'common_biallelic_vcf',\n",
       "    'fasta': 'reference_sequence',\n",
       "    'input_state': {'valueFrom': 'gatk_applybqsr_readgroups'},\n",
       "    'job_uuid': 'job_uuid',\n",
       "    'known_snp': 'known_snp',\n",
       "    'thread_count': 'thread_count'},\n",
       "   'out': ['sqlite']},\n",
       "  'merge_all_sqlite': {'run': '../../tools/merge_sqlite.cwl',\n",
       "   'in': {'source_sqlite': {'source': ['merge_sqlite_fastq_clean_pe/destination_sqlite',\n",
       "      'merge_sqlite_fastq_clean_se/destination_sqlite',\n",
       "      'merge_sqlite_bwa_pe/destination_sqlite',\n",
       "      'merge_sqlite_bwa_se/destination_sqlite',\n",
       "      'dup_outputs_decider/sqlite',\n",
       "      'picard_validatesamfile_bqsr_to_sqlite/sqlite',\n",
       "      'metrics/sqlite'],\n",
       "     'valueFrom': '${\\n   var res = [];\\n   for (var i = 0; i<self.length; i++) {\\n     if(self[i] !== null) {\\n       res.push(self[i])\\n     }\\n   }\\n   return(res);\\n }\\n'},\n",
       "    'job_uuid': 'job_uuid'},\n",
       "   'out': ['destination_sqlite', 'log']}}}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = yaml.safe_load(open('/home/estorrs/pecgs-pipeline/submodules/gdc-dnaseq-cwl/subworkflows/main/gdc_dnaseq_main_workflow.cwl'))\n",
    "t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "192c8672",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'align_dnaseq',\n",
    "    'label': 'align_dnaseq',\n",
    "    'inputs': {\n",
    "        'bam_name': 'string',\n",
    "        'job_uuid': 'string',\n",
    "        'readgroup_fastq_pe_file_list': {\n",
    "            'type': {\n",
    "                'type': 'array',\n",
    "                'items': '../../submodules/gdc-dnaseq-cwl/tools/readgroup.yml#readgroup_fastq_file'\n",
    "            }\n",
    "        },\n",
    "       'known_snp': {\n",
    "           'type': 'File',\n",
    "           'secondaryFiles': ['.tbi']\n",
    "       },\n",
    "       'run_markduplicates': {\n",
    "           'type': 'boolean?',\n",
    "           'default': True\n",
    "       },\n",
    "       'reference_sequence': {\n",
    "           'type': 'File',\n",
    "           'secondaryFiles': [\n",
    "               '.amb',\n",
    "               '.ann',\n",
    "               '.bwt',\n",
    "               '.fai',\n",
    "               '.pac',\n",
    "               '.sa',\n",
    "               '^.dict'\n",
    "           ]\n",
    "       },\n",
    "       'thread_count': 'long'\n",
    "    },\n",
    "    'outputs': {\n",
    "        'output_bam': {\n",
    "            'type': 'File',\n",
    "            'outputSource': 'gatk_applybqsr/output_bam'\n",
    "        }\n",
    "    },\n",
    "    'steps': {\n",
    "        'fastq_clean_pe': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/utils/fastq_clean.cwl',\n",
    "            'scatter': 'input',\n",
    "            'in': {'input': 'readgroup_fastq_pe_file_list', 'job_uuid': 'job_uuid'},\n",
    "            'out': ['output']\n",
    "        },\n",
    "        'fastq_clean_se': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/utils/fastq_clean.cwl',\n",
    "            'scatter': 'input',\n",
    "            'in': {'input': 'readgroup_fastq_se_file_list', 'job_uuid': 'job_uuid'},\n",
    "            'out': ['output']\n",
    "        },\n",
    "        'readgroups_bam_to_readgroups_fastq_lists': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/utils/readgroups_bam_to_readgroups_fastq_lists.cwl',\n",
    "            'scatter': 'readgroups_bam_file',\n",
    "            'in': {'readgroups_bam_file': 'readgroups_bam_file_list'},\n",
    "            'out': ['pe_file_list', 'se_file_list', 'o1_file_list', 'o2_file_list']\n",
    "        },\n",
    "        'merge_fastq_arrays': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/utils/merge_fastq_array_workflow.cwl',\n",
    "            'in': {\n",
    "                'bam_pe_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/pe_file_list',\n",
    "                'bam_se_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/se_file_list',\n",
    "                'bam_o1_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/o1_file_list',\n",
    "                'bam_o2_fastqs': 'readgroups_bam_to_readgroups_fastq_lists/o2_file_list',\n",
    "                'fastqs_pe': 'fastq_clean_pe/output',\n",
    "                'fastqs_se': 'fastq_clean_se/output'\n",
    "            },\n",
    "            'out': ['merged_pe_fastq_array', 'merged_se_fastq_array']\n",
    "        },\n",
    "        'bwa_pe': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/align/bwa_pe.cwl',\n",
    "            'scatter': 'readgroup_fastq_pe',\n",
    "            'in': {\n",
    "                'job_uuid': 'job_uuid',\n",
    "                'reference_sequence': 'reference_sequence',\n",
    "                'readgroup_fastq_pe': 'merge_fastq_arrays/merged_pe_fastq_array',\n",
    "                'thread_count': 'thread_count'\n",
    "            },\n",
    "            'out': ['bam']\n",
    "        },\n",
    "        'bwa_se': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/align/bwa_se.cwl',\n",
    "            'scatter': 'readgroup_fastq_se',\n",
    "            'in': {\n",
    "                'job_uuid': 'job_uuid',\n",
    "                'reference_sequence': 'reference_sequence',\n",
    "                'readgroup_fastq_se': 'merge_fastq_arrays/merged_se_fastq_array',\n",
    "                'thread_count': 'thread_count'\n",
    "            },\n",
    "            'out': ['bam']\n",
    "        },\n",
    "        'dup_branch_decider': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/tools/decider_markdup_input.cwl',\n",
    "            'in': {\n",
    "                'run_markdups': 'run_markduplicates',\n",
    "                'bam': {'source': ['bwa_pe/bam', 'bwa_se/bam']}\n",
    "            },\n",
    "            'out': ['do_markdup_workflow', 'skip_markdup_workflow', 'out_bam']\n",
    "        },\n",
    "        'conditional_markduplicates': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/utils/conditional_markduplicates.cwl',\n",
    "            'scatter': 'run_markduplicates',\n",
    "            'in': {\n",
    "                'bam': 'dup_branch_decider/out_bam',\n",
    "                'job_uuid': 'job_uuid',\n",
    "                'thread_count': 'thread_count',\n",
    "                'bam_name': 'bam_name',\n",
    "                'run_markduplicates': 'dup_branch_decider/do_markdup_workflow'\n",
    "            },\n",
    "            'out': ['output', 'sqlite']\n",
    "        },\n",
    "        'conditional_skip_markduplicates': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/subworkflows/utils/conditional_skip_markduplicates.cwl',\n",
    "            'scatter': 'skip_markduplicates',\n",
    "            'in': {\n",
    "                'bam': 'dup_branch_decider/out_bam',\n",
    "                'job_uuid': 'job_uuid',\n",
    "                'thread_count': 'thread_count',\n",
    "                'bam_name': 'bam_name',\n",
    "                'skip_markduplicates': 'dup_branch_decider/skip_markdup_workflow'\n",
    "            },\n",
    "            'out': ['output']\n",
    "        },\n",
    "        'dup_outputs_decider': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/tools/decider_markdup_output.cwl',\n",
    "            'in': {\n",
    "                'markdup_bam': 'conditional_markduplicates/output',\n",
    "                'markdup_sqlite': 'conditional_markduplicates/sqlite',\n",
    "                'skip_markdup_bam': 'conditional_skip_markduplicates/output'\n",
    "            },\n",
    "            'out': ['bam']\n",
    "        },\n",
    "        'gatk_baserecalibrator': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/tools/gatk4_baserecalibrator.cwl',\n",
    "            'in': {\n",
    "                'input': 'dup_outputs_decider/bam',\n",
    "                'known-sites': 'known_snp',\n",
    "                'reference': 'reference_sequence'\n",
    "            },\n",
    "            'out': ['output_grp']\n",
    "        },\n",
    "        'gatk_applybqsr': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/tools/gatk4_applybqsr.cwl',\n",
    "            'in': {'input': 'dup_outputs_decider/bam', 'bqsr-recal-file': 'gatk_baserecalibrator/output_grp'},\n",
    "            'out': ['output_bam']\n",
    "        },\n",
    "        'picard_validatesamfile_bqsr': {\n",
    "            'run': '../../submodules/gdc-dnaseq-cwl/tools/picard_validatesamfile.cwl',\n",
    "            'in': {'INPUT': 'gatk_applybqsr/output_bam', 'VALIDATION_STRINGENCY': {'valueFrom': 'STRICT'}},\n",
    "            'out': ['OUTPUT']\n",
    "        },\n",
    "    },\n",
    "   'requirements': [\n",
    "       {'class': 'InlineJavascriptRequirement'},\n",
    "       {'class': 'MultipleInputFeatureRequirement'},\n",
    "       {'class': 'ScatterFeatureRequirement'},\n",
    "       {\n",
    "           'class': 'SchemaDefRequirement',\n",
    "           'types': [\n",
    "               {'$import': '../../submodules/gdc-dnaseq-cwl/tools/readgroup.yml'}\n",
    "           ]\n",
    "       },\n",
    "      {'class': 'StepInputExpressionRequirement'},\n",
    "      {'class': 'SubworkflowFeatureRequirement'}\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86656a9e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c34f7ed6",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'align_dnaseq_workflow.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77b64330",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
