{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "f03b205f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import yaml\n",
    "import uuid\n",
    "import os\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "262987e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "2f9d0660",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "faef0fc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "f403b2e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import wombat.pecgs as pecgs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d519673c",
   "metadata": {},
   "source": [
    "#### make pecgs pipeline cwl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "0608748d",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl_dir = '/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "011c36d1",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### pecgs_TN_wxs_fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "6fe734c5",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "template = {\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int',\n",
    "    'tumor_wxs_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_wxs_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_fq_1': {\n",
    "        'class': 'File', \n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },  \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    \n",
    "    ## align dna-seq\n",
    "    'known_sites': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/vcf.gz',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/vcf.gz.tbi'}\n",
    "        ]\n",
    "    },\n",
    "    'wxs_normal_flowcell': 'a_string',\n",
    "    'wxs_normal_lane': 'a_string',\n",
    "    'wxs_normal_index_sequencer': 'a_string',\n",
    "    'wxs_normal_library_preparation': 'a_string',\n",
    "    'wxs_normal_platform': 'a_string',\n",
    "    'wxs_tumor_flowcell': 'a_string',\n",
    "    'wxs_tumor_lane': 'a_string',\n",
    "    'wxs_tumor_index_sequencer': 'a_string',\n",
    "    'wxs_tumor_library_preparation': 'a_string',\n",
    "    'wxs_tumor_platform': 'a_string',\n",
    "    \n",
    "    # CNV\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 'a_int',\n",
    "    'minimal_microsatellite_size': 'a_int',\n",
    "    \n",
    "    # TinDaisy\n",
    "    'rescue_clinvar': False,\n",
    "    'rescue_cosmic': False,\n",
    "    'vep_cache_version': 'a_string',\n",
    "    'vep_cache_gz': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'clinvar_annotation': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'classification_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'af_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'call_regions': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'normal_barcode': 'a_string',\n",
    "    'tumor_barcode': 'a_string',\n",
    "    'tindaisy_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'strelka_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'centromere_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'assembly': 'a_string',\n",
    "    'varscan_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    \n",
    "    # TinJasmine\n",
    "    'sample_barcode': 'a_string',\n",
    "    'centromere': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'tinjasmine_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'Canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'ROI_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'varscan_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config_template': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'gatk_filter_config': {'class': 'File', 'path': 'a/file/path'}, \n",
    "    \n",
    "    # neoscan\n",
    "    'neoscan_ref_dir': {'class': 'Directory', 'path': 'a/dir/path'},\n",
    "    'neoscan_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "d27d6418",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_TN_wxs_fq.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "638ce397",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_TN_wxs_fq',\n",
    "    'label': 'pecgs_TN_wxs_fq',\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'cpu',\n",
    "            'type': 'int?',\n",
    "            'default': 40\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # align dna-seq\n",
    "        {\n",
    "            'id': 'tumor_sample',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).WXS.T'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_sample',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).WXS.N'\n",
    "        },\n",
    "        {\n",
    "            'id': 'known_sites',\n",
    "            'type': 'File',\n",
    "           'secondaryFiles': ['.tbi']\n",
    "        },\n",
    "        {\n",
    "            'id': 'reference',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '.amb',\n",
    "               '.ann',\n",
    "               '.bwt',\n",
    "               '.fai',\n",
    "               '.pac',\n",
    "               '.sa',\n",
    "               '^.dict'\n",
    "           ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_flowcell',\n",
    "            'type': 'string?',\n",
    "            'default': 'flowcellABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_lane',\n",
    "            'type': 'string?',\n",
    "            'default': '1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_index_sequencer',\n",
    "            'type': 'string?',\n",
    "            'default': 'sequencerABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_library_preparation',\n",
    "            'type': 'string?',\n",
    "            'default': 'lib1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_platform',\n",
    "            'type': 'string?',\n",
    "            'default': 'ILLUMINA'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_flowcell',\n",
    "            'type': 'string?',\n",
    "            'default': 'flowcellABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_lane',\n",
    "            'type': 'string?',\n",
    "            'default': '1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_index_sequencer',\n",
    "            'type': 'string?',\n",
    "            'default': 'sequencerABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_library_preparation',\n",
    "            'type': 'string?',\n",
    "            'default': 'lib1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_platform',\n",
    "            'type': 'string?',\n",
    "            'default': 'ILLUMINA'\n",
    "        },\n",
    "        \n",
    "        # CNV\n",
    "        {\n",
    "            'id': 'reference_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        \n",
    "        # msisensor\n",
    "        {\n",
    "            'id': 'microsatellite',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_homopolymer_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_microsatellite_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        \n",
    "        # TinDaisy\n",
    "        {\n",
    "            'id': 'rescue_clinvar',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'rescue_cosmic',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_version',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_gz',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'clinvar_annotation',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'classification_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'af_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'call_regions',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).N',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).T',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'strelka_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'assembly',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # TinJasmine\n",
    "        {\n",
    "            'id': 'sample_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample)'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'Canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'ROI_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config_template',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gatk_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # neoscan\n",
    "        {\n",
    "            'id': 'neoscan_ref_dir',\n",
    "            'type': 'Directory'\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_input_type',\n",
    "            'type': 'string?',\n",
    "            'default': 'dna'\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'tumor_wxs_output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'align_tumor_wxs/output_bam',\n",
    "            'secondaryFiles': ['^.bai']\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'align_normal_wxs/output_bam',\n",
    "            'secondaryFiles': ['^.bai']\n",
    "        },\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_cnv/gene_level_cnv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_summary'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_dis',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_dis'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_germline',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_germline'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_somatic',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_somatic'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_maf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_all'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_MAF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/allCall_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'charger_output_tsv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_charger/charger_tsv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_snv_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_neoscan/snv_summary',\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_indel_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_neoscan/indel_summary',\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'align_tumor_wxs',\n",
    "            'label': 'align_tumor_wxs',\n",
    "            'run': '../../submodules/align-dnaseq/cwl/align_dnaseq.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'tumor_sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_wxs_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_wxs_fq_2'},\n",
    "                {'id': 'known_sites', 'source': 'known_sites'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'flowcell', 'source': 'wxs_tumor_flowcell'},\n",
    "                {'id': 'lane', 'source': 'wxs_tumor_lane'},\n",
    "                {'id': 'index_sequencer', 'source': 'wxs_tumor_index_sequencer'},\n",
    "                {'id': 'library_preparation', 'source': 'wxs_tumor_library_preparation'},\n",
    "                {'id': 'platform', 'source': 'wxs_tumor_platform'},  \n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'align_normal_wxs',\n",
    "            'label': 'align_normal_wxs',\n",
    "            'run': '../../submodules/align-dnaseq/cwl/align_dnaseq.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'normal_sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'normal_wxs_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'normal_wxs_fq_2'},\n",
    "                {'id': 'known_sites', 'source': 'known_sites'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'flowcell', 'source': 'wxs_normal_flowcell'},\n",
    "                {'id': 'lane', 'source': 'wxs_normal_lane'},\n",
    "                {'id': 'index_sequencer', 'source': 'wxs_normal_index_sequencer'},\n",
    "                {'id': 'library_preparation', 'source': 'wxs_normal_library_preparation'},\n",
    "                {'id': 'platform', 'source': 'wxs_normal_platform'},  \n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_cnv',\n",
    "            'label': 'run_cnv',\n",
    "            'run': '../../submodules/pecgs-cnv/cwl/cnv_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference_dir', 'source': 'reference_dir'},\n",
    "                {'id': 'target_interval_list', 'source': 'target_interval_list'},\n",
    "                {'id': 'common_biallelic', 'source': 'common_biallelic'},\n",
    "                {'id': 'protein_coding_gene', 'source': 'protein_coding_gene'},\n",
    "                {'id': 'pool_of_normals', 'source': 'pool_of_normals'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'gene_level_cnv'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_msisensor',\n",
    "            'label': 'run_msisensor',\n",
    "            'run': '../msisensor/msisensor_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'threads', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'microsatellite', 'source': 'microsatellite'},\n",
    "                {'id': 'minimal_homopolymer_size', 'source': 'minimal_homopolymer_size'},\n",
    "                {'id': 'minimal_microsatellite_size', 'source': 'minimal_microsatellite_size'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_summary'},\n",
    "                {'id': 'output_dis'},\n",
    "                {'id': 'output_germline'},\n",
    "                {'id': 'output_somatic'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tindaisy',\n",
    "            'label': 'run_tindaisy',\n",
    "            'run': '../../submodules/TinDaisy/cwl/workflows/tindaisy2.cwl',\n",
    "            'in': [\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference_fasta', 'source': 'reference'},\n",
    "                {'id': 'pindel_config', 'source': 'pindel_config'},\n",
    "                {'id': 'varscan_config', 'source': 'varscan_config'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'centromere_bed', 'source': 'centromere_bed'},\n",
    "                {'id': 'strelka_config', 'source': 'strelka_config'},\n",
    "                {'id': 'chrlist', 'source': 'tindaisy_chrlist'},\n",
    "                {'id': 'tumor_barcode', 'source': 'tumor_barcode'},\n",
    "                {'id': 'normal_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'canonical_BED', 'source': 'canonical_BED'},\n",
    "                {'id': 'call_regions', 'source': 'call_regions'},\n",
    "                {'id': 'af_config', 'source': 'af_config'},\n",
    "                {'id': 'classification_config', 'source': 'classification_config'},\n",
    "                {'id': 'clinvar_annotation', 'source': 'clinvar_annotation'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'rescue_cosmic', 'source': 'rescue_cosmic'},\n",
    "                {'id': 'rescue_clinvar', 'source': 'rescue_clinvar'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_maf_clean'},\n",
    "                {'id': 'output_vcf_clean'},\n",
    "                {'id': 'output_vcf_all'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tinjasmine',\n",
    "            'label': 'run_tinjasmine',\n",
    "            'run': '../../submodules/TinJasmine/cwl/TinJasmine.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'gatk_filter_config', 'source': 'gatk_filter_config'},\n",
    "                {'id': 'pindel_config_template', 'source': 'pindel_config_template'},\n",
    "                {'id': 'pindel_filter_config', 'source': 'pindel_filter_config'},\n",
    "                {'id': 'varscan_filter_config', 'source': 'varscan_filter_config'},\n",
    "                {'id': 'ROI_BED', 'source': 'ROI_BED'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'Canonical_BED', 'source': 'Canonical_BED'},\n",
    "                {'id': 'chrlist', 'source': 'tinjasmine_chrlist'},\n",
    "                {'id': 'centromere', 'source': 'centromere'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'clean_VCF'},\n",
    "                {'id': 'allCall_VCF'},\n",
    "                {'id': 'clean_MAF'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_charger',\n",
    "            'label': 'run_charger',\n",
    "            'run': '../charger/charger.cwl',\n",
    "            'in': [\n",
    "                {'id': 'vcf', 'source': 'run_tinjasmine/clean_VCF'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'charger_tsv'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_neoscan',\n",
    "            'label': 'run_neoscan',\n",
    "            'run': '../../submodules/pecgs-neoscan/cwl/neoscan.cwl',\n",
    "            'in': [\n",
    "                {'id': 'maf', 'source': 'run_tindaisy/output_maf_clean'},\n",
    "                {'id': 'bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'ref_dir', 'source': 'neoscan_ref_dir'},\n",
    "                {'id': 'bed', 'source': 'neoscan_bed'},\n",
    "                {'id': 'input_type', 'source': 'neoscan_input_type'}\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'snv_summary'},\n",
    "                {'id': 'indel_summary'},\n",
    "            ]\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "   'requirements': [\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "7a0b0826",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_TN_wxs_fq.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31d91738",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### pecgs_TN_wxs_bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "e7180047",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "template = {\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int',\n",
    "    'tumor_wxs_bam': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_bam': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    }, \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    \n",
    "    # CNV\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 'a_int',\n",
    "    'minimal_microsatellite_size': 'a_int',\n",
    "    \n",
    "    # TinDaisy\n",
    "    'rescue_clinvar': False,\n",
    "    'rescue_cosmic': False,\n",
    "    'vep_cache_version': 'a_string',\n",
    "    'vep_cache_gz': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'clinvar_annotation': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'classification_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'af_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'call_regions': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'normal_barcode': 'a_string',\n",
    "    'tumor_barcode': 'a_string',\n",
    "    'tindaisy_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'strelka_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'centromere_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'assembly': 'a_string',\n",
    "    'varscan_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    \n",
    "    # TinJasmine\n",
    "    'sample_barcode': 'a_string',\n",
    "    'centromere': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'tinjasmine_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'Canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'ROI_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'varscan_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config_template': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'gatk_filter_config': {'class': 'File', 'path': 'a/file/path'}, \n",
    "    \n",
    "    # neoscan\n",
    "    'neoscan_ref_dir': {'class': 'Directory', 'path': 'a/dir/path'},\n",
    "    'neoscan_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "2291dda9",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_TN_wxs_bam.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "ec4f7b55",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_TN_wxs_bam',\n",
    "    'label': 'pecgs_TN_wxs_bam',\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'cpu',\n",
    "            'type': 'int?',\n",
    "            'default': 40\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_bam',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_bam',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        {\n",
    "            'id': 'reference',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '.amb',\n",
    "               '.ann',\n",
    "               '.bwt',\n",
    "               '.fai',\n",
    "               '.pac',\n",
    "               '.sa',\n",
    "               '^.dict'\n",
    "           ]\n",
    "        },\n",
    "        \n",
    "        # CNV\n",
    "        {\n",
    "            'id': 'reference_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        \n",
    "        # msisensor\n",
    "        {\n",
    "            'id': 'microsatellite',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_homopolymer_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_microsatellite_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        \n",
    "        # TinDaisy\n",
    "        {\n",
    "            'id': 'rescue_clinvar',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'rescue_cosmic',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_version',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_gz',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'clinvar_annotation',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'classification_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'af_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'call_regions',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).N',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).T',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'strelka_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'assembly',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # TinJasmine\n",
    "        {\n",
    "            'id': 'sample_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample)'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'Canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'ROI_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config_template',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gatk_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # neoscan\n",
    "        {\n",
    "            'id': 'neoscan_ref_dir',\n",
    "            'type': 'Directory'\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_input_type',\n",
    "            'type': 'string?',\n",
    "            'default': 'dna'\n",
    "        },\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_cnv/gene_level_cnv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_summary'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_dis',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_dis'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_germline',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_germline'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_somatic',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_somatic'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_maf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_all'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_MAF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/allCall_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'charger_output_tsv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_charger/charger_tsv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_snv_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_neoscan/snv_summary',\n",
    "        },\n",
    "        {\n",
    "            'id': 'neoscan_indel_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_neoscan/indel_summary',\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'run_cnv',\n",
    "            'label': 'run_cnv',\n",
    "            'run': '../../submodules/pecgs-cnv/cwl/cnv_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'reference_dir', 'source': 'reference_dir'},\n",
    "                {'id': 'target_interval_list', 'source': 'target_interval_list'},\n",
    "                {'id': 'common_biallelic', 'source': 'common_biallelic'},\n",
    "                {'id': 'protein_coding_gene', 'source': 'protein_coding_gene'},\n",
    "                {'id': 'pool_of_normals', 'source': 'pool_of_normals'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'gene_level_cnv'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_msisensor',\n",
    "            'label': 'run_msisensor',\n",
    "            'run': '../msisensor/msisensor_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'threads', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'microsatellite', 'source': 'microsatellite'},\n",
    "                {'id': 'minimal_homopolymer_size', 'source': 'minimal_homopolymer_size'},\n",
    "                {'id': 'minimal_microsatellite_size', 'source': 'minimal_microsatellite_size'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_summary'},\n",
    "                {'id': 'output_dis'},\n",
    "                {'id': 'output_germline'},\n",
    "                {'id': 'output_somatic'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tindaisy',\n",
    "            'label': 'run_tindaisy',\n",
    "            'run': '../../submodules/TinDaisy/cwl/workflows/tindaisy2.cwl',\n",
    "            'in': [\n",
    "                {'id': 'tumor_bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'reference_fasta', 'source': 'reference'},\n",
    "                {'id': 'pindel_config', 'source': 'pindel_config'},\n",
    "                {'id': 'varscan_config', 'source': 'varscan_config'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'centromere_bed', 'source': 'centromere_bed'},\n",
    "                {'id': 'strelka_config', 'source': 'strelka_config'},\n",
    "                {'id': 'chrlist', 'source': 'tindaisy_chrlist'},\n",
    "                {'id': 'tumor_barcode', 'source': 'tumor_barcode'},\n",
    "                {'id': 'normal_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'canonical_BED', 'source': 'canonical_BED'},\n",
    "                {'id': 'call_regions', 'source': 'call_regions'},\n",
    "                {'id': 'af_config', 'source': 'af_config'},\n",
    "                {'id': 'classification_config', 'source': 'classification_config'},\n",
    "                {'id': 'clinvar_annotation', 'source': 'clinvar_annotation'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'rescue_cosmic', 'source': 'rescue_cosmic'},\n",
    "                {'id': 'rescue_clinvar', 'source': 'rescue_clinvar'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_maf_clean'},\n",
    "                {'id': 'output_vcf_clean'},\n",
    "                {'id': 'output_vcf_all'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tinjasmine',\n",
    "            'label': 'run_tinjasmine',\n",
    "            'run': '../../submodules/TinJasmine/cwl/TinJasmine.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'gatk_filter_config', 'source': 'gatk_filter_config'},\n",
    "                {'id': 'pindel_config_template', 'source': 'pindel_config_template'},\n",
    "                {'id': 'pindel_filter_config', 'source': 'pindel_filter_config'},\n",
    "                {'id': 'varscan_filter_config', 'source': 'varscan_filter_config'},\n",
    "                {'id': 'ROI_BED', 'source': 'ROI_BED'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'Canonical_BED', 'source': 'Canonical_BED'},\n",
    "                {'id': 'chrlist', 'source': 'tinjasmine_chrlist'},\n",
    "                {'id': 'centromere', 'source': 'centromere'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'clean_VCF'},\n",
    "                {'id': 'allCall_VCF'},\n",
    "                {'id': 'clean_MAF'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_charger',\n",
    "            'label': 'run_charger',\n",
    "            'run': '../charger/charger.cwl',\n",
    "            'in': [\n",
    "                {'id': 'vcf', 'source': 'run_tinjasmine/clean_VCF'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'charger_tsv'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_neoscan',\n",
    "            'label': 'run_neoscan',\n",
    "            'run': '../../submodules/pecgs-neoscan/cwl/neoscan.cwl',\n",
    "            'in': [\n",
    "                {'id': 'maf', 'source': 'run_tindaisy/output_maf_clean'},\n",
    "                {'id': 'bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'ref_dir', 'source': 'neoscan_ref_dir'},\n",
    "                {'id': 'bed', 'source': 'neoscan_bed'},\n",
    "                {'id': 'input_type', 'source': 'neoscan_input_type'}\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'snv_summary'},\n",
    "                {'id': 'indel_summary'},\n",
    "            ]\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "   'requirements': [\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "845c4db4",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_TN_wxs_bam.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6162655e",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d86395a9",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36d6d70f",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "eb7307b7",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### pecgs_TN_wgs_bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "ed30d5ff",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "template = {\n",
    "#     'sample': 'a_string',\n",
    "#     'cpu': 'a_int', # not currently actually used for anything\n",
    "    'tumor_wgs_bam': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wgs_bam': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    }, \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "    },\n",
    "    \n",
    "    # SomaticSV\n",
    "    'generate_evidence_bam': 'a_bool'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "2aa31f0d",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_TN_wgs_bam.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "61256337",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "\n",
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_TN_wgs_bam',\n",
    "    'label': 'pecgs_TN_wgs_bam',\n",
    "    'inputs': [\n",
    "#         {\n",
    "#             'id': 'sample',\n",
    "#             'type': 'string',\n",
    "#         },\n",
    "#         {\n",
    "#             'id': 'cpu', # not currently actually used for anything\n",
    "#             'type': 'int?',\n",
    "#             'default': 40\n",
    "#         },\n",
    "        {\n",
    "            'id': 'tumor_wgs_bam',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '$(self.basename).bai',\n",
    "           ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wgs_bam',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '$(self.basename).bai',\n",
    "           ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'reference',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '$(self.basename).fai',\n",
    "           ]\n",
    "        },\n",
    "        \n",
    "        # SomaticSV\n",
    "        {\n",
    "            'id': 'generate_evidence_bam',\n",
    "            'type': 'boolean?',\n",
    "            'default': True,\n",
    "        },\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'somatic_sv_vcf',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_somatic_sv/output'\n",
    "        },\n",
    "        {\n",
    "            'id': 'somatic_sv_evidence_bam',\n",
    "            'type': [\n",
    "                'null',\n",
    "                'File',\n",
    "                {'type': 'array', 'items': 'File'}\n",
    "            ],\n",
    "            'outputSource': 'run_somatic_sv/output_evidence'\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'run_somatic_sv',\n",
    "            'label': 'run_somatic_sv',\n",
    "            'run': '../../submodules/SomaticSV/cwl/SomaticSV.cwl',\n",
    "            'in': [\n",
    "                {'id': 'tumor', 'source': 'tumor_wgs_bam'},\n",
    "                {'id': 'normal', 'source': 'normal_wgs_bam'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'generateEvidenceBam', 'source': 'generate_evidence_bam'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output'},\n",
    "                {'id': 'output_evidence'}\n",
    "            ]\n",
    "        },\n",
    "    ],\n",
    "   'requirements': [\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "644b222c",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_TN_wgs_bam.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3580930a",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### pecgs_T_rna_fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "076264ee",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "template = {\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int',\n",
    "    'tumor_rna_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },   \n",
    "    \n",
    "    # fusion\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # bulk expression\n",
    "    'bulk_expression_fq_1_list': [\n",
    "        {\n",
    "            'class': 'File',\n",
    "            'path': 'a/file/path'\n",
    "        }\n",
    "    ],\n",
    "    'bulk_expression_fq_2_list': [\n",
    "        {\n",
    "            'class': 'File',\n",
    "            'path': 'a/file/path'\n",
    "        }\n",
    "    ],\n",
    "    'star_index': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/dir/path'\n",
    "    },\n",
    "    'gtf': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'gene_info': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "86275926",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_T_rna_fq.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "34d08faa",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_T_rna_fq',\n",
    "    'label': 'pecgs_T_rna_fq',\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'cpu',\n",
    "            'type': 'int?',\n",
    "            'default': 40\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # fusion\n",
    "        {\n",
    "            'id': 'genome_lib_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'genome_db',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'bwts',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_executable',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_fasta',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_annotations',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'filter_database',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'fusion_annotator_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        \n",
    "        # bulk_expression\n",
    "        {\n",
    "            'id': 'bulk_expression_fq_1_list',\n",
    "            'type': 'File[]'\n",
    "        },\n",
    "        {\n",
    "            'id': 'bulk_expression_fq_2_list',\n",
    "            'type': 'File[]'\n",
    "        },\n",
    "        {\n",
    "            'id': 'star_index',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'gtf',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        { \n",
    "            'id': 'gene_info',\n",
    "            'type': 'File',\n",
    "        },\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'filtered_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/filtered_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'total_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/total_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'readcounts_and_fpkm_tsv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_bulk_expression/readcounts_and_fpkm_tsv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_bulk_expression/output_bam'\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'run_fusion',\n",
    "            'label': 'run_fusion',\n",
    "            'run': '../../submodules/pecgs-fusion/cwl/fusion.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_rna_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_rna_fq_2'},\n",
    "                {'id': 'filter_database', 'source': 'filter_database'},\n",
    "                {'id': 'bwts', 'source': 'bwts'},\n",
    "                {'id': 'fusion_annotator_dir', 'source': 'fusion_annotator_dir'},\n",
    "                {'id': 'genome_db', 'source': 'genome_db'},\n",
    "                {'id': 'genome_lib_dir', 'source': 'genome_lib_dir'},\n",
    "                {'id': 'integrate_annotations', 'source': 'integrate_annotations'},\n",
    "                {'id': 'integrate_executable', 'source': 'integrate_executable'},\n",
    "                {'id': 'integrate_fasta', 'source': 'integrate_fasta'}\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'filtered_fusions'},\n",
    "                {'id': 'total_fusions'}\n",
    "            ]\n",
    "        },\n",
    "        { \n",
    "            'id': 'run_bulk_expression',\n",
    "            'label': 'run_bulk_expression',\n",
    "            'run': '../../submodules/pecgs-bulk-expression/cwl/bulk_expression.cwl',\n",
    "            'in': [\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'bulk_expression_fq_1_list'},\n",
    "                {'id': 'fq_2', 'source': 'bulk_expression_fq_2_list'},\n",
    "                {'id': 'star_index', 'source': 'star_index'},\n",
    "                {'id': 'gtf', 'source': 'gtf'},\n",
    "                {'id': 'gene_info', 'source': 'gene_info'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'readcounts_and_fpkm_tsv'},\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        }, \n",
    "    ],\n",
    "   'requirements': [\n",
    "#        {'class': 'MultipleInputFeatureRequirement'}\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "4e381cfe",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_T_rna_fq.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4723440f",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c19eaba",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d9f421e",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6625a3a6",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "426424fb",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1003a523",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5542b47",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "9035519b",
   "metadata": {},
   "source": [
    "#### test runs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b047f1d",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### test run pecgs_TN_wxs_fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "92978c75",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b71c6c1e</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>afe9f9e5-2921-4539-9578-fd01b71c6c1e</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>eafb3178-e459-44c0-a374-8e17a8107fce</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>9923349f-dc3c-40d4-a097-803c4828f732</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>78777884-505c-4b2f-9170-107662c1effa</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>2699158f-25fe-4f19-a7cf-2a4e1cf687b3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  afe9f9e5-2921-4539-9578-fd01b71c6c1e   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  eafb3178-e459-44c0-a374-8e17a8107fce   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  9923349f-dc3c-40d4-a097-803c4828f732   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  78777884-505c-4b2f-9170-107662c1effa   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_afe9f9e5-2921-4539-9578-fd01b7...  2699158f-25fe-4f19-a7cf-2a4e1cf687b3  "
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = pd.read_csv('../tests/data/pecgs_TN_wxs_fq/run_list.txt', sep='\\t', index_col='run_id')\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "391a9227",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>lane</th>\n",
       "      <th>index_sequencer</th>\n",
       "      <th>library_preparation</th>\n",
       "      <th>platform</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>tumor</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>normal</td>\n",
       "      <td>HH7KNDSXY</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTATCGCT-CTTGAAGGTT</td>\n",
       "      <td>TWCE-HT191P1-JM1D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           sample_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "\n",
       "                                                   experimental_strategy  \\\n",
       "run_id                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "\n",
       "                                                   sample_type   flowcell  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       tumor  HFMFWDSXY   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...      normal  HH7KNDSXY   \n",
       "\n",
       "                                                    lane  \\\n",
       "run_id                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...     2   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...     1   \n",
       "\n",
       "                                                          index_sequencer  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CCAGTAGCGT-ATGTATTGGC   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CATTATCGCT-CTTGAAGGTT   \n",
       "\n",
       "                                                               library_preparation  \\\n",
       "run_id                                                                               \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  TWCE-HT191P1-S1H1A3Y3D1_1-lib1   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       TWCE-HT191P1-JM1D1_1-lib1   \n",
       "\n",
       "                                                    platform  \n",
       "run_id                                                        \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  "
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequencing_info = pd.read_csv('../tests/data/pecgs_TN_wxs_fq/sequencing_info.txt', sep='\\t', index_col='run_id')\n",
    "sequencing_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "161d0ddd",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "run_dir = '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_fq'\n",
    "tool_root = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline'\n",
    "out_dir = '../tests/data/pecgs_TN_wxs_fq'\n",
    "run_list = 'run_list.txt'\n",
    "sequencing_info = 'sequencing_info.txt'\n",
    "pipeline_variant = 'pecgs_TN_wxs_fq'\n",
    "queue = 'dinglab'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "ea70ec67",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "launch_cmds, make_cmd, tidy_cmd, summarize_cmd = pecgs.create_run_setup_scripts(\n",
    "    tool_root, out_dir, pipeline_variant, run_list, run_dir, sequencing_info=sequencing_info, queue=queue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "2cfa22a2",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_fq\n",
      "export LSF_DOCKER_VOLUMES=\"/storage1/fs1/dinglab/Active:/storage1/fs1/dinglab/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"\n",
      "export PATH=\"/miniconda/envs/pecgs/bin:$PATH\"\n",
      "bsub -q dinglab-interactive -G compute-dinglab -Is -a 'docker(estorrs/pecgs-pipeline:0.0.1)' '/bin/bash'\n"
     ]
    }
   ],
   "source": [
    "for c in launch_cmds: print(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "6e8232a4",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py make-run --sequencing-info sequencing_info.txt --queue dinglab pecgs_TN_wxs_fq run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_fq'"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "make_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "0ebf2a6f",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py tidy-run pecgs_TN_wxs_fq run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_fq'"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tidy_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "4110a622",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py summarize-run pecgs_TN_wxs_fq run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_fq'"
      ]
     },
     "execution_count": 242,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summarize_cmd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c6a9b37",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### test run pecgs_TN_wxs_bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "008cf235",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>wxs_normal_bam.filepath</th>\n",
       "      <th>wxs_normal_bam.uuid</th>\n",
       "      <th>wxs_tumor_bam.filepath</th>\n",
       "      <th>wxs_tumor_bam.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621</th>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>2a9edf1e-76df-4595-9125-fb9b9d919621</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>8919accb-6973-4ec5-b29f-9d895a1713d4</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/estorrs/...</td>\n",
       "      <td>6e0bde4a-22c1-4e1f-a38a-dff275b61472</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  C3L-00677   \n",
       "\n",
       "                                                                            run_uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  2a9edf1e-76df-4595-9125-fb9b9d919621   \n",
       "\n",
       "                                                                          wxs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                 wxs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  8919accb-6973-4ec5-b29f-9d895a1713d4   \n",
       "\n",
       "                                                                           wxs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  /storage1/fs1/dinglab/Active/Projects/estorrs/...   \n",
       "\n",
       "                                                                  wxs_tumor_bam.uuid  \n",
       "run_id                                                                                \n",
       "C3L-00677_2a9edf1e-76df-4595-9125-fb9b9d919621  6e0bde4a-22c1-4e1f-a38a-dff275b61472  "
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = pd.read_csv('../tests/data/pecgs_TN_wxs_bam/run_list.txt', sep='\\t', index_col='run_id')\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "2f55b5cf",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "run_dir = '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_bam'\n",
    "tool_root = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline'\n",
    "out_dir = '../tests/data/pecgs_TN_wxs_bam'\n",
    "run_list = 'run_list.txt'\n",
    "pipeline_variant = 'pecgs_TN_wxs_bam'\n",
    "queue = 'dinglab'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "a9968aa8",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "launch_cmds, make_cmd, tidy_cmd, summarize_cmd = pecgs.create_run_setup_scripts(\n",
    "    tool_root, out_dir, pipeline_variant, run_list, run_dir, queue=queue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "b7e29ecd",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_bam\n",
      "export LSF_DOCKER_VOLUMES=\"/storage1/fs1/dinglab/Active:/storage1/fs1/dinglab/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"\n",
      "export PATH=\"/miniconda/envs/pecgs/bin:$PATH\"\n",
      "bsub -q dinglab-interactive -G compute-dinglab -Is -a 'docker(estorrs/pecgs-pipeline:0.0.1)' '/bin/bash'\n"
     ]
    }
   ],
   "source": [
    "for c in launch_cmds: print(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "a142daf3",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py make-run  --queue dinglab pecgs_TN_wxs_bam run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_bam'"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "make_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "b9c7a9a9",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py tidy-run pecgs_TN_wxs_bam run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_bam'"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tidy_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "9d44420f",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py summarize-run pecgs_TN_wxs_bam run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wxs_bam'"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summarize_cmd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71b666c5",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### test run pecgs_TN_wgs_bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "f63249f5",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>wgs_normal_bam.filepath</th>\n",
       "      <th>wgs_normal_bam.uuid</th>\n",
       "      <th>wgs_tumor_bam.filepath</th>\n",
       "      <th>wgs_tumor_bam.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1</th>\n",
       "      <td>C3N-05929</td>\n",
       "      <td>1fd2054c-fc78-433d-8f99-32f8e6909db1</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>7d8f4ede-1911-456e-9eee-2fafe7132f50</td>\n",
       "      <td>/storage1/fs1/m.wyczalkowski/Active/Primary/CP...</td>\n",
       "      <td>d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  case_id  \\\n",
       "run_id                                                      \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  C3N-05929   \n",
       "\n",
       "                                                                            run_uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  1fd2054c-fc78-433d-8f99-32f8e6909db1   \n",
       "\n",
       "                                                                          wgs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                 wgs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  7d8f4ede-1911-456e-9eee-2fafe7132f50   \n",
       "\n",
       "                                                                           wgs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  /storage1/fs1/m.wyczalkowski/Active/Primary/CP...   \n",
       "\n",
       "                                                                  wgs_tumor_bam.uuid  \n",
       "run_id                                                                                \n",
       "C3N-05929_1fd2054c-fc78-433d-8f99-32f8e6909db1  d3ddd33b-c7a3-4caf-bdd7-c05e98a19d83  "
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = pd.read_csv('../tests/data/pecgs_TN_wgs_bam/run_list.txt', sep='\\t', index_col='run_id')\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "3a0ecc8d",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "run_dir = '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wgs_bam'\n",
    "tool_root = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline'\n",
    "out_dir = '../tests/data/pecgs_TN_wgs_bam'\n",
    "run_list = 'run_list.txt'\n",
    "pipeline_variant = 'pecgs_TN_wgs_bam'\n",
    "queue = 'dinglab'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "a8253036",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "launch_cmds, make_cmd, tidy_cmd, summarize_cmd = pecgs.create_run_setup_scripts(\n",
    "    tool_root, out_dir, pipeline_variant, run_list, run_dir, queue=queue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "9b2f1643",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wgs_bam',\n",
       " 'export LSF_DOCKER_VOLUMES=\"/storage1/fs1/dinglab/Active:/storage1/fs1/dinglab/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"',\n",
       " 'export PATH=\"/miniconda/envs/pecgs/bin:$PATH\"',\n",
       " \"bsub -q dinglab-interactive -G compute-dinglab -Is -a 'docker(estorrs/pecgs-pipeline:0.0.1)' '/bin/bash'\"]"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "launch_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "327d4f69",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py make-run  --queue dinglab pecgs_TN_wgs_bam run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_TN_wgs_bam'"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "make_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "540fdd56",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "1085d177",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### test run pecgs_T_rna_fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "f4cedadc",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5bb5519</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>be6b68e6-365d-4451-8195-f1b3d5bb5519</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>fb1f4b8a-a56b-4bc6-b059-864c2b066dc8</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>80ce8102-e8a4-4529-85b0-701a4dc022ea</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             case_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  be6b68e6-365d-4451-8195-f1b3d5bb5519   \n",
       "\n",
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  fb1f4b8a-a56b-4bc6-b059-864c2b066dc8   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_be6b68e6-365d-4451-8195-f1b3d5...  80ce8102-e8a4-4529-85b0-701a4dc022ea  "
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = pd.read_csv('../tests/data/pecgs_T_rna_fq/run_list.txt', sep='\\t', index_col='run_id')\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "7b5c7dde",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "run_dir = '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_T_rna_fq'\n",
    "tool_root = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline'\n",
    "out_dir = '../tests/data/pecgs_T_rna_fq'\n",
    "run_list = 'run_list.txt'\n",
    "pipeline_variant = 'pecgs_T_rna_fq'\n",
    "queue = 'dinglab'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "5f63041a",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "launch_cmds, make_cmd, tidy_cmd, summarize_cmd = pecgs.create_run_setup_scripts(\n",
    "    tool_root, out_dir, pipeline_variant, run_list, run_dir, queue=queue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "69d7f7dc",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'python /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/src/compute1/generate_run_commands.py make-run  --queue dinglab pecgs_T_rna_fq run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs_T_rna_fq'"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "make_cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37bb7658",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
