{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3006866f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import yaml\n",
    "import uuid\n",
    "import os\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "24d7280e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ba0589e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cc4c27d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "27eb1f60",
   "metadata": {},
   "outputs": [],
   "source": [
    "import wombat.pecgs as pecgs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a862f3b",
   "metadata": {},
   "source": [
    "###### make pecgs pipeline cwl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a0ca8b20",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl_dir = '/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "15c9b45d",
   "metadata": {},
   "outputs": [],
   "source": [
    "template = {\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int',\n",
    "    'tumor_wxs_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_wxs_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_fq_1': {\n",
    "        'class': 'File', \n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    ## align dna-seq\n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    'known_sites': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/vcf.gz',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/vcf.gz.tbi'}\n",
    "        ]\n",
    "    },\n",
    "    'wxs_normal_flowcell': 'a_string',\n",
    "    'wxs_normal_lane': 'a_string',\n",
    "    'wxs_normal_index_sequencer': 'a_string',\n",
    "    'wxs_normal_library_preparation': 'a_string',\n",
    "    'wxs_normal_platform': 'a_string',\n",
    "    'wxs_tumor_flowcell': 'a_string',\n",
    "    'wxs_tumor_lane': 'a_string',\n",
    "    'wxs_tumor_index_sequencer': 'a_string',\n",
    "    'wxs_tumor_library_preparation': 'a_string',\n",
    "    'wxs_tumor_platform': 'a_string',\n",
    "    \n",
    "    # fusion\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # CNV\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 'a_int',\n",
    "    'minimal_microsatellite_size': 'a_int',\n",
    "    \n",
    "    # TinDaisy\n",
    "    'rescue_clinvar': False,\n",
    "    'rescue_cosmic': False,\n",
    "    'vep_cache_version': 'a_string',\n",
    "    'vep_cache_gz': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'clinvar_annotation': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'classification_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'af_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'call_regions': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'normal_barcode': 'a_string',\n",
    "    'tumor_barcode': 'a_string',\n",
    "    'tindaisy_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'strelka_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'centromere_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'assembly': 'a_string',\n",
    "    'varscan_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    \n",
    "    # TinJasmine\n",
    "    'sample_barcode': 'a_string',\n",
    "    'centromere': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'tinjasmine_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'Canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'ROI_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'varscan_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config_template': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'gatk_filter_config': {'class': 'File', 'path': 'a/file/path'}, \n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "584f1bc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_TN_wxs_fq_T_rna_fq.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eac1e9f0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "1c4222a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_TN_wxs_fq_T_rna_fq',\n",
    "    'label': 'pecgs_TN_wxs_fq_T_rna_fq',\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'cpu',\n",
    "            'type': 'int?',\n",
    "            'default': 40\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # align dna-seq\n",
    "        {\n",
    "            'id': 'tumor_sample',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).WXS.T'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_sample',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).WXS.N'\n",
    "        },\n",
    "        {\n",
    "            'id': 'known_sites',\n",
    "            'type': 'File',\n",
    "           'secondaryFiles': ['.tbi']\n",
    "        },\n",
    "        {\n",
    "            'id': 'reference',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '.amb',\n",
    "               '.ann',\n",
    "               '.bwt',\n",
    "               '.fai',\n",
    "               '.pac',\n",
    "               '.sa',\n",
    "               '^.dict'\n",
    "           ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_flowcell',\n",
    "            'type': 'string?',\n",
    "            'default': 'flowcellABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_lane',\n",
    "            'type': 'string?',\n",
    "            'default': '1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_index_sequencer',\n",
    "            'type': 'string?',\n",
    "            'default': 'sequencerABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_library_preparation',\n",
    "            'type': 'string?',\n",
    "            'default': 'lib1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_platform',\n",
    "            'type': 'string?',\n",
    "            'default': 'ILLUMINA'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_flowcell',\n",
    "            'type': 'string?',\n",
    "            'default': 'flowcellABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_lane',\n",
    "            'type': 'string?',\n",
    "            'default': '1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_index_sequencer',\n",
    "            'type': 'string?',\n",
    "            'default': 'sequencerABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_library_preparation',\n",
    "            'type': 'string?',\n",
    "            'default': 'lib1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_platform',\n",
    "            'type': 'string?',\n",
    "            'default': 'ILLUMINA'\n",
    "        },\n",
    "        \n",
    "        # fusion\n",
    "        {\n",
    "            'id': 'genome_lib_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'genome_db',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'bwts',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_executable',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_fasta',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_annotations',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'filter_database',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'fusion_annotator_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        \n",
    "        # CNV\n",
    "        {\n",
    "            'id': 'reference_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        \n",
    "        # msisensor\n",
    "        {\n",
    "            'id': 'microsatellite',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_homopolymer_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_microsatellite_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        \n",
    "        # TinDaisy\n",
    "        {\n",
    "            'id': 'rescue_clinvar',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'rescue_cosmic',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_version',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_gz',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'clinvar_annotation',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'classification_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'af_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'call_regions',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).N',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).T',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'strelka_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'assembly',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # TinJasmine\n",
    "        {\n",
    "            'id': 'sample_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample)'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'Canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'ROI_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config_template',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gatk_filter_config',\n",
    "            'type': 'File'\n",
    "        }\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'tumor_wxs_output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'align_tumor_wxs/output_bam',\n",
    "            'secondaryFiles': ['^.bai']\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'align_normal_wxs/output_bam',\n",
    "            'secondaryFiles': ['^.bai']\n",
    "        },\n",
    "        {\n",
    "            'id': 'filtered_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/filtered_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'total_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/total_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_cnv/gene_level_cnv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_summary'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_dis',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_dis'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_germline',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_germline'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_somatic',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_somatic'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_maf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_all'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_MAF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/allCall_VCF'\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'align_tumor_wxs',\n",
    "            'label': 'align_tumor_wxs',\n",
    "            'run': '../../submodules/align-dnaseq/cwl/align_dnaseq.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'tumor_sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_wxs_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_wxs_fq_2'},\n",
    "                {'id': 'known_sites', 'source': 'known_sites'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'flowcell', 'source': 'wxs_tumor_flowcell'},\n",
    "                {'id': 'lane', 'source': 'wxs_tumor_lane'},\n",
    "                {'id': 'index_sequencer', 'source': 'wxs_tumor_index_sequencer'},\n",
    "                {'id': 'library_preparation', 'source': 'wxs_tumor_library_preparation'},\n",
    "                {'id': 'platform', 'source': 'wxs_tumor_platform'},  \n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'align_normal_wxs',\n",
    "            'label': 'align_normal_wxs',\n",
    "            'run': '../../submodules/align-dnaseq/cwl/align_dnaseq.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'normal_sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'normal_wxs_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'normal_wxs_fq_2'},\n",
    "                {'id': 'known_sites', 'source': 'known_sites'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'flowcell', 'source': 'wxs_normal_flowcell'},\n",
    "                {'id': 'lane', 'source': 'wxs_normal_lane'},\n",
    "                {'id': 'index_sequencer', 'source': 'wxs_normal_index_sequencer'},\n",
    "                {'id': 'library_preparation', 'source': 'wxs_normal_library_preparation'},\n",
    "                {'id': 'platform', 'source': 'wxs_normal_platform'},  \n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_fusion',\n",
    "            'label': 'run_fusion',\n",
    "            'run': '../../submodules/pecgs-fusion/cwl/fusion.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_rna_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_rna_fq_2'},\n",
    "                {'id': 'filter_database', 'source': 'filter_database'},\n",
    "                {'id': 'bwts', 'source': 'bwts'},\n",
    "                {'id': 'fusion_annotator_dir', 'source': 'fusion_annotator_dir'},\n",
    "                {'id': 'genome_db', 'source': 'genome_db'},\n",
    "                {'id': 'genome_lib_dir', 'source': 'genome_lib_dir'},\n",
    "                {'id': 'integrate_annotations', 'source': 'integrate_annotations'},\n",
    "                {'id': 'integrate_executable', 'source': 'integrate_executable'},\n",
    "                {'id': 'integrate_fasta', 'source': 'integrate_fasta'}\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'filtered_fusions'},\n",
    "                {'id': 'total_fusions'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_cnv',\n",
    "            'label': 'run_cnv',\n",
    "            'run': '../../submodules/pecgs-cnv/cwl/cnv_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference_dir', 'source': 'reference_dir'},\n",
    "                {'id': 'target_interval_list', 'source': 'target_interval_list'},\n",
    "                {'id': 'common_biallelic', 'source': 'common_biallelic'},\n",
    "                {'id': 'protein_coding_gene', 'source': 'protein_coding_gene'},\n",
    "                {'id': 'pool_of_normals', 'source': 'pool_of_normals'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'gene_level_cnv'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_msisensor',\n",
    "            'label': 'run_msisensor',\n",
    "            'run': '../msisensor/msisensor_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'threads', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'microsatellite', 'source': 'microsatellite'},\n",
    "                {'id': 'minimal_homopolymer_size', 'source': 'minimal_homopolymer_size'},\n",
    "                {'id': 'minimal_microsatellite_size', 'source': 'minimal_microsatellite_size'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_summary'},\n",
    "                {'id': 'output_dis'},\n",
    "                {'id': 'output_germline'},\n",
    "                {'id': 'output_somatic'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tindaisy',\n",
    "            'label': 'run_tindaisy',\n",
    "            'run': '../../submodules/TinDaisy/cwl/workflows/tindaisy2.cwl',\n",
    "            'in': [\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference_fasta', 'source': 'reference'},\n",
    "                {'id': 'pindel_config', 'source': 'pindel_config'},\n",
    "                {'id': 'varscan_config', 'source': 'varscan_config'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'centromere_bed', 'source': 'centromere_bed'},\n",
    "                {'id': 'strelka_config', 'source': 'strelka_config'},\n",
    "                {'id': 'chrlist', 'source': 'tindaisy_chrlist'},\n",
    "                {'id': 'tumor_barcode', 'source': 'tumor_barcode'},\n",
    "                {'id': 'normal_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'canonical_BED', 'source': 'canonical_BED'},\n",
    "                {'id': 'call_regions', 'source': 'call_regions'},\n",
    "                {'id': 'af_config', 'source': 'af_config'},\n",
    "                {'id': 'classification_config', 'source': 'classification_config'},\n",
    "                {'id': 'clinvar_annotation', 'source': 'clinvar_annotation'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'rescue_cosmic', 'source': 'rescue_cosmic'},\n",
    "                {'id': 'rescue_clinvar', 'source': 'rescue_clinvar'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_maf_clean'},\n",
    "                {'id': 'output_vcf_clean'},\n",
    "                {'id': 'output_vcf_all'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tinjasmine',\n",
    "            'label': 'run_tinjasmine',\n",
    "            'run': '../../submodules/TinJasmine/cwl/TinJasmine.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'gatk_filter_config', 'source': 'gatk_filter_config'},\n",
    "                {'id': 'pindel_config_template', 'source': 'pindel_config_template'},\n",
    "                {'id': 'pindel_filter_config', 'source': 'pindel_filter_config'},\n",
    "                {'id': 'varscan_filter_config', 'source': 'varscan_filter_config'},\n",
    "                {'id': 'ROI_BED', 'source': 'ROI_BED'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'Canonical_BED', 'source': 'Canonical_BED'},\n",
    "                {'id': 'chrlist', 'source': 'tinjasmine_chrlist'},\n",
    "                {'id': 'centromere', 'source': 'centromere'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'clean_VCF'},\n",
    "                {'id': 'allCall_VCF'},\n",
    "                {'id': 'clean_MAF'},\n",
    "            ]\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "   'requirements': [\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "6112b930",
   "metadata": {},
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_TN_wxs_fq_T_rna_fq.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fd2adc4",
   "metadata": {},
   "source": [
    "###### test run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a8d3093f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed</th>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>ab2eb486-8926-4329-b14c-81e9605edcc4</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>530103e2-ecc2-4400-b89b-329fb16631b0</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>ee5aa566-1492-49bd-a44f-70a28a9bd6e1</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>08739a15-9bb4-4ab1-82e4-67ac523d65e4</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>53ccd273-07b5-416f-a68a-bd4953424612</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>6fb1f30a-a19a-4702-9707-7d072d13222e</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>23402d1f-7ea6-48d0-8553-02d5f4163aed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  ab2eb486-8926-4329-b14c-81e9605edcc4   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  530103e2-ecc2-4400-b89b-329fb16631b0   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  ee5aa566-1492-49bd-a44f-70a28a9bd6e1   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  08739a15-9bb4-4ab1-82e4-67ac523d65e4   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  53ccd273-07b5-416f-a68a-bd4953424612   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  6fb1f30a-a19a-4702-9707-7d072d13222e   \n",
       "\n",
       "                                                           sample_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4...  23402d1f-7ea6-48d0-8553-02d5f4163aed  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = pd.read_csv('../tests/data/pecgs_pipeline/run_list.txt', sep='\\t', index_col=0)\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ea4a69f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed': {'rna-seq_tumor_R1': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "  'rna-seq_tumor_R2': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "  'wxs_normal_R1': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R1_001.fastq.gz',\n",
       "  'wxs_normal_R2': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R2_001.fastq.gz',\n",
       "  'wxs_tumor_R1': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz',\n",
       "  'wxs_tumor_R2': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz'}}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_map = run_list.transpose().to_dict()\n",
    "run_map = {k:{c.replace('.filepath', ''):val for c, val in v.items() if 'filepath' in c}\n",
    "          for k, v in run_map.items()}\n",
    "run_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "df550fc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed': {'normal': {'flowcell': 'HH7KNDSXY',\n",
       "   'index_sequencer': 'CATTATCGCT-CTTGAAGGTT',\n",
       "   'lane': '1',\n",
       "   'library_preparation': 'TWCE-HT191P1-JM1D1_1-lib1',\n",
       "   'platform': 'ILLUMINA'},\n",
       "  'tumor': {'flowcell': 'HFMFWDSXY',\n",
       "   'index_sequencer': 'CCAGTAGCGT-ATGTATTGGC',\n",
       "   'lane': '2',\n",
       "   'library_preparation': 'TWCE-HT191P1-S1H1A3Y3D1_1-lib1',\n",
       "   'platform': 'ILLUMINA'}}}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequencing_info = yaml.safe_load(open('../tests/data/pecgs_pipeline/sequencing_info.yaml'))\n",
    "sequencing_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4ef3e79c",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_dir = '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs'\n",
    "proxy_run_dir = '../tests/data/pecgs_pipeline/run_dir'\n",
    "tool_root = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8bfc639d",
   "metadata": {},
   "outputs": [],
   "source": [
    "start_cmds, server_cmds, job_cmds = pecgs.from_run_list_TN_wxs_fq_T_rna_fq(\n",
    "    run_map, run_dir, tool_root, sequencing_info_map=sequencing_info, proxy_run_dir=proxy_run_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "25d6b192",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['source /opt/ibm/lsfsuite/lsf/conf/lsf.conf',\n",
       " 'export LSF_DOCKER_NETWORK=host',\n",
       " 'export LSF_DOCKER_VOLUMES=\"/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs:/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline:/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline /storage1/fs1/dinglab/Active:/storage1/fs1/dinglab/Active /storage1/fs1/m.wyczalkowski/Active:/storage1/fs1/m.wyczalkowski/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"',\n",
       " 'bgadd -L 50 /estorrs/default_pecgs_pipeline',\n",
       " \"bsub -n 1 -q dinglab-interactive -G compute-dinglab -a 'docker(mwyczalkowski/cromwell-runner)' -g /estorrs/default_pecgs_pipeline -J b70f2017-e818-43e2-bfa8-009a831b08a6 -Is '/bin/bash'\"]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "5da8084d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/usr/bin/java -Dconfig.file=/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/inputs/server-cromwell-config.compute1.dat -jar /usr/local/cromwell/cromwell-47.jar server'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "server_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "50d64fd6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/logs',\n",
       " 'mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed/cromwell-workdir/logs',\n",
       " 'source /opt/ibm/lsfsuite/lsf/conf/lsf.conf',\n",
       " 'export LSF_DOCKER_NETWORK=host',\n",
       " 'export LSF_DOCKER_VOLUMES=\"/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs:/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline:/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline /storage1/fs1/dinglab/Active:/storage1/fs1/dinglab/Active /storage1/fs1/m.wyczalkowski/Active:/storage1/fs1/m.wyczalkowski/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"',\n",
       " \"bsub -n 1 -q dinglab -G compute-dinglab -a 'docker(mwyczalkowski/cromwell-runner)' -g /estorrs/default_pecgs_pipeline -J cromwell_launch_HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed -oo /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/logs/HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed.log '/usr/bin/java -Dconfig.file=/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/inputs/HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed.cromwell-config-db.compute1.dat -Djavax.net.ssl.trustStorePassword=changeit -Djavax.net.ssl.trustStore=/gscmnt/gc2560/core/genome/cromwell/cromwell.truststore -jar /usr/local/cromwell/cromwell-47.jar run -t cwl -i /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/inputs/HT191P1-S1H1A3Y3_23402d1f-7ea6-48d0-8553-02d5f4163aed.input.yaml /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/cwl/pecgs_workflows/pecgs_TN_wxs_fq_T_rna_fq.cwl'\"]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "job_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08d907b4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8405bc0e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
