{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f03b205f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import yaml\n",
    "import uuid\n",
    "import os\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "262987e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2f9d0660",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "faef0fc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f403b2e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import wombat.pecgs as pecgs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce669c67",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### make cwl from bam wxs and rnaseq tumor fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "30d69b28",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl_dir = '/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8114e89b",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "template = {\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int',\n",
    "    'tumor_wxs_bam': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_bam': {\n",
    "        'class': 'File', \n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    \n",
    "    # fusion\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # CNV\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 'a_int',\n",
    "    'minimal_microsatellite_size': 'a_int',\n",
    "    \n",
    "    # TinDaisy\n",
    "    'rescue_clinvar': False,\n",
    "    'rescue_cosmic': False,\n",
    "    'vep_cache_version': 'a_string',\n",
    "    'vep_cache_gz': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'clinvar_annotation': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'classification_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'af_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'call_regions': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'normal_barcode': 'a_string',\n",
    "    'tumor_barcode': 'a_string',\n",
    "    'tindaisy_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'strelka_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'centromere_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'assembly': 'a_string',\n",
    "    'varscan_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    \n",
    "    # TinJasmine\n",
    "    'sample_barcode': 'a_string',\n",
    "    'centromere': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'tinjasmine_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'Canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'ROI_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'varscan_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config_template': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'gatk_filter_config': {'class': 'File', 'path': 'a/file/path'}, \n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b41fe371",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_TN_wxs_bam_T_rna_fq.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3cd971d5",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_TN_wxs_bam_T_rna_fq',\n",
    "    'label': 'pecgs_TN_wxs_bam_T_rna_fq',\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'cpu',\n",
    "            'type': 'int?',\n",
    "            'default': 40\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_bam',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_bam',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        {\n",
    "            'id': 'reference',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '.amb',\n",
    "               '.ann',\n",
    "               '.bwt',\n",
    "               '.fai',\n",
    "               '.pac',\n",
    "               '.sa',\n",
    "               '^.dict'\n",
    "           ]\n",
    "        },\n",
    "        \n",
    "        # fusion\n",
    "        {\n",
    "            'id': 'genome_lib_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'genome_db',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'bwts',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_executable',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_fasta',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_annotations',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'filter_database',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'fusion_annotator_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        \n",
    "        # CNV\n",
    "        {\n",
    "            'id': 'reference_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        \n",
    "        # msisensor\n",
    "        {\n",
    "            'id': 'microsatellite',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_homopolymer_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_microsatellite_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        \n",
    "        # TinDaisy\n",
    "        {\n",
    "            'id': 'rescue_clinvar',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'rescue_cosmic',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_version',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_gz',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'clinvar_annotation',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'classification_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'af_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'call_regions',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).N',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).T',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'strelka_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'assembly',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # TinJasmine\n",
    "        {\n",
    "            'id': 'sample_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample)'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'Canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'ROI_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config_template',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gatk_filter_config',\n",
    "            'type': 'File'\n",
    "        }\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'filtered_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/filtered_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'total_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/total_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_cnv/gene_level_cnv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_summary'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_dis',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_dis'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_germline',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_germline'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_somatic',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_somatic'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_maf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_all'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_MAF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/allCall_VCF'\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'run_fusion',\n",
    "            'label': 'run_fusion',\n",
    "            'run': '../../submodules/pecgs-fusion/cwl/fusion.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_rna_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_rna_fq_2'},\n",
    "                {'id': 'filter_database', 'source': 'filter_database'},\n",
    "                {'id': 'bwts', 'source': 'bwts'},\n",
    "                {'id': 'fusion_annotator_dir', 'source': 'fusion_annotator_dir'},\n",
    "                {'id': 'genome_db', 'source': 'genome_db'},\n",
    "                {'id': 'genome_lib_dir', 'source': 'genome_lib_dir'},\n",
    "                {'id': 'integrate_annotations', 'source': 'integrate_annotations'},\n",
    "                {'id': 'integrate_executable', 'source': 'integrate_executable'},\n",
    "                {'id': 'integrate_fasta', 'source': 'integrate_fasta'}\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'filtered_fusions'},\n",
    "                {'id': 'total_fusions'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_cnv',\n",
    "            'label': 'run_cnv',\n",
    "            'run': '../../submodules/pecgs-cnv/cwl/cnv_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'reference_dir', 'source': 'reference_dir'},\n",
    "                {'id': 'target_interval_list', 'source': 'target_interval_list'},\n",
    "                {'id': 'common_biallelic', 'source': 'common_biallelic'},\n",
    "                {'id': 'protein_coding_gene', 'source': 'protein_coding_gene'},\n",
    "                {'id': 'pool_of_normals', 'source': 'pool_of_normals'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'gene_level_cnv'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_msisensor',\n",
    "            'label': 'run_msisensor',\n",
    "            'run': '../msisensor/msisensor_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'threads', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'microsatellite', 'source': 'microsatellite'},\n",
    "                {'id': 'minimal_homopolymer_size', 'source': 'minimal_homopolymer_size'},\n",
    "                {'id': 'minimal_microsatellite_size', 'source': 'minimal_microsatellite_size'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_summary'},\n",
    "                {'id': 'output_dis'},\n",
    "                {'id': 'output_germline'},\n",
    "                {'id': 'output_somatic'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tindaisy',\n",
    "            'label': 'run_tindaisy',\n",
    "            'run': '../../submodules/TinDaisy/cwl/workflows/tindaisy2.cwl',\n",
    "            'in': [\n",
    "                {'id': 'tumor_bam', 'source': 'tumor_wxs_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'tumor_normal_bam'},\n",
    "                {'id': 'reference_fasta', 'source': 'reference'},\n",
    "                {'id': 'pindel_config', 'source': 'pindel_config'},\n",
    "                {'id': 'varscan_config', 'source': 'varscan_config'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'centromere_bed', 'source': 'centromere_bed'},\n",
    "                {'id': 'strelka_config', 'source': 'strelka_config'},\n",
    "                {'id': 'chrlist', 'source': 'tindaisy_chrlist'},\n",
    "                {'id': 'tumor_barcode', 'source': 'tumor_barcode'},\n",
    "                {'id': 'normal_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'canonical_BED', 'source': 'canonical_BED'},\n",
    "                {'id': 'call_regions', 'source': 'call_regions'},\n",
    "                {'id': 'af_config', 'source': 'af_config'},\n",
    "                {'id': 'classification_config', 'source': 'classification_config'},\n",
    "                {'id': 'clinvar_annotation', 'source': 'clinvar_annotation'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'rescue_cosmic', 'source': 'rescue_cosmic'},\n",
    "                {'id': 'rescue_clinvar', 'source': 'rescue_clinvar'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_maf_clean'},\n",
    "                {'id': 'output_vcf_clean'},\n",
    "                {'id': 'output_vcf_all'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tinjasmine',\n",
    "            'label': 'run_tinjasmine',\n",
    "            'run': '../../submodules/TinJasmine/cwl/TinJasmine.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'bam', 'source': 'normal_wxs_bam'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'gatk_filter_config', 'source': 'gatk_filter_config'},\n",
    "                {'id': 'pindel_config_template', 'source': 'pindel_config_template'},\n",
    "                {'id': 'pindel_filter_config', 'source': 'pindel_filter_config'},\n",
    "                {'id': 'varscan_filter_config', 'source': 'varscan_filter_config'},\n",
    "                {'id': 'ROI_BED', 'source': 'ROI_BED'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'Canonical_BED', 'source': 'Canonical_BED'},\n",
    "                {'id': 'chrlist', 'source': 'tinjasmine_chrlist'},\n",
    "                {'id': 'centromere', 'source': 'centromere'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'clean_VCF'},\n",
    "                {'id': 'allCall_VCF'},\n",
    "                {'id': 'clean_MAF'},\n",
    "            ]\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "   'requirements': [\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1fc767b2",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_TN_wxs_bam_T_rna_fq.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ffc992f",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e858a6a5",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba187ac3",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6cb96f1",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca92f294",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc5e2e90",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2674ec8d",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "2b047f1d",
   "metadata": {},
   "source": [
    "###### test run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "38f1bc8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "      <th>wxs_normal_bam.filepath</th>\n",
       "      <th>wxs_normal_bam.uuid</th>\n",
       "      <th>wxs_tumor_bam.filepath</th>\n",
       "      <th>wxs_tumor_bam.uuid</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1</th>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/multiple...</td>\n",
       "      <td>77a6ca24-e9b7-4c2b-b7f3-2ac9fd17f0b9</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/multiple...</td>\n",
       "      <td>8ee74ac6-a768-4c89-a1ac-0f94b0a337e9</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/cliu/MMR...</td>\n",
       "      <td>80f84b14-e98c-433b-bf64-7db51d4b21a5</td>\n",
       "      <td>/storage1/fs1/dinglab/Active/Projects/cliu/MMR...</td>\n",
       "      <td>e7395d87-2524-4712-90c4-d555d693f2b8</td>\n",
       "      <td>MMRF_1250</td>\n",
       "      <td>74405909-4c72-4cca-8991-34d0af397ac1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                              \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  /storage1/fs1/dinglab/Active/Projects/multiple...   \n",
       "\n",
       "                                                               rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                 \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  77a6ca24-e9b7-4c2b-b7f3-2ac9fd17f0b9   \n",
       "\n",
       "                                                                        rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                              \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  /storage1/fs1/dinglab/Active/Projects/multiple...   \n",
       "\n",
       "                                                               rna-seq_tumor_R2.uuid  \\\n",
       "run_id                                                                                 \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  8ee74ac6-a768-4c89-a1ac-0f94b0a337e9   \n",
       "\n",
       "                                                                          wxs_normal_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  /storage1/fs1/dinglab/Active/Projects/cliu/MMR...   \n",
       "\n",
       "                                                                 wxs_normal_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  80f84b14-e98c-433b-bf64-7db51d4b21a5   \n",
       "\n",
       "                                                                           wxs_tumor_bam.filepath  \\\n",
       "run_id                                                                                              \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  /storage1/fs1/dinglab/Active/Projects/cliu/MMR...   \n",
       "\n",
       "                                                                  wxs_tumor_bam.uuid  \\\n",
       "run_id                                                                                 \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  e7395d87-2524-4712-90c4-d555d693f2b8   \n",
       "\n",
       "                                                sample_id  \\\n",
       "run_id                                                      \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  MMRF_1250   \n",
       "\n",
       "                                                                            run_uuid  \n",
       "run_id                                                                                \n",
       "MMRF_1250_74405909-4c72-4cca-8991-34d0af397ac1  74405909-4c72-4cca-8991-34d0af397ac1  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_uuid = str(uuid.uuid4())\n",
    "sample_id = 'MMRF_1250'\n",
    "data = [\n",
    "    [f'{sample_id}_{run_uuid}', '/storage1/fs1/dinglab/Active/Projects/multiple_myeloma/MMRF/RNASeq_tumor_FASTQ/SRR1567020_1.fastq.gz', str(uuid.uuid4()), '/storage1/fs1/dinglab/Active/Projects/multiple_myeloma/MMRF/RNASeq_tumor_FASTQ/SRR1567020_2.fastq.gz', str(uuid.uuid4()), '/storage1/fs1/dinglab/Active/Projects/cliu/MMRF/bulk_dna_alignment/20220130_WXS/outputs/job_40/mark_dup_bam/SRR1566924.WXS.N.bam', str(uuid.uuid4()), '/storage1/fs1/dinglab/Active/Projects/cliu/MMRF/bulk_dna_alignment/20220130_WXS/outputs/job_4/mark_dup_bam/SRR1566923.WXS.T.bam', str(uuid.uuid4()), sample_id, run_uuid]\n",
    "]\n",
    "run_list = pd.DataFrame(data=data,\n",
    "                       columns=['run_id', 'rna-seq_tumor_R1.filepath', 'rna-seq_tumor_R1.uuid', 'rna-seq_tumor_R2.filepath', 'rna-seq_tumor_R2.uuid', 'wxs_normal_bam.filepath', 'wxs_normal_bam.uuid', 'wxs_tumor_bam.filepath', 'wxs_tumor_bam.uuid', 'sample_id', 'run_uuid'])\n",
    "run_list = run_list.set_index('run_id')\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "47c08ae4",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p /home/estorrs/pecgs-pipeline/examples/MMRF_1250"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4bf6dbd1",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_list.to_csv('/home/estorrs/pecgs-pipeline/examples/MMRF_1250/runlist.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a13039b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3dc533cc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
