{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f03b205f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import yaml\n",
    "import uuid\n",
    "import os\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "262987e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2f9d0660",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "faef0fc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f403b2e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import wombat.pecgs as pecgs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d519673c",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### make pecgs pipeline cwl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0608748d",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl_dir = '/home/estorrs/pecgs-pipeline/cwl/pecgs_workflows'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "564856a7",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "template = {\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int',\n",
    "    'tumor_wxs_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_wxs_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_fq_1': {\n",
    "        'class': 'File', \n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'normal_wxs_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'tumor_rna_fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    ## align dna-seq\n",
    "    'reference': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/GRCh38.d1.vd1.fa',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.amb'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.ann'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.bwt'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.fai'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.pac'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.fa.sa'},\n",
    "            {'class': 'File', 'path': 'path/to/GRCh38.d1.vd1.dict'}\n",
    "        ]\n",
    "    },\n",
    "    'known_sites': {\n",
    "        'class': 'File',\n",
    "        'path': 'path/to/vcf.gz',\n",
    "        'secondaryFiles': [\n",
    "            {'class': 'File', 'path': 'path/to/vcf.gz.tbi'}\n",
    "        ]\n",
    "    },\n",
    "    'wxs_normal_flowcell': 'a_string',\n",
    "    'wxs_normal_lane': 'a_string',\n",
    "    'wxs_normal_index_sequencer': 'a_string',\n",
    "    'wxs_normal_library_preparation': 'a_string',\n",
    "    'wxs_normal_platform': 'a_string',\n",
    "    'wxs_tumor_flowcell': 'a_string',\n",
    "    'wxs_tumor_lane': 'a_string',\n",
    "    'wxs_tumor_index_sequencer': 'a_string',\n",
    "    'wxs_tumor_library_preparation': 'a_string',\n",
    "    'wxs_tumor_platform': 'a_string',\n",
    "    \n",
    "    # fusion\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # CNV\n",
    "    'reference_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'target_interval_list': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'common_biallelic': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'protein_coding_gene': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'pool_of_normals': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    \n",
    "    # msisensor\n",
    "    'microsatellite': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'minimal_homopolymer_size': 'a_int',\n",
    "    'minimal_microsatellite_size': 'a_int',\n",
    "    \n",
    "    # TinDaisy\n",
    "    'rescue_clinvar': False,\n",
    "    'rescue_cosmic': False,\n",
    "    'vep_cache_version': 'a_string',\n",
    "    'vep_cache_gz': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'clinvar_annotation': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'classification_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'af_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'call_regions': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'normal_barcode': 'a_string',\n",
    "    'tumor_barcode': 'a_string',\n",
    "    'tindaisy_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'strelka_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'centromere_bed': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'assembly': 'a_string',\n",
    "    'varscan_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    \n",
    "    # TinJasmine\n",
    "    'sample_barcode': 'a_string',\n",
    "    'centromere': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'tinjasmine_chrlist': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'Canonical_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'ROI_BED': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'varscan_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_filter_config': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'pindel_config_template': {'class': 'File', 'path': 'a/file/path'},\n",
    "    'gatk_filter_config': {'class': 'File', 'path': 'a/file/path'}, \n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "8217b772",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(template, open(os.path.join(cwl_dir, 'template.pecgs_TN_wxs_fq_T_rna_fq.yaml'), 'w'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65d50337",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "2dca867c",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'class': 'Workflow',\n",
    "    'id': 'pecgs_TN_wxs_fq_T_rna_fq',\n",
    "    'label': 'pecgs_TN_wxs_fq_T_rna_fq',\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'cpu',\n",
    "            'type': 'int?',\n",
    "            'default': 40\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_wxs_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_1',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_rna_fq_2',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # align dna-seq\n",
    "        {\n",
    "            'id': 'tumor_sample',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).WXS.T'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_sample',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).WXS.N'\n",
    "        },\n",
    "        {\n",
    "            'id': 'known_sites',\n",
    "            'type': 'File',\n",
    "           'secondaryFiles': ['.tbi']\n",
    "        },\n",
    "        {\n",
    "            'id': 'reference',\n",
    "            'type': 'File',\n",
    "            'secondaryFiles': [\n",
    "               '.amb',\n",
    "               '.ann',\n",
    "               '.bwt',\n",
    "               '.fai',\n",
    "               '.pac',\n",
    "               '.sa',\n",
    "               '^.dict'\n",
    "           ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_flowcell',\n",
    "            'type': 'string?',\n",
    "            'default': 'flowcellABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_lane',\n",
    "            'type': 'string?',\n",
    "            'default': '1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_index_sequencer',\n",
    "            'type': 'string?',\n",
    "            'default': 'sequencerABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_library_preparation',\n",
    "            'type': 'string?',\n",
    "            'default': 'lib1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_normal_platform',\n",
    "            'type': 'string?',\n",
    "            'default': 'ILLUMINA'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_flowcell',\n",
    "            'type': 'string?',\n",
    "            'default': 'flowcellABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_lane',\n",
    "            'type': 'string?',\n",
    "            'default': '1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_index_sequencer',\n",
    "            'type': 'string?',\n",
    "            'default': 'sequencerABCDE'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_library_preparation',\n",
    "            'type': 'string?',\n",
    "            'default': 'lib1'\n",
    "        },\n",
    "        {\n",
    "            'id': 'wxs_tumor_platform',\n",
    "            'type': 'string?',\n",
    "            'default': 'ILLUMINA'\n",
    "        },\n",
    "        \n",
    "        # fusion\n",
    "        {\n",
    "            'id': 'genome_lib_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'genome_db',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'bwts',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_executable',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_fasta',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'integrate_annotations',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'filter_database',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'fusion_annotator_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        \n",
    "        # CNV\n",
    "        {\n",
    "            'id': 'reference_dir',\n",
    "            'type': 'Directory',\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        \n",
    "        # msisensor\n",
    "        {\n",
    "            'id': 'microsatellite',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_homopolymer_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        {\n",
    "            'id': 'minimal_microsatellite_size',\n",
    "            'type': 'int?',\n",
    "            'default': 1\n",
    "        },\n",
    "        \n",
    "        # TinDaisy\n",
    "        {\n",
    "            'id': 'rescue_clinvar',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'rescue_cosmic',\n",
    "            'type': 'boolean?',\n",
    "            'default': False,\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_version',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'vep_cache_gz',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'clinvar_annotation',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'classification_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'af_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'call_regions',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).N',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample).T',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'strelka_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere_bed',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'assembly',\n",
    "            'type': 'string'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        \n",
    "        # TinJasmine\n",
    "        {\n",
    "            'id': 'sample_barcode',\n",
    "            'type': 'string?',\n",
    "            'default': '$(inputs.sample)'\n",
    "        },\n",
    "        {\n",
    "            'id': 'centromere',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_chrlist',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'Canonical_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'ROI_BED',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'varscan_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_filter_config',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'pindel_config_template',\n",
    "            'type': 'File'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gatk_filter_config',\n",
    "            'type': 'File'\n",
    "        }\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'tumor_wxs_output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'align_tumor_wxs/output_bam',\n",
    "            'secondaryFiles': ['^.bai']\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_wxs_output_bam',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'align_normal_wxs/output_bam',\n",
    "            'secondaryFiles': ['^.bai']\n",
    "        },\n",
    "        {\n",
    "            'id': 'filtered_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/filtered_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'total_fusions',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_fusion/total_fusions'\n",
    "        },\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_cnv/gene_level_cnv'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_summary',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_summary'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_dis',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_dis'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_germline',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_germline'\n",
    "        },\n",
    "        {\n",
    "            'id': 'msisensor_output_somatic',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_msisensor/output_somatic'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_maf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_clean'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tindaisy_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tindaisy/output_vcf_all'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_maf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_MAF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_clean',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/clean_VCF'\n",
    "        },\n",
    "        {\n",
    "            'id': 'tinjasmine_output_vcf_all',\n",
    "            'type': 'File',\n",
    "            'outputSource': 'run_tinjasmine/allCall_VCF'\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'align_tumor_wxs',\n",
    "            'label': 'align_tumor_wxs',\n",
    "            'run': '../../submodules/align-dnaseq/cwl/align_dnaseq.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'tumor_sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_wxs_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_wxs_fq_2'},\n",
    "                {'id': 'known_sites', 'source': 'known_sites'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'flowcell', 'source': 'wxs_tumor_flowcell'},\n",
    "                {'id': 'lane', 'source': 'wxs_tumor_lane'},\n",
    "                {'id': 'index_sequencer', 'source': 'wxs_tumor_index_sequencer'},\n",
    "                {'id': 'library_preparation', 'source': 'wxs_tumor_library_preparation'},\n",
    "                {'id': 'platform', 'source': 'wxs_tumor_platform'},  \n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'align_normal_wxs',\n",
    "            'label': 'align_normal_wxs',\n",
    "            'run': '../../submodules/align-dnaseq/cwl/align_dnaseq.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'normal_sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'normal_wxs_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'normal_wxs_fq_2'},\n",
    "                {'id': 'known_sites', 'source': 'known_sites'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'flowcell', 'source': 'wxs_normal_flowcell'},\n",
    "                {'id': 'lane', 'source': 'wxs_normal_lane'},\n",
    "                {'id': 'index_sequencer', 'source': 'wxs_normal_index_sequencer'},\n",
    "                {'id': 'library_preparation', 'source': 'wxs_normal_library_preparation'},\n",
    "                {'id': 'platform', 'source': 'wxs_normal_platform'},  \n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_bam'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_fusion',\n",
    "            'label': 'run_fusion',\n",
    "            'run': '../../submodules/pecgs-fusion/cwl/fusion.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'fq_1', 'source': 'tumor_rna_fq_1'},\n",
    "                {'id': 'fq_2', 'source': 'tumor_rna_fq_2'},\n",
    "                {'id': 'filter_database', 'source': 'filter_database'},\n",
    "                {'id': 'bwts', 'source': 'bwts'},\n",
    "                {'id': 'fusion_annotator_dir', 'source': 'fusion_annotator_dir'},\n",
    "                {'id': 'genome_db', 'source': 'genome_db'},\n",
    "                {'id': 'genome_lib_dir', 'source': 'genome_lib_dir'},\n",
    "                {'id': 'integrate_annotations', 'source': 'integrate_annotations'},\n",
    "                {'id': 'integrate_executable', 'source': 'integrate_executable'},\n",
    "                {'id': 'integrate_fasta', 'source': 'integrate_fasta'}\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'filtered_fusions'},\n",
    "                {'id': 'total_fusions'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_cnv',\n",
    "            'label': 'run_cnv',\n",
    "            'run': '../../submodules/pecgs-cnv/cwl/cnv_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'cpu', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference_dir', 'source': 'reference_dir'},\n",
    "                {'id': 'target_interval_list', 'source': 'target_interval_list'},\n",
    "                {'id': 'common_biallelic', 'source': 'common_biallelic'},\n",
    "                {'id': 'protein_coding_gene', 'source': 'protein_coding_gene'},\n",
    "                {'id': 'pool_of_normals', 'source': 'pool_of_normals'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'gene_level_cnv'}\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_msisensor',\n",
    "            'label': 'run_msisensor',\n",
    "            'run': '../msisensor/msisensor_workflow.cwl',\n",
    "            'in': [\n",
    "                {'id': 'threads', 'source': 'cpu'},\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'microsatellite', 'source': 'microsatellite'},\n",
    "                {'id': 'minimal_homopolymer_size', 'source': 'minimal_homopolymer_size'},\n",
    "                {'id': 'minimal_microsatellite_size', 'source': 'minimal_microsatellite_size'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_summary'},\n",
    "                {'id': 'output_dis'},\n",
    "                {'id': 'output_germline'},\n",
    "                {'id': 'output_somatic'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tindaisy',\n",
    "            'label': 'run_tindaisy',\n",
    "            'run': '../../submodules/TinDaisy/cwl/workflows/tindaisy2.cwl',\n",
    "            'in': [\n",
    "                {'id': 'tumor_bam', 'source': 'align_tumor_wxs/output_bam'},\n",
    "                {'id': 'normal_bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference_fasta', 'source': 'reference'},\n",
    "                {'id': 'pindel_config', 'source': 'pindel_config'},\n",
    "                {'id': 'varscan_config', 'source': 'varscan_config'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'centromere_bed', 'source': 'centromere_bed'},\n",
    "                {'id': 'strelka_config', 'source': 'strelka_config'},\n",
    "                {'id': 'chrlist', 'source': 'tindaisy_chrlist'},\n",
    "                {'id': 'tumor_barcode', 'source': 'tumor_barcode'},\n",
    "                {'id': 'normal_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'canonical_BED', 'source': 'canonical_BED'},\n",
    "                {'id': 'call_regions', 'source': 'call_regions'},\n",
    "                {'id': 'af_config', 'source': 'af_config'},\n",
    "                {'id': 'classification_config', 'source': 'classification_config'},\n",
    "                {'id': 'clinvar_annotation', 'source': 'clinvar_annotation'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'rescue_cosmic', 'source': 'rescue_cosmic'},\n",
    "                {'id': 'rescue_clinvar', 'source': 'rescue_clinvar'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'output_maf_clean'},\n",
    "                {'id': 'output_vcf_clean'},\n",
    "                {'id': 'output_vcf_all'},\n",
    "            ]\n",
    "        },\n",
    "        {\n",
    "            'id': 'run_tinjasmine',\n",
    "            'label': 'run_tinjasmine',\n",
    "            'run': '../../submodules/TinJasmine/cwl/TinJasmine.cwl',\n",
    "            'in': [\n",
    "                {'id': 'sample_barcode', 'source': 'normal_barcode'},\n",
    "                {'id': 'bam', 'source': 'align_normal_wxs/output_bam'},\n",
    "                {'id': 'reference', 'source': 'reference'},\n",
    "                {'id': 'gatk_filter_config', 'source': 'gatk_filter_config'},\n",
    "                {'id': 'pindel_config_template', 'source': 'pindel_config_template'},\n",
    "                {'id': 'pindel_filter_config', 'source': 'pindel_filter_config'},\n",
    "                {'id': 'varscan_filter_config', 'source': 'varscan_filter_config'},\n",
    "                {'id': 'ROI_BED', 'source': 'ROI_BED'},\n",
    "                {'id': 'vep_cache_gz', 'source': 'vep_cache_gz'},\n",
    "                {'id': 'vep_cache_version', 'source': 'vep_cache_version'},\n",
    "                {'id': 'assembly', 'source': 'assembly'},\n",
    "                {'id': 'Canonical_BED', 'source': 'Canonical_BED'},\n",
    "                {'id': 'chrlist', 'source': 'tinjasmine_chrlist'},\n",
    "                {'id': 'centromere', 'source': 'centromere'},\n",
    "            ],\n",
    "            'out': [\n",
    "                {'id': 'clean_VCF'},\n",
    "                {'id': 'allCall_VCF'},\n",
    "                {'id': 'clean_MAF'},\n",
    "            ]\n",
    "        },\n",
    "        \n",
    "    ],\n",
    "   'requirements': [\n",
    "   ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0e54a71f",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "yaml.safe_dump(cwl, open(os.path.join(cwl_dir, 'pecgs_TN_wxs_fq_T_rna_fq.cwl'), 'w'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b047f1d",
   "metadata": {},
   "source": [
    "###### test run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "92978c75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna-seq_tumor_R1.filepath</th>\n",
       "      <th>rna-seq_tumor_R1.uuid</th>\n",
       "      <th>rna-seq_tumor_R2.filepath</th>\n",
       "      <th>rna-seq_tumor_R2.uuid</th>\n",
       "      <th>wxs_normal_R1.filepath</th>\n",
       "      <th>wxs_normal_R1.uuid</th>\n",
       "      <th>wxs_normal_R2.filepath</th>\n",
       "      <th>wxs_normal_R2.uuid</th>\n",
       "      <th>wxs_tumor_R1.filepath</th>\n",
       "      <th>wxs_tumor_R1.uuid</th>\n",
       "      <th>wxs_tumor_R2.filepath</th>\n",
       "      <th>wxs_tumor_R2.uuid</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>fb1f4b8a-a56b-4bc6-b059-864c2b066dc8</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>80ce8102-e8a4-4529-85b0-701a4dc022ea</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>eafb3178-e459-44c0-a374-8e17a8107fce</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>9923349f-dc3c-40d4-a097-803c4828f732</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>78777884-505c-4b2f-9170-107662c1effa</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/pecgs_resources/...</td>\n",
       "      <td>2699158f-25fe-4f19-a7cf-2a4e1cf687b3</td>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            rna-seq_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  fb1f4b8a-a56b-4bc6-b059-864c2b066dc8   \n",
       "\n",
       "                                                                            rna-seq_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                   rna-seq_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  80ce8102-e8a4-4529-85b0-701a4dc022ea   \n",
       "\n",
       "                                                                               wxs_normal_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  eafb3178-e459-44c0-a374-8e17a8107fce   \n",
       "\n",
       "                                                                               wxs_normal_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                      wxs_normal_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  9923349f-dc3c-40d4-a097-803c4828f732   \n",
       "\n",
       "                                                                                wxs_tumor_R1.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R1.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  78777884-505c-4b2f-9170-107662c1effa   \n",
       "\n",
       "                                                                                wxs_tumor_R2.filepath  \\\n",
       "run_id                                                                                                  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  /scratch1/fs1/dinglab/estorrs/pecgs_resources/...   \n",
       "\n",
       "                                                                       wxs_tumor_R2.uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  2699158f-25fe-4f19-a7cf-2a4e1cf687b3   \n",
       "\n",
       "                                                           sample_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \n",
       "run_id                                                                                    \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_list = pd.read_csv('../tests/data/pecgs_pipeline/run_list.txt', sep='\\t', index_col='run_id')\n",
    "run_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "dc040bf1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8': {'rna-seq_tumor_R1': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R1.fastq.gz',\n",
       "  'rna-seq_tumor_R2': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/rna-seq/ht191p1-s1h1a3y3e1_1.AAGGTGTTAG-TAAGCGCGTG.HFWJGDSXY_AAGGTGTTAG-TAAGCGCGTG_L001_R2.fastq.gz',\n",
       "  'wxs_normal_R1': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R1_001.fastq.gz',\n",
       "  'wxs_normal_R2': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CATTATCGCT-CTTGAAGGTT_S34_L001_R2_001.fastq.gz',\n",
       "  'wxs_tumor_R1': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R1_001.fastq.gz',\n",
       "  'wxs_tumor_R2': '/scratch1/fs1/dinglab/estorrs/pecgs_resources/test_samples/HT191P1-S1H1A3Y3/wxs/CCAGTAGCGT-ATGTATTGGC_S53_L002_R2_001.fastq.gz'}}"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_map = run_list.transpose().to_dict()\n",
    "run_map = {k:{c.replace('.filepath', ''):val for c, val in v.items() if 'filepath' in c}\n",
    "          for k, v in run_map.items()}\n",
    "run_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "391a9227",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>lane</th>\n",
       "      <th>index_sequencer</th>\n",
       "      <th>library_preparation</th>\n",
       "      <th>platform</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>tumor</td>\n",
       "      <td>HFMFWDSXY</td>\n",
       "      <td>2</td>\n",
       "      <td>CCAGTAGCGT-ATGTATTGGC</td>\n",
       "      <td>TWCE-HT191P1-S1H1A3Y3D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8</th>\n",
       "      <td>HT191P1-S1H1A3Y3</td>\n",
       "      <td>419b8098-e4b2-4318-883d-c233cb7e06c8</td>\n",
       "      <td>wxs</td>\n",
       "      <td>normal</td>\n",
       "      <td>HH7KNDSXY</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTATCGCT-CTTGAAGGTT</td>\n",
       "      <td>TWCE-HT191P1-JM1D1_1-lib1</td>\n",
       "      <td>ILLUMINA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           sample_id  \\\n",
       "run_id                                                                 \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  HT191P1-S1H1A3Y3   \n",
       "\n",
       "                                                                                run_uuid  \\\n",
       "run_id                                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  419b8098-e4b2-4318-883d-c233cb7e06c8   \n",
       "\n",
       "                                                   experimental_strategy  \\\n",
       "run_id                                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...                   wxs   \n",
       "\n",
       "                                                   sample_type   flowcell  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       tumor  HFMFWDSXY   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...      normal  HH7KNDSXY   \n",
       "\n",
       "                                                    lane  \\\n",
       "run_id                                                     \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...     2   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...     1   \n",
       "\n",
       "                                                          index_sequencer  \\\n",
       "run_id                                                                      \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CCAGTAGCGT-ATGTATTGGC   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  CATTATCGCT-CTTGAAGGTT   \n",
       "\n",
       "                                                               library_preparation  \\\n",
       "run_id                                                                               \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  TWCE-HT191P1-S1H1A3Y3D1_1-lib1   \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...       TWCE-HT191P1-JM1D1_1-lib1   \n",
       "\n",
       "                                                    platform  \n",
       "run_id                                                        \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  \n",
       "HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb...  ILLUMINA  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequencing_info = pd.read_csv('../tests/data/pecgs_pipeline/sequencing_info.txt', sep='\\t', index_col='run_id')\n",
    "sequencing_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "161d0ddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_dir = '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs'\n",
    "proxy_run_dir = '../tests/data/pecgs_pipeline/run_dir'\n",
    "tool_root = '/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "7ebda8cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "start_cmds, server_cmds, job_cmds = pecgs.from_run_list_TN_wxs_fq_T_rna_fq(\n",
    "    run_map, run_dir, tool_root, sequencing_info=sequencing_info, proxy_run_dir=proxy_run_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "07758f9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['source /opt/ibm/lsfsuite/lsf/conf/lsf.conf',\n",
       " 'export LSF_DOCKER_NETWORK=host',\n",
       " 'export LSF_DOCKER_VOLUMES=\"/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs:/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline:/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline /storage1/fs1/dinglab:/storage1/fs1/dinglab /storage1/fs1/m.wyczalkowski/Active:/storage1/fs1/m.wyczalkowski/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"',\n",
       " \"bsub -n 1 -q dinglab-interactive -G compute-dinglab -a 'docker(mwyczalkowski/cromwell-runner)' -Is '/bin/bash'\"]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "4120583c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/usr/bin/java -Dconfig.file=/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/inputs/server-cromwell-config.compute1.dat -jar /usr/local/cromwell/cromwell-47.jar server'"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "server_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "9ece31d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/logs',\n",
       " 'mkdir -p /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/logs',\n",
       " 'source /opt/ibm/lsfsuite/lsf/conf/lsf.conf',\n",
       " 'export LSF_DOCKER_NETWORK=host',\n",
       " 'export LSF_DOCKER_VOLUMES=\"/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs:/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline:/storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline /storage1/fs1/dinglab:/storage1/fs1/dinglab /storage1/fs1/m.wyczalkowski/Active:/storage1/fs1/m.wyczalkowski/Active /scratch1/fs1/dinglab:/scratch1/fs1/dinglab\"',\n",
       " \"bsub -n 1 -q dinglab -G compute-dinglab -a 'docker(mwyczalkowski/cromwell-runner)' -oo /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/logs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.log '/usr/bin/java -Dconfig.file=/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/inputs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.cromwell-config-db.compute1.dat -Djavax.net.ssl.trustStorePassword=changeit -Djavax.net.ssl.trustStore=/gscmnt/gc2560/core/genome/cromwell/cromwell.truststore -jar /usr/local/cromwell/cromwell-47.jar run -t cwl -i /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/inputs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.input.yaml /storage1/fs1/dinglab/Active/Projects/estorrs/pecgs-pipeline/cwl/pecgs_workflows/pecgs_TN_wxs_fq_T_rna_fq.cwl'\"]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "job_cmds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "192ac111",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pecgs.tidy_run_commands(run_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "bc5606d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from wombat.utils import listfiles\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "adb78479",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'tinjasmine_output_vcf_clean': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_tinjasmine/TinJasmine.cwl/d58a5ca5-8797-48a3-958c-003030f50394/call-canonical_filter/execution/output/HotspotFiltered.vcf',\n",
       " 'filtered_fusions': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_fusion/execution/glob-360b64828b25dc1ce023ac7581421672/Filtered_Fusions_in_HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.tsv',\n",
       " 'msisensor_output_somatic': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_msisensor/msisensor_workflow.cwl/d08a70a7-aeae-4ae6-9563-72661bb277a1/call-msisensor/execution/output_somatic',\n",
       " 'tinjasmine_output_vcf_all': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_tinjasmine/TinJasmine.cwl/d58a5ca5-8797-48a3-958c-003030f50394/call-vep_annotate/execution/results/vep/output_vep.vcf',\n",
       " 'tumor_wxs_output_bam': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-align_tumor_wxs/execution/output.bam',\n",
       " 'tindaisy_output_maf_clean': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_tindaisy/tindaisy2.cwl/83249bbf-668d-4d3a-a397-f713bc4ef58a/call-vcf2maf/execution/result.maf',\n",
       " 'msisensor_output_dis': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_msisensor/msisensor_workflow.cwl/d08a70a7-aeae-4ae6-9563-72661bb277a1/call-msisensor/execution/output_dis',\n",
       " 'normal_wxs_output_bam': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-align_normal_wxs/execution/output.bam',\n",
       " 'gene_level_cnv': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_cnv/cnv_workflow.cwl/ead7d4f3-ad20-4ced-ba06-9c077e9c3c5d/call-cnv/execution/glob-ee91797f7d054f32de1c4f0eb6182ea0/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.T.geneLevel.from_seg.cn',\n",
       " 'msisensor_output_germline': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_msisensor/msisensor_workflow.cwl/d08a70a7-aeae-4ae6-9563-72661bb277a1/call-msisensor/execution/output_germline',\n",
       " 'tindaisy_output_vcf_clean': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_tindaisy/tindaisy2.cwl/83249bbf-668d-4d3a-a397-f713bc4ef58a/call-canonical_filter/execution/output/HotspotFiltered.vcf',\n",
       " 'msisensor_output_summary': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_msisensor/msisensor_workflow.cwl/d08a70a7-aeae-4ae6-9563-72661bb277a1/call-msisensor/execution/output',\n",
       " 'total_fusions': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_fusion/execution/glob-cb2a3e6dbb8d92bdc0ba93689a5dcd80/Total_Fusions_in_HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.tsv',\n",
       " 'tindaisy_output_vcf_all': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_tindaisy/tindaisy2.cwl/83249bbf-668d-4d3a-a397-f713bc4ef58a/call-snp_indel_proximity_filter/execution/output/ProximityFiltered.vcf',\n",
       " 'tinjasmine_output_maf_clean': '/scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs/runs/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8/cromwell-workdir/cromwell-executions/pecgs_TN_wxs_fq_T_rna_fq.cwl/8b125db5-03f0-4cf7-be1a-7663e61e458b/call-run_tinjasmine/TinJasmine.cwl/d58a5ca5-8797-48a3-958c-003030f50394/call-vcf2maf/execution/result.maf'}"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def parse_output_from_log(log_fp, workflow_name):\n",
    "    f = open(log_fp)\n",
    "    identifier = None\n",
    "    m = {}\n",
    "    for line in f:\n",
    "        line = line.strip()\n",
    "        if f'\"{workflow_name}.cwl' in line:\n",
    "            identifier = re.sub(r'^.*\".*cwl.(.*)\".*$', r'\\1', line)\n",
    "        if '\"location\": ' in line and identifier is not None:\n",
    "            location = re.sub(r'^.*location.*\"(.*)\".*$', r'\\1', line)\n",
    "            m[identifier] = location\n",
    "            identifier = None\n",
    "    return m\n",
    "\n",
    "workflow_name = 'pecgs_TN_wxs_fq_T_rna_fq'\n",
    "log_fp = '/data/estorrs/sandbox/HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8.log'\n",
    "m = parse_output_from_log(log_fp, workflow_name)\n",
    "m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "88ac12ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# summary format\n",
    "# sample_id cwl_step output_name data_path filesize result_uuid run_id run_uuid run_date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "f4dee6e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>workflow_step</th>\n",
       "      <th>result_name</th>\n",
       "      <th>data_path</th>\n",
       "      <th>filesize</th>\n",
       "      <th>result_uuid</th>\n",
       "      <th>run_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>run_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_tinjasmine</td>\n",
       "      <td>tinjasmine_output_vcf_clean</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>e4b9cf29-ad1c-405d-9ea7-d8d65e8dc126</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_fusion</td>\n",
       "      <td>filtered_fusions</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>7f69e7e3-8bff-49e1-aa2c-f3185682219b</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_msisensor</td>\n",
       "      <td>msisensor_output_somatic</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>6ac4038a-6730-44a6-a13d-0d1166aa9795</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_tinjasmine</td>\n",
       "      <td>tinjasmine_output_vcf_all</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>5e3d54cc-e31e-494e-9aff-21eef560073d</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sample</td>\n",
       "      <td>align_tumor_wxs</td>\n",
       "      <td>tumor_wxs_output_bam</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>0ba74a48-3240-426b-b85d-f8959fe431cd</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_tindaisy</td>\n",
       "      <td>tindaisy_output_maf_clean</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>842cb861-b234-49e3-a29c-3eb1533d5d2a</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_msisensor</td>\n",
       "      <td>msisensor_output_dis</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>d0695897-e749-4050-9034-20079ef38a85</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>sample</td>\n",
       "      <td>align_normal_wxs</td>\n",
       "      <td>normal_wxs_output_bam</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>16ac7507-9401-4791-8c8c-c653e8852bc5</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_cnv</td>\n",
       "      <td>gene_level_cnv</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>e23f7b7b-8491-4548-ae27-ce6a088ebb6e</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_msisensor</td>\n",
       "      <td>msisensor_output_germline</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>76540bb7-6915-4adc-bd4f-353acd38c4ae</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_tindaisy</td>\n",
       "      <td>tindaisy_output_vcf_clean</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>b623c611-8225-45a6-a0b5-e6eacb7896b8</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_msisensor</td>\n",
       "      <td>msisensor_output_summary</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>1e60ef76-3006-4d6e-86af-427f308cdcf4</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_fusion</td>\n",
       "      <td>total_fusions</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>496ec571-b9c9-43af-8664-db95878ea270</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_tindaisy</td>\n",
       "      <td>tindaisy_output_vcf_all</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>89aad415-392d-47cc-879f-3637b73db7f9</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>sample</td>\n",
       "      <td>run_tinjasmine</td>\n",
       "      <td>tinjasmine_output_maf_clean</td>\n",
       "      <td>/scratch1/fs1/dinglab/estorrs/cromwell-data/pe...</td>\n",
       "      <td>filesize</td>\n",
       "      <td>5d546f41-3994-4cfe-9158-869a4338f26f</td>\n",
       "      <td>dummy</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     workflow_step                  result_name  \\\n",
       "0     sample    run_tinjasmine  tinjasmine_output_vcf_clean   \n",
       "1     sample        run_fusion             filtered_fusions   \n",
       "2     sample     run_msisensor     msisensor_output_somatic   \n",
       "3     sample    run_tinjasmine    tinjasmine_output_vcf_all   \n",
       "4     sample   align_tumor_wxs         tumor_wxs_output_bam   \n",
       "5     sample      run_tindaisy    tindaisy_output_maf_clean   \n",
       "6     sample     run_msisensor         msisensor_output_dis   \n",
       "7     sample  align_normal_wxs        normal_wxs_output_bam   \n",
       "8     sample           run_cnv               gene_level_cnv   \n",
       "9     sample     run_msisensor    msisensor_output_germline   \n",
       "10    sample      run_tindaisy    tindaisy_output_vcf_clean   \n",
       "11    sample     run_msisensor     msisensor_output_summary   \n",
       "12    sample        run_fusion                total_fusions   \n",
       "13    sample      run_tindaisy      tindaisy_output_vcf_all   \n",
       "14    sample    run_tinjasmine  tinjasmine_output_maf_clean   \n",
       "\n",
       "                                            data_path  filesize  \\\n",
       "0   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "1   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "2   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "3   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "4   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "5   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "6   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "7   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "8   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "9   /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "10  /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "11  /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "12  /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "13  /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "14  /scratch1/fs1/dinglab/estorrs/cromwell-data/pe...  filesize   \n",
       "\n",
       "                             result_uuid run_id run_uuid    run_date  \n",
       "0   e4b9cf29-ad1c-405d-9ea7-d8d65e8dc126  dummy    dummy  2022-02-23  \n",
       "1   7f69e7e3-8bff-49e1-aa2c-f3185682219b  dummy    dummy  2022-02-23  \n",
       "2   6ac4038a-6730-44a6-a13d-0d1166aa9795  dummy    dummy  2022-02-23  \n",
       "3   5e3d54cc-e31e-494e-9aff-21eef560073d  dummy    dummy  2022-02-23  \n",
       "4   0ba74a48-3240-426b-b85d-f8959fe431cd  dummy    dummy  2022-02-23  \n",
       "5   842cb861-b234-49e3-a29c-3eb1533d5d2a  dummy    dummy  2022-02-23  \n",
       "6   d0695897-e749-4050-9034-20079ef38a85  dummy    dummy  2022-02-23  \n",
       "7   16ac7507-9401-4791-8c8c-c653e8852bc5  dummy    dummy  2022-02-23  \n",
       "8   e23f7b7b-8491-4548-ae27-ce6a088ebb6e  dummy    dummy  2022-02-23  \n",
       "9   76540bb7-6915-4adc-bd4f-353acd38c4ae  dummy    dummy  2022-02-23  \n",
       "10  b623c611-8225-45a6-a0b5-e6eacb7896b8  dummy    dummy  2022-02-23  \n",
       "11  1e60ef76-3006-4d6e-86af-427f308cdcf4  dummy    dummy  2022-02-23  \n",
       "12  496ec571-b9c9-43af-8664-db95878ea270  dummy    dummy  2022-02-23  \n",
       "13  89aad415-392d-47cc-879f-3637b73db7f9  dummy    dummy  2022-02-23  \n",
       "14  5d546f41-3994-4cfe-9158-869a4338f26f  dummy    dummy  2022-02-23  "
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_step(filepath):\n",
    "    return re.sub(r'^.*' + workflow_name + r'.cwl/[^/]*/call-([^/]*)/.*$', r'\\1', filepath)\n",
    "\n",
    "data = []\n",
    "sample_id = 'sample'\n",
    "run_date = str(datetime.datetime.today()).split(' ')[0]\n",
    "run_id, run_uuid = 'dummy', 'dummy'\n",
    "for k, v in m.items():\n",
    "    result_uuid = str(uuid.uuid4())\n",
    "    data.append([sample_id, get_step(v), k, v, 'filesize', result_uuid, run_id, run_uuid, run_date])\n",
    "analysis_summary = pd.DataFrame(data, columns=['sample_id', 'workflow_step', 'result_name', 'data_path',\n",
    "                                               'filesize', 'result_uuid', 'run_id', 'run_uuid', 'run_date'])\n",
    "analysis_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "5327bacd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "c27ce7f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "def get_pipeline_info():\n",
    "    output = subprocess.check_output(('git', 'log', '-1')).decode('utf-8').split('\\n')[0]\n",
    "    commit_id = re.sub(r'^commit\\s(.*)\\s.*$', r'\\1', output)\n",
    "    version = re.sub(r'^commit\\s.*tag:\\s(),\\s.*$', r'\\1', output)\n",
    "    return commit_id, version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89adb409",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_workflow_root(fp):\n",
    "    return re.sub(r'^(.*)', r'\\1', fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "ba869ab0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# pipeline run format\n",
    "# run_id sample_id run_uuid run_date pipeline_name pipeline_version pipeline_git_commit_id run_workflow_root run_input_config_filepath run_log_filepath  batch_runlist_filepath run_input_uuids\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "bd2fd7be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_uuid</th>\n",
       "      <th>run_data</th>\n",
       "      <th>pipeline_name</th>\n",
       "      <th>pipeline_version</th>\n",
       "      <th>pipeline_commit_id</th>\n",
       "      <th>run_root</th>\n",
       "      <th>run_input_config_filepath</th>\n",
       "      <th>run_log_filepath</th>\n",
       "      <th>runlist_filepath</th>\n",
       "      <th>run_input_uuids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dummy</td>\n",
       "      <td>sample</td>\n",
       "      <td>dummy</td>\n",
       "      <td>2022-02-23</td>\n",
       "      <td>pecgs_TN_wxs_fq_T_rna_fq</td>\n",
       "      <td>commit fbf2f35a5f65208364afa7c05552436e9641e4fb</td>\n",
       "      <td>commit fbf2f35a5f65208364afa7c05552436e9641e4fb</td>\n",
       "      <td>root</td>\n",
       "      <td>config_fp</td>\n",
       "      <td>/data/estorrs/sandbox/HT191P1-S1H1A3Y3_419b809...</td>\n",
       "      <td>runlist</td>\n",
       "      <td>concated uuids</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  run_id sample_id run_uuid    run_data             pipeline_name  \\\n",
       "0  dummy    sample    dummy  2022-02-23  pecgs_TN_wxs_fq_T_rna_fq   \n",
       "\n",
       "                                  pipeline_version  \\\n",
       "0  commit fbf2f35a5f65208364afa7c05552436e9641e4fb   \n",
       "\n",
       "                                pipeline_commit_id run_root  \\\n",
       "0  commit fbf2f35a5f65208364afa7c05552436e9641e4fb     root   \n",
       "\n",
       "  run_input_config_filepath  \\\n",
       "0                 config_fp   \n",
       "\n",
       "                                    run_log_filepath runlist_filepath  \\\n",
       "0  /data/estorrs/sandbox/HT191P1-S1H1A3Y3_419b809...          runlist   \n",
       "\n",
       "  run_input_uuids  \n",
       "0  concated uuids  "
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_data = []\n",
    "commit_id, version = get_pipeline_info()\n",
    "run_data.append([run_id, sample_id, run_uuid, run_date, workflow_name, version, commit_id, 'root', 'config_fp', log_fp, 'runlist', 'concated uuids'])\n",
    "run_df = pd.DataFrame(data=run_data,\n",
    "                     columns=['run_id', 'sample_id', 'run_uuid', 'run_data', 'pipeline_name', 'pipeline_version',\n",
    "                             'pipeline_commit_id', 'run_root', 'run_input_config_filepath', 'run_log_filepath', 'runlist_filepath', 'run_input_uuids'])\n",
    "run_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "11071981",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'fb1f4b8a-a56b-4bc6-b059-864c2b066dc8, 80ce8102-e8a4-4529-85b0-701a4dc022ea, eafb3178-e459-44c0-a374-8e17a8107fce, 9923349f-dc3c-40d4-a097-803c4828f732, 78777884-505c-4b2f-9170-107662c1effa, 2699158f-25fe-4f19-a7cf-2a4e1cf687b3'"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uuid_cols = [c for c in run_list.columns if '.uuid' in c]\n",
    "', '.join(run_list.loc['HT191P1-S1H1A3Y3_419b8098-e4b2-4318-883d-c233cb7e06c8', uuid_cols].to_list())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57b7cb66",
   "metadata": {},
   "outputs": [],
   "source": [
    "python ../../src/compute1/generate_run_commands.py summarize-run pecgs_TN_wxs_fq_T_rna_fq run_list.txt /scratch1/fs1/dinglab/estorrs/cromwell-data/pecgs/testing/pecgs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
